--- 
title: "MGS microbiota data analysis based on XMAS package"
author: "Hua Zou"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
always_allow_html: true
description: "MGS microbiota data analysis based on XMAS package"
---

# Prerequisites

Here, we give users one example to practice the MGS microbiota data analysis workflow by **XMAS 2.0**. By the way, we also recommend users handling your own microbiota data in a reasonable manner when you utilize this package or workflow. Pay attention to whether your data fit the methods this package provided.


## Preparing for the tutorial

This tutorial requires that the users should have basic knowledge on R language and statistics. If you are new to microbial research, we recommend the **Statistical Analysis of Microbiome Data with R** [@xia2018statistical] as guide book. 

We suggest users using local laptop because our own server have not deployed **XMAS 2.0** package yet. In addition, installing the following software before using **XMAS 2.0**.

* R 3.6.3 or later release [Download link](https://mirrors.tuna.tsinghua.edu.cn/CRAN/).

* Rstudio Desktop [Download link](https://www.rstudio.com/products/rstudio/download/#download).


## Workflow description

The standard data analysis of 16S and MGS data by **XMAS 2.0**. The upstream process is performed by in-house pipeline. and this example just shows how to perform downstream data analysis. In briefly, the example comprises the following steps:

1. Converting inputs into phyloseq object;

2. Quality Evaluation;

3. Pre-Processing Data;

4. Diversity analysis;

5. Ordination analysis;

6. Composition analysis;

7. Differential analysis.
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.height=8, fig.align="center", fig.cap="Functions of XMAS 2.0 in MGS"}
DiagrammeR::grViz(diagram = "digraph flowchart { 

# initiate graph
graph [layout = dot, rankdir = TB, label = '[dada2_res|metadata|tree]\n\n', labelloc = t]


# global node settings
node [shape = rectangle, style = filled, fillcolor = Linen, fontname = arial, width = 4]
 
tab1 [label = '@@1']  
tab2 [label = '@@2']  
tab3 [label = '@@3']
tab4 [label = '@@4']  
tab5 [label = '@@5']  
tab6 [label = '@@6']
tab7 [label = '@@7']  
tab8 [label = '@@8']  
tab9 [label = '@@9']
tab10 [label = '@@10']  
tab11 [label = '@@11']  
tab12 [label = '@@12']
tab13 [label = '@@13'] 
 
tab1 -> tab2 -> tab3 -> tab4 -> tab5 -> tab6 -> tab7 -> tab8 -> tab9 -> tab10 -> tab11 -> tab12 -> tab13;  
}  

  
[1]: paste0('Importing data: import_dada2()/summarize_phyloseq()')
[2]: paste0('Quality Control(RefCheck/Remove): summarize_taxa()/run_RefCheck()/get_GroupPhyloseq()')
[3]: paste0('Quality Control(Rarefaction curves): plot_RarefCure()')  
[4]: paste0('Quality Control(Reads Track): plot_Dada2Track()')  
[5]: paste0('Quality Control(Variability): plot_taxa_cv()/plot_read_distribution()')  
[6]: paste0('Data processing(Rarefy): normalize()') 
[7]: paste0('Data processing(Extract specific taxonomic level): summarize_taxa()') 
[8]: paste0('Data processing(Count2Relative): summarize_taxa(absolute = FALSE)') 
[9]: paste0('Data processing(Filter&trim): run_filter()/run_trim()') 
[10]: paste0('GlobalView(Alpha diveristy): run_alpha_diversity()/plot_boxplot()') 
[11]: paste0('GlobalView(Beta diveristy): run_beta_diversity()/run_permanova()/run_ordination()/plot_Ordination()') 
[12]: paste0('GlobalView(microbial composition): plot_StackBarPlot()/plot_core_taxa()/core_members()')
[13]: paste0('DifferentialAnalysis: run_lefse()/plot_lefse()/run_wilcox()/plot_volcano()')
")
```
