<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Convert inputdata into phyloseq object | MGS microbiota data analysis based on XMAS package</title>
  <meta name="description" content="Chapter 2 Convert inputdata into phyloseq object | MGS microbiota data analysis based on XMAS package" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Convert inputdata into phyloseq object | MGS microbiota data analysis based on XMAS package" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Chapter 2 Convert inputdata into phyloseq object | MGS microbiota data analysis based on XMAS package" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Convert inputdata into phyloseq object | MGS microbiota data analysis based on XMAS package" />
  
  <meta name="twitter:description" content="Chapter 2 Convert inputdata into phyloseq object | MGS microbiota data analysis based on XMAS package" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2022-06-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="quality-control.html"/>
<script src="libs/header-attrs-2.13/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.21/datatables.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">MGS microbiota data analysis based on XMAS package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-tutorial"><i class="fa fa-check"></i><b>1.1</b> Preparing for the tutorial<span></span></a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#workflow-description"><i class="fa fa-check"></i><b>1.2</b> Workflow description<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html"><i class="fa fa-check"></i><b>2</b> Convert inputdata into phyloseq object<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metaphlan23"><i class="fa fa-check"></i><b>2.1</b> Metaphlan2/3<span></span></a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results"><i class="fa fa-check"></i><b>2.1.1</b> Importing results<span></span></a></li>
<li class="chapter" data-level="2.1.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table"><i class="fa fa-check"></i><b>2.1.2</b> taxa table<span></span></a></li>
<li class="chapter" data-level="2.1.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table"><i class="fa fa-check"></i><b>2.1.3</b> otu table<span></span></a></li>
<li class="chapter" data-level="2.1.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table"><i class="fa fa-check"></i><b>2.1.4</b> metadata table<span></span></a></li>
<li class="chapter" data-level="2.1.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object"><i class="fa fa-check"></i><b>2.1.5</b> phyloseq object<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#transform-the-abundance-of-taxa-whose-relative-abundance-under-limit-of-detection-lod-into-zeros"><i class="fa fa-check"></i><b>2.2</b> Transform the abundance of taxa whose relative abundance under Limit Of Detection (LOD) into Zeros<span></span></a></li>
<li class="chapter" data-level="2.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#systematic-information"><i class="fa fa-check"></i><b>2.3</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>3</b> Quality Control<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="quality-control.html"><a href="quality-control.html#spike-in-sample-brs"><i class="fa fa-check"></i><b>3.1</b> spike-in sample (BRS)<span></span></a></li>
<li class="chapter" data-level="3.2" data-path="quality-control.html"><a href="quality-control.html#spike-in-samples-brs-remove"><i class="fa fa-check"></i><b>3.2</b> Spike-in sample’s (BRS) remove<span></span></a></li>
<li class="chapter" data-level="3.3" data-path="quality-control.html"><a href="quality-control.html#systematic-information-1"><i class="fa fa-check"></i><b>3.3</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data-processing.html"><a href="data-processing.html"><i class="fa fa-check"></i><b>4</b> Data processing<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-processing.html"><a href="data-processing.html#extracting-specific-taxonomic-level"><i class="fa fa-check"></i><b>4.1</b> Extracting specific taxonomic level<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="data-processing.html"><a href="data-processing.html#systematic-information-2"><i class="fa fa-check"></i><b>4.2</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="diversity-analysis.html"><a href="diversity-analysis.html"><i class="fa fa-check"></i><b>5</b> Diversity analysis<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="diversity-analysis.html"><a href="diversity-analysis.html#alpha-diveristy"><i class="fa fa-check"></i><b>5.1</b> Alpha diveristy<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="diversity-analysis.html"><a href="diversity-analysis.html#beta-diversity"><i class="fa fa-check"></i><b>5.2</b> Beta diversity<span></span></a></li>
<li class="chapter" data-level="5.3" data-path="diversity-analysis.html"><a href="diversity-analysis.html#systematic-information-3"><i class="fa fa-check"></i><b>5.3</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html"><i class="fa fa-check"></i><b>6</b> PERMANOVA + beta dispersion + Ordination<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#permanova"><i class="fa fa-check"></i><b>6.1</b> PERMANOVA<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#beta-dispersion"><i class="fa fa-check"></i><b>6.2</b> beta dispersion<span></span></a></li>
<li class="chapter" data-level="6.3" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#ordination"><i class="fa fa-check"></i><b>6.3</b> Ordination<span></span></a></li>
<li class="chapter" data-level="6.4" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#systematic-information-4"><i class="fa fa-check"></i><b>6.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="microbial-composition.html"><a href="microbial-composition.html"><i class="fa fa-check"></i><b>7</b> Microbial composition<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="microbial-composition.html"><a href="microbial-composition.html#stacked-barplot"><i class="fa fa-check"></i><b>7.1</b> Stacked barplot<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="microbial-composition.html"><a href="microbial-composition.html#core-microbiota"><i class="fa fa-check"></i><b>7.2</b> Core microbiota<span></span></a></li>
<li class="chapter" data-level="7.3" data-path="microbial-composition.html"><a href="microbial-composition.html#systematic-information-5"><i class="fa fa-check"></i><b>7.3</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-analysis.html"><a href="differential-analysis.html"><i class="fa fa-check"></i><b>8</b> Differential Analysis<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="differential-analysis.html"><a href="differential-analysis.html#filtering-trimming"><i class="fa fa-check"></i><b>8.1</b> Filtering &amp; Trimming<span></span></a></li>
<li class="chapter" data-level="8.2" data-path="differential-analysis.html"><a href="differential-analysis.html#filtering-the-low-relative-abundance-or-unclassified-taxa-by-the-threshold-total-counts-1e-4"><i class="fa fa-check"></i><b>8.2</b> Filtering the low relative abundance or unclassified taxa by the threshold (total counts &lt; 1e-4)<span></span></a></li>
<li class="chapter" data-level="8.3" data-path="differential-analysis.html"><a href="differential-analysis.html#trimming-the-taxa-with-low-occurrence-less-than-threshold"><i class="fa fa-check"></i><b>8.3</b> Trimming the taxa with low occurrence less than threshold<span></span></a></li>
<li class="chapter" data-level="8.4" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse"><i class="fa fa-check"></i><b>8.4</b> Liner discriminant analysis (LDA) effect size (LEfSe)<span></span></a></li>
<li class="chapter" data-level="8.5" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-test"><i class="fa fa-check"></i><b>8.5</b> Wilcoxon Rank-Sum test<span></span></a></li>
<li class="chapter" data-level="8.6" data-path="differential-analysis.html"><a href="differential-analysis.html#dominant-taxa"><i class="fa fa-check"></i><b>8.6</b> Dominant taxa<span></span></a></li>
<li class="chapter" data-level="8.7" data-path="differential-analysis.html"><a href="differential-analysis.html#multiple-differential-analysis-by-one-function"><i class="fa fa-check"></i><b>8.7</b> Multiple differential analysis by one function<span></span></a></li>
<li class="chapter" data-level="8.8" data-path="differential-analysis.html"><a href="differential-analysis.html#systematic-information-6"><i class="fa fa-check"></i><b>8.8</b> Systematic Information<span></span></a></li>
</ul></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">MGS microbiota data analysis based on XMAS package</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="convert-inputdata-into-phyloseq-object" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Convert inputdata into phyloseq object<a href="convert-inputdata-into-phyloseq-object.html#convert-inputdata-into-phyloseq-object" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><em>r Biocpkg(“phyloseq”)</em> <span class="citation">(<a href="#ref-mcmurdie2013phyloseq" role="doc-biblioref">McMurdie and Holmes 2013</a>)</span> is the most popular <a href="https://bioconductor.org/">Biocondcutor</a> package used by the microbiome research field, and <code>phyloseq-class</code> objects are a great data-standard for microbiota data in R. Therefore, the core functions in <code>XMAS</code> take <code>phyloseq-class</code> object as input. In the phyloseq object, information on OTU abundances, taxonomy of OTUs, the phylogenetic tree and metadata is stored.</p>
<p>This tutorial will introduce you the basic steps to convert results from the in-house pipeline into <em>phyloseq-class object</em>. More importantly on how to look at your data and filter appropriately. We will use the inputs from <strong>/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/</strong>.</p>
<p><strong>Loading packages</strong></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="convert-inputdata-into-phyloseq-object.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(XMAS2)</span>
<span id="cb1-2"><a href="convert-inputdata-into-phyloseq-object.html#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-3"><a href="convert-inputdata-into-phyloseq-object.html#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb1-4"><a href="convert-inputdata-into-phyloseq-object.html#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span></code></pre></div>
<div id="metaphlan23" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Metaphlan2/3<a href="convert-inputdata-into-phyloseq-object.html#metaphlan23" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The result of the in-house <strong><a href="https://huttenhower.sph.harvard.edu/metaphlan2">Metaphlan2/3</a></strong> pipeline.</p>
<div id="importing-results" class="section level3 hasAnchor" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Importing results<a href="convert-inputdata-into-phyloseq-object.html#importing-results" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p>/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/MGS/metaphlan2_merged.tsv</p></li>
<li><p>/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/MGS/metadata.txt</p></li>
</ol>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="convert-inputdata-into-phyloseq-object.html#cb2-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_res <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb2-2"><a href="convert-inputdata-into-phyloseq-object.html#cb2-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system.file</span>(</span>
<span id="cb2-3"><a href="convert-inputdata-into-phyloseq-object.html#cb2-3" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;metaphlan2_merged.tsv&quot;</span>,</span>
<span id="cb2-4"><a href="convert-inputdata-into-phyloseq-object.html#cb2-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">package =</span> <span class="st">&quot;XMAS2&quot;</span></span>
<span id="cb2-5"><a href="convert-inputdata-into-phyloseq-object.html#cb2-5" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb2-6"><a href="convert-inputdata-into-phyloseq-object.html#cb2-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cn">TRUE</span>, </span>
<span id="cb2-7"><a href="convert-inputdata-into-phyloseq-object.html#cb2-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span></span>
<span id="cb2-8"><a href="convert-inputdata-into-phyloseq-object.html#cb2-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-9"><a href="convert-inputdata-into-phyloseq-object.html#cb2-9" aria-hidden="true" tabindex="-1"></a>metaphlan2_sam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb2-10"><a href="convert-inputdata-into-phyloseq-object.html#cb2-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system.file</span>(</span>
<span id="cb2-11"><a href="convert-inputdata-into-phyloseq-object.html#cb2-11" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;metaphlan2_metadata.tsv&quot;</span>,</span>
<span id="cb2-12"><a href="convert-inputdata-into-phyloseq-object.html#cb2-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">package =</span> <span class="st">&quot;XMAS2&quot;</span></span>
<span id="cb2-13"><a href="convert-inputdata-into-phyloseq-object.html#cb2-13" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb2-14"><a href="convert-inputdata-into-phyloseq-object.html#cb2-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb2-15"><a href="convert-inputdata-into-phyloseq-object.html#cb2-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-16"><a href="convert-inputdata-into-phyloseq-object.html#cb2-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span></span>
<span id="cb2-17"><a href="convert-inputdata-into-phyloseq-object.html#cb2-17" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="taxa-table" class="section level3 hasAnchor" number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> taxa table<a href="convert-inputdata-into-phyloseq-object.html#taxa-table" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="convert-inputdata-into-phyloseq-object.html#cb3-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_res_list <span class="ot">&lt;-</span> <span class="fu">import_metaphlan_taxa</span>(metaphlan2_res, <span class="at">taxa_level =</span> <span class="st">&quot;Species&quot;</span>)</span>
<span id="cb3-2"><a href="convert-inputdata-into-phyloseq-object.html#cb3-2" aria-hidden="true" tabindex="-1"></a>tax_tab <span class="ot">&lt;-</span> metaphlan2_res_list<span class="sc">$</span>tax_tab</span>
<span id="cb3-3"><a href="convert-inputdata-into-phyloseq-object.html#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(tax_tab)</span></code></pre></div>
<pre><code>##                                  Kingdom            Phylum             Class              Order              Family          Genus
## s__Actinomyces_graevenitzii  k__Bacteria p__Actinobacteria c__Actinobacteria o__Actinomycetales f__Actinomycetaceae g__Actinomyces
## s__Actinomyces_johnsonii     k__Bacteria p__Actinobacteria c__Actinobacteria o__Actinomycetales f__Actinomycetaceae g__Actinomyces
## s__Actinomyces_massiliensis  k__Bacteria p__Actinobacteria c__Actinobacteria o__Actinomycetales f__Actinomycetaceae g__Actinomyces
## s__Actinomyces_odontolyticus k__Bacteria p__Actinobacteria c__Actinobacteria o__Actinomycetales f__Actinomycetaceae g__Actinomyces
## s__Actinomyces_oris          k__Bacteria p__Actinobacteria c__Actinobacteria o__Actinomycetales f__Actinomycetaceae g__Actinomyces
## s__Actinomyces_viscosus      k__Bacteria p__Actinobacteria c__Actinobacteria o__Actinomycetales f__Actinomycetaceae g__Actinomyces
##                                                   Species
## s__Actinomyces_graevenitzii   s__Actinomyces_graevenitzii
## s__Actinomyces_johnsonii         s__Actinomyces_johnsonii
## s__Actinomyces_massiliensis   s__Actinomyces_massiliensis
## s__Actinomyces_odontolyticus s__Actinomyces_odontolyticus
## s__Actinomyces_oris                   s__Actinomyces_oris
## s__Actinomyces_viscosus           s__Actinomyces_viscosus</code></pre>
</div>
<div id="otu-table" class="section level3 hasAnchor" number="2.1.3">
<h3><span class="header-section-number">2.1.3</span> otu table<a href="convert-inputdata-into-phyloseq-object.html#otu-table" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="convert-inputdata-into-phyloseq-object.html#cb5-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_res_list <span class="ot">&lt;-</span> <span class="fu">import_metaphlan_taxa</span>(metaphlan2_res, <span class="at">taxa_level =</span> <span class="st">&quot;Species&quot;</span>)</span>
<span id="cb5-2"><a href="convert-inputdata-into-phyloseq-object.html#cb5-2" aria-hidden="true" tabindex="-1"></a>otu_tab <span class="ot">&lt;-</span> metaphlan2_res_list<span class="sc">$</span>abu_tab</span>
<span id="cb5-3"><a href="convert-inputdata-into-phyloseq-object.html#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(otu_tab)</span></code></pre></div>
<pre><code>##                                    s1 s2      s3 s4 s5 s6       s7      s8        s9 s10      s11   s12      s13     s14 s15 s16 s17
## s__Actinomyces_graevenitzii  3.58e-05  0 0.0e+00  0  0  0 7.20e-06 0.0e+00 0.0000000   0 1.40e-05 0e+00 0.00e+00 0.0e+00   0   0   0
## s__Actinomyces_johnsonii     0.00e+00  0 0.0e+00  0  0  0 0.00e+00 0.0e+00 0.0000000   0 0.00e+00 0e+00 1.15e-05 0.0e+00   0   0   0
## s__Actinomyces_massiliensis  0.00e+00  0 0.0e+00  0  0  0 2.95e-05 0.0e+00 0.0000000   0 0.00e+00 0e+00 0.00e+00 0.0e+00   0   0   0
## s__Actinomyces_odontolyticus 7.70e-05  0 1.7e-06  0  0  0 2.47e-05 5.8e-06 0.0001175   0 6.16e-05 0e+00 1.66e-05 3.9e-06   0   0   0
## s__Actinomyces_oris          0.00e+00  0 0.0e+00  0  0  0 2.81e-05 0.0e+00 0.0000135   0 0.00e+00 0e+00 0.00e+00 0.0e+00   0   0   0
## s__Actinomyces_viscosus      4.60e-06  0 0.0e+00  0  0  0 0.00e+00 0.0e+00 0.0000039   0 6.20e-06 7e-06 1.10e-06 0.0e+00   0   0   0
##                                  s18      s19 s20 s21      s22 refE
## s__Actinomyces_graevenitzii  6.1e-06 1.60e-06   0   0 0.00e+00    0
## s__Actinomyces_johnsonii     0.0e+00 0.00e+00   0   0 0.00e+00    0
## s__Actinomyces_massiliensis  0.0e+00 0.00e+00   0   0 2.52e-05    0
## s__Actinomyces_odontolyticus 0.0e+00 4.19e-05   0   0 0.00e+00    0
## s__Actinomyces_oris          0.0e+00 0.00e+00   0   0 7.40e-06    0
## s__Actinomyces_viscosus      0.0e+00 0.00e+00   0   0 7.28e-05    0</code></pre>
</div>
<div id="metadata-table" class="section level3 hasAnchor" number="2.1.4">
<h3><span class="header-section-number">2.1.4</span> metadata table<a href="convert-inputdata-into-phyloseq-object.html#metadata-table" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="convert-inputdata-into-phyloseq-object.html#cb7-1" aria-hidden="true" tabindex="-1"></a>sam_tab <span class="ot">&lt;-</span> metaphlan2_sam <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb7-2"><a href="convert-inputdata-into-phyloseq-object.html#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sam_tab)</span></code></pre></div>
<pre><code>##    Group phynotype
## s1    BB      0.00
## s2    AA      2.50
## s3    BB      0.00
## s4    AA      1.25
## s5    AA     30.00
## s6    AA     15.00</code></pre>
</div>
<div id="phyloseq-object" class="section level3 hasAnchor" number="2.1.5">
<h3><span class="header-section-number">2.1.5</span> phyloseq object<a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="convert-inputdata-into-phyloseq-object.html#cb9-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps <span class="ot">&lt;-</span> <span class="fu">get_metaphlan_phyloseq</span>(<span class="at">otu_tab =</span> otu_tab, </span>
<span id="cb9-2"><a href="convert-inputdata-into-phyloseq-object.html#cb9-2" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">sam_tab =</span> sam_tab,</span>
<span id="cb9-3"><a href="convert-inputdata-into-phyloseq-object.html#cb9-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">tax_tab =</span> tax_tab)</span>
<span id="cb9-4"><a href="convert-inputdata-into-phyloseq-object.html#cb9-4" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 326 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 326 taxa by 7 taxonomic ranks ]</code></pre>
<p>we obtain the <strong>phyloseq-class object</strong> and then use it to perform data analysis.</p>
<p>Here, the phyloseq object comprises five components (<em>OTU Table</em>, <em>Sample Data</em>, and <em>Taxonomy Table</em>).</p>
</div>
</div>
<div id="transform-the-abundance-of-taxa-whose-relative-abundance-under-limit-of-detection-lod-into-zeros" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Transform the abundance of taxa whose relative abundance under Limit Of Detection (LOD) into Zeros<a href="convert-inputdata-into-phyloseq-object.html#transform-the-abundance-of-taxa-whose-relative-abundance-under-limit-of-detection-lod-into-zeros" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="convert-inputdata-into-phyloseq-object.html#cb11-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_Species_LOD <span class="ot">&lt;-</span> <span class="fu">aggregate_LOD_taxa</span>(metaphlan2_ps, </span>
<span id="cb11-2"><a href="convert-inputdata-into-phyloseq-object.html#cb11-2" aria-hidden="true" tabindex="-1"></a>                                                <span class="at">taxa_level =</span> <span class="st">&quot;Species&quot;</span>, </span>
<span id="cb11-3"><a href="convert-inputdata-into-phyloseq-object.html#cb11-3" aria-hidden="true" tabindex="-1"></a>                                                <span class="at">cutoff =</span> <span class="fl">1e-04</span>)</span>
<span id="cb11-4"><a href="convert-inputdata-into-phyloseq-object.html#cb11-4" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_Species_LOD</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 192 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 192 taxa by 7 taxonomic ranks ]</code></pre>
</div>
<div id="systematic-information" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Systematic Information<a href="convert-inputdata-into-phyloseq-object.html#systematic-information" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="convert-inputdata-into-phyloseq-object.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.1.2 (2021-11-01)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Monterey 12.2.1
## 
## Matrix products: default
## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] magrittr_2.0.2              glue_1.6.2                  Gmisc_3.0.0                 htmlTable_2.4.0            
##  [5] Rcpp_1.0.8.2                SummarizedExperiment_1.24.0 Biobase_2.54.0              GenomicRanges_1.46.1       
##  [9] GenomeInfoDb_1.30.1         IRanges_2.28.0              S4Vectors_0.32.3            BiocGenerics_0.40.0        
## [13] MatrixGenerics_1.6.0        matrixStats_0.61.0          ggpubr_0.4.0                ggplot2_3.3.5              
## [17] phyloseq_1.38.0             tibble_3.1.6                dplyr_1.0.8                 XMAS2_2.1.3                
## 
## loaded via a namespace (and not attached):
##   [1] backports_1.4.1        Hmisc_4.6-0            plyr_1.8.6             igraph_1.2.11          splines_4.1.2         
##   [6] crosstalk_1.2.0        BiocParallel_1.28.3    TH.data_1.1-0          digest_0.6.29          foreach_1.5.2         
##  [11] htmltools_0.5.2        fansi_1.0.2            checkmate_2.0.0        memoise_2.0.1          cluster_2.1.2         
##  [16] limma_3.50.1           Biostrings_2.62.0      annotate_1.72.0        bayesm_3.1-4           sandwich_3.0-1        
##  [21] metagenomeSeq_1.36.0   jpeg_0.1-9             colorspace_2.0-3       ggrepel_0.9.1          blob_1.2.2            
##  [26] xfun_0.30              crayon_1.5.0           RCurl_1.98-1.6         jsonlite_1.8.0         libcoin_1.0-9         
##  [31] genefilter_1.76.0      survival_3.3-1         zoo_1.8-9              iterators_1.0.14       ape_5.6-2             
##  [36] gtable_0.3.0           zlibbioc_1.40.0        XVector_0.34.0         compositions_2.0-4     DelayedArray_0.20.0   
##  [41] car_3.0-12             Rhdf5lib_1.16.0        shape_1.4.6            DEoptimR_1.0-10        abind_1.4-5           
##  [46] scales_1.1.1           pheatmap_1.0.12        mvtnorm_1.1-3          DBI_1.1.2              edgeR_3.36.0          
##  [51] rstatix_0.7.0          xtable_1.8-4           foreign_0.8-82         bit_4.0.4              Formula_1.2-4         
##  [56] DT_0.21                glmnet_4.1-3           htmlwidgets_1.5.4      httr_1.4.2             DiagrammeR_1.0.9      
##  [61] gplots_3.1.1           RColorBrewer_1.1-2     modeltools_0.2-23      ellipsis_0.3.2         farver_2.1.0          
##  [66] pkgconfig_2.0.3        XML_3.99-0.9           sass_0.4.0             nnet_7.3-17            locfit_1.5-9.5        
##  [71] utf8_1.2.2             labeling_0.4.2         tidyselect_1.1.2       rlang_1.0.2            reshape2_1.4.4        
##  [76] AnnotationDbi_1.56.2   munsell_0.5.0          tools_4.1.2            visNetwork_2.1.0       cachem_1.0.6          
##  [81] cli_3.2.0              generics_0.1.2         RSQLite_2.2.10         ade4_1.7-18            broom_0.7.12          
##  [86] evaluate_0.15          biomformat_1.22.0      stringr_1.4.0          fastmap_1.1.0          yaml_2.3.5            
##  [91] knitr_1.37             bit64_4.0.5            robustbase_0.93-9      caTools_1.18.2         forestplot_2.0.1      
##  [96] purrr_0.3.4            KEGGREST_1.34.0        coin_1.4-2             nlme_3.1-155           compiler_4.1.2        
## [101] rstudioapi_0.13        png_0.1-7              ggsignif_0.6.3         geneplotter_1.72.0     bslib_0.3.1           
## [106] stringi_1.7.6          highr_0.9              lattice_0.20-45        Matrix_1.4-0           tensorA_0.36.2        
## [111] vegan_2.5-7            permute_0.9-7          multtest_2.50.0        vctrs_0.3.8            pillar_1.7.0          
## [116] lifecycle_1.0.1        rhdf5filters_1.6.0     jquerylib_0.1.4        cowplot_1.1.1          data.table_1.14.2     
## [121] bitops_1.0-7           R6_2.5.1               latticeExtra_0.6-29    bookdown_0.24          KernSmooth_2.23-20    
## [126] gridExtra_2.3          codetools_0.2-18       MASS_7.3-55            gtools_3.9.2           assertthat_0.2.1      
## [131] Wrench_1.12.0          rhdf5_2.38.1           DESeq2_1.34.0          withr_2.5.0            multcomp_1.4-18       
## [136] GenomeInfoDbData_1.2.7 mgcv_1.8-39            parallel_4.1.2         rpart_4.1.16           tidyr_1.2.0           
## [141] rmarkdown_2.13         carData_3.0-5          lubridate_1.8.0        base64enc_0.1-3</code></pre>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-mcmurdie2013phyloseq" class="csl-entry">
McMurdie, Paul J, and Susan Holmes. 2013. <span>“Phyloseq: An r Package for Reproducible Interactive Analysis and Graphics of Microbiome Census Data.”</span> <em>PloS One</em> 8 (4): e61217.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="quality-control.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/MGS_DataAnalysis_template/blob/main/01-Load-data.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["MGS microbiota data analysis based on XMAS package.pdf", "MGS microbiota data analysis based on XMAS package.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
