<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Differential Analysis | MGS microbiota data analysis based on XMAS package</title>
  <meta name="description" content="Chapter 8 Differential Analysis | MGS microbiota data analysis based on XMAS package" />
  <meta name="generator" content="bookdown 0.27 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Differential Analysis | MGS microbiota data analysis based on XMAS package" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Chapter 8 Differential Analysis | MGS microbiota data analysis based on XMAS package" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Differential Analysis | MGS microbiota data analysis based on XMAS package" />
  
  <meta name="twitter:description" content="Chapter 8 Differential Analysis | MGS microbiota data analysis based on XMAS package" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2022-08-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="microbial-composition.html"/>
<link rel="next" href="network-analysis.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.21/datatables.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">MGS microbiota data analysis based on XMAS package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-tutorial"><i class="fa fa-check"></i><b>1.1</b> Preparing for the tutorial</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#workflow-description"><i class="fa fa-check"></i><b>1.2</b> Workflow description</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html"><i class="fa fa-check"></i><b>2</b> Convert inputdata into phyloseq object</a>
<ul>
<li class="chapter" data-level="2.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metaphlan23"><i class="fa fa-check"></i><b>2.1</b> Metaphlan2/3</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results"><i class="fa fa-check"></i><b>2.1.1</b> Importing results</a></li>
<li class="chapter" data-level="2.1.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table"><i class="fa fa-check"></i><b>2.1.2</b> taxa table</a></li>
<li class="chapter" data-level="2.1.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table"><i class="fa fa-check"></i><b>2.1.3</b> otu table</a></li>
<li class="chapter" data-level="2.1.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table"><i class="fa fa-check"></i><b>2.1.4</b> metadata table</a></li>
<li class="chapter" data-level="2.1.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object"><i class="fa fa-check"></i><b>2.1.5</b> phyloseq object</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#transform-the-abundance-of-taxa-whose-relative-abundance-under-limit-of-detection-lod-into-zeros"><i class="fa fa-check"></i><b>2.2</b> Transform the abundance of taxa whose relative abundance under Limit Of Detection (LOD) into Zeros</a></li>
<li class="chapter" data-level="2.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#systematic-information"><i class="fa fa-check"></i><b>2.3</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>3</b> Quality Control</a>
<ul>
<li class="chapter" data-level="3.1" data-path="quality-control.html"><a href="quality-control.html#spike-in-sample-brs"><i class="fa fa-check"></i><b>3.1</b> spike-in sample (BRS)</a></li>
<li class="chapter" data-level="3.2" data-path="quality-control.html"><a href="quality-control.html#spike-in-samples-brs-remove"><i class="fa fa-check"></i><b>3.2</b> Spike-in sample’s (BRS) remove</a></li>
<li class="chapter" data-level="3.3" data-path="quality-control.html"><a href="quality-control.html#systematic-information-1"><i class="fa fa-check"></i><b>3.3</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data-processing.html"><a href="data-processing.html"><i class="fa fa-check"></i><b>4</b> Data processing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-processing.html"><a href="data-processing.html#extracting-specific-taxonomic-level"><i class="fa fa-check"></i><b>4.1</b> Extracting specific taxonomic level</a></li>
<li class="chapter" data-level="4.2" data-path="data-processing.html"><a href="data-processing.html#systematic-information-2"><i class="fa fa-check"></i><b>4.2</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="diversity-analysis.html"><a href="diversity-analysis.html"><i class="fa fa-check"></i><b>5</b> Diversity analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="diversity-analysis.html"><a href="diversity-analysis.html#alpha-diveristy"><i class="fa fa-check"></i><b>5.1</b> Alpha diveristy</a></li>
<li class="chapter" data-level="5.2" data-path="diversity-analysis.html"><a href="diversity-analysis.html#beta-diversity"><i class="fa fa-check"></i><b>5.2</b> Beta diversity</a></li>
<li class="chapter" data-level="5.3" data-path="diversity-analysis.html"><a href="diversity-analysis.html#systematic-information-3"><i class="fa fa-check"></i><b>5.3</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html"><i class="fa fa-check"></i><b>6</b> PERMANOVA + beta dispersion + Ordination</a>
<ul>
<li class="chapter" data-level="6.1" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#permanova"><i class="fa fa-check"></i><b>6.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="6.2" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#beta-dispersion"><i class="fa fa-check"></i><b>6.2</b> beta dispersion</a></li>
<li class="chapter" data-level="6.3" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#ordination"><i class="fa fa-check"></i><b>6.3</b> Ordination</a></li>
<li class="chapter" data-level="6.4" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#systematic-information-4"><i class="fa fa-check"></i><b>6.4</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="microbial-composition.html"><a href="microbial-composition.html"><i class="fa fa-check"></i><b>7</b> Microbial composition</a>
<ul>
<li class="chapter" data-level="7.1" data-path="microbial-composition.html"><a href="microbial-composition.html#stacked-barplot"><i class="fa fa-check"></i><b>7.1</b> Stacked barplot</a></li>
<li class="chapter" data-level="7.2" data-path="microbial-composition.html"><a href="microbial-composition.html#core-microbiota"><i class="fa fa-check"></i><b>7.2</b> Core microbiota</a></li>
<li class="chapter" data-level="7.3" data-path="microbial-composition.html"><a href="microbial-composition.html#systematic-information-5"><i class="fa fa-check"></i><b>7.3</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-analysis.html"><a href="differential-analysis.html"><i class="fa fa-check"></i><b>8</b> Differential Analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="differential-analysis.html"><a href="differential-analysis.html#filtering-trimming"><i class="fa fa-check"></i><b>8.1</b> Filtering &amp; Trimming</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="differential-analysis.html"><a href="differential-analysis.html#filtering-the-low-relative-abundance-or-unclassified-taxa-by-the-threshold-total-counts-1e-4"><i class="fa fa-check"></i><b>8.1.1</b> Filtering the low relative abundance or unclassified taxa by the threshold (total counts &lt; 1e-4)</a></li>
<li class="chapter" data-level="8.1.2" data-path="differential-analysis.html"><a href="differential-analysis.html#trimming-the-taxa-with-low-occurrence-less-than-threshold"><i class="fa fa-check"></i><b>8.1.2</b> Trimming the taxa with low occurrence less than threshold</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse"><i class="fa fa-check"></i><b>8.2</b> Liner discriminant analysis (LDA) effect size (LEfSe)</a></li>
<li class="chapter" data-level="8.3" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-test"><i class="fa fa-check"></i><b>8.3</b> Wilcoxon Rank-Sum test</a></li>
<li class="chapter" data-level="8.4" data-path="differential-analysis.html"><a href="differential-analysis.html#dominant-taxa"><i class="fa fa-check"></i><b>8.4</b> Dominant taxa</a></li>
<li class="chapter" data-level="8.5" data-path="differential-analysis.html"><a href="differential-analysis.html#multiple-differential-analysis-by-one-function"><i class="fa fa-check"></i><b>8.5</b> Multiple differential analysis by one function</a></li>
<li class="chapter" data-level="8.6" data-path="differential-analysis.html"><a href="differential-analysis.html#systematic-information-6"><i class="fa fa-check"></i><b>8.6</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="network-analysis.html"><a href="network-analysis.html"><i class="fa fa-check"></i><b>9</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="network-analysis.html"><a href="network-analysis.html#importing-data"><i class="fa fa-check"></i><b>9.1</b> Importing data</a></li>
<li class="chapter" data-level="9.2" data-path="network-analysis.html"><a href="network-analysis.html#single-network-with-pearson-correlation-as-association-measure"><i class="fa fa-check"></i><b>9.2</b> Single network with Pearson correlation as association measure</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="network-analysis.html"><a href="network-analysis.html#building-network-model"><i class="fa fa-check"></i><b>9.2.1</b> Building network model</a></li>
<li class="chapter" data-level="9.2.2" data-path="network-analysis.html"><a href="network-analysis.html#analyzing-the-constructed-network"><i class="fa fa-check"></i><b>9.2.2</b> Analyzing the constructed network</a></li>
<li class="chapter" data-level="9.2.3" data-path="network-analysis.html"><a href="network-analysis.html#visualizing-the-network"><i class="fa fa-check"></i><b>9.2.3</b> Visualizing the network</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="network-analysis.html"><a href="network-analysis.html#dissimilarity-based-networks"><i class="fa fa-check"></i><b>9.3</b> Dissimilarity-based Networks</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="network-analysis.html"><a href="network-analysis.html#building-network-module"><i class="fa fa-check"></i><b>9.3.1</b> Building network module</a></li>
<li class="chapter" data-level="9.3.2" data-path="network-analysis.html"><a href="network-analysis.html#analyzing-the-constructed-network-1"><i class="fa fa-check"></i><b>9.3.2</b> Analyzing the constructed network</a></li>
<li class="chapter" data-level="9.3.3" data-path="network-analysis.html"><a href="network-analysis.html#visualizing-the-network-1"><i class="fa fa-check"></i><b>9.3.3</b> Visualizing the network</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="network-analysis.html"><a href="network-analysis.html#systematic-information-7"><i class="fa fa-check"></i><b>9.4</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="test-example.html"><a href="test-example.html"><i class="fa fa-check"></i><b>10</b> Test Example</a>
<ul>
<li class="chapter" data-level="10.1" data-path="test-example.html"><a href="test-example.html#loading-packages"><i class="fa fa-check"></i><b>10.1</b> Loading packages</a></li>
<li class="chapter" data-level="10.2" data-path="test-example.html"><a href="test-example.html#loading-data"><i class="fa fa-check"></i><b>10.2</b> Loading data</a></li>
<li class="chapter" data-level="10.3" data-path="test-example.html"><a href="test-example.html#step1-convert-inputs-into-phyloseq-data"><i class="fa fa-check"></i><b>10.3</b> Step1: Convert inputs into phyloseq data</a></li>
<li class="chapter" data-level="10.4" data-path="test-example.html"><a href="test-example.html#step2-transform-limit-of-detection-lod-into-zeros"><i class="fa fa-check"></i><b>10.4</b> Step2: Transform limit of detection (LOD) into Zeros</a></li>
<li class="chapter" data-level="10.5" data-path="test-example.html"><a href="test-example.html#step3-brs-checking"><i class="fa fa-check"></i><b>10.5</b> Step3: BRS checking</a></li>
<li class="chapter" data-level="10.6" data-path="test-example.html"><a href="test-example.html#step4-extracting-specific-taxonomic-level"><i class="fa fa-check"></i><b>10.6</b> Step4: Extracting specific taxonomic level</a></li>
<li class="chapter" data-level="10.7" data-path="test-example.html"><a href="test-example.html#step5-globalview"><i class="fa fa-check"></i><b>10.7</b> Step5: GlobalView</a></li>
<li class="chapter" data-level="10.8" data-path="test-example.html"><a href="test-example.html#step6-differential-analysis"><i class="fa fa-check"></i><b>10.8</b> Step6: Differential Analysis</a></li>
<li class="chapter" data-level="10.9" data-path="test-example.html"><a href="test-example.html#systematic-information-8"><i class="fa fa-check"></i><b>10.9</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">MGS microbiota data analysis based on XMAS package</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="differential-analysis" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">Chapter 8</span> Differential Analysis<a href="differential-analysis.html#differential-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><strong>Loading packages</strong></p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="differential-analysis.html#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(XMAS2)</span>
<span id="cb59-2"><a href="differential-analysis.html#cb59-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb59-3"><a href="differential-analysis.html#cb59-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb59-4"><a href="differential-analysis.html#cb59-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb59-5"><a href="differential-analysis.html#cb59-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb59-6"><a href="differential-analysis.html#cb59-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span></code></pre></div>
<p>There are more than 10 approaches to perform differential analysis. Here, we choose two of them and recommend users going to <strong>Chapter 10</strong> to see more details.</p>
<div id="filtering-trimming" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">8.1</span> Filtering &amp; Trimming<a href="differential-analysis.html#filtering-trimming" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We suggest that filtering taxa with low abundance (the summarized value under cutoff: <code>1e-4</code>) and trimming taxa with low prevalence (default: <code>0.1</code>).</p>
<div id="filtering-the-low-relative-abundance-or-unclassified-taxa-by-the-threshold-total-counts-1e-4" class="section level3 hasAnchor" number="8.1.1">
<h3><span class="header-section-number">8.1.1</span> Filtering the low relative abundance or unclassified taxa by the threshold (total counts &lt; 1e-4)<a href="differential-analysis.html#filtering-the-low-relative-abundance-or-unclassified-taxa-by-the-threshold-total-counts-1e-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>filter by sum relative abundance</li>
</ul>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="differential-analysis.html#cb60-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species_filter <span class="ot">&lt;-</span> <span class="fu">run_filter</span>(<span class="at">ps =</span> metaphlan2_ps_LOD_species_remove_BRS, </span>
<span id="cb60-2"><a href="differential-analysis.html#cb60-2" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">cutoff =</span> <span class="fl">1e-4</span>, </span>
<span id="cb60-3"><a href="differential-analysis.html#cb60-3" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">unclass =</span> <span class="cn">TRUE</span>)</span>
<span id="cb60-4"><a href="differential-analysis.html#cb60-4" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species_filter </span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 171 taxa and 22 samples ]
## sample_data() Sample Data:       [ 22 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 171 taxa by 7 taxonomic ranks ]</code></pre>
<ul>
<li>filter by following two criterion <span class="citation">(<a href="#ref-thingholm2019obese" role="doc-biblioref">Thingholm et al. 2019</a>)</span></li>
</ul>
<blockquote>
<p>Species from taxonomic profiles were retained for further analysis if their mean relative abundance exceeded 0.005
(0.5%) across the dataset with a minimum abundance of 0.05 (5%) in at least one sample and non-zero abundance in at least
60% of samples.</p>
</blockquote>
<ol style="list-style-type: decimal">
<li><p>Mean relative abundance: 0.005;</p></li>
<li><p>Minimum relative abundance: 0.05;</p></li>
</ol>
<p>Here, we use 0.01 (the 1e-4 regarded as 0.01 compared to the Referece because Metaphlan2 data had been divided 100).</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="differential-analysis.html#cb62-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species_filter2 <span class="ot">&lt;-</span> <span class="fu">run_filter2</span>(<span class="at">ps =</span> metaphlan2_ps_LOD_species_remove_BRS, </span>
<span id="cb62-2"><a href="differential-analysis.html#cb62-2" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">cutoff_mean =</span> <span class="fl">1e-04</span>, </span>
<span id="cb62-3"><a href="differential-analysis.html#cb62-3" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">cutoff_one =</span> <span class="fl">1e-03</span>, </span>
<span id="cb62-4"><a href="differential-analysis.html#cb62-4" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">unclass =</span> <span class="cn">TRUE</span>)</span>
<span id="cb62-5"><a href="differential-analysis.html#cb62-5" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species_filter2</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 144 taxa and 22 samples ]
## sample_data() Sample Data:       [ 22 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 144 taxa by 7 taxonomic ranks ]</code></pre>
</div>
<div id="trimming-the-taxa-with-low-occurrence-less-than-threshold" class="section level3 hasAnchor" number="8.1.2">
<h3><span class="header-section-number">8.1.2</span> Trimming the taxa with low occurrence less than threshold<a href="differential-analysis.html#trimming-the-taxa-with-low-occurrence-less-than-threshold" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="differential-analysis.html#cb64-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species_filter_trim <span class="ot">&lt;-</span> <span class="fu">run_trim</span>(<span class="at">object =</span> metaphlan2_ps_species_filter, </span>
<span id="cb64-2"><a href="differential-analysis.html#cb64-2" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">cutoff =</span> <span class="fl">0.1</span>, </span>
<span id="cb64-3"><a href="differential-analysis.html#cb64-3" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">trim =</span> <span class="st">&quot;feature&quot;</span>)</span>
<span id="cb64-4"><a href="differential-analysis.html#cb64-4" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species_filter_trim</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 149 taxa and 22 samples ]
## sample_data() Sample Data:       [ 22 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 149 taxa by 7 taxonomic ranks ]</code></pre>
</div>
</div>
<div id="liner-discriminant-analysis-lda-effect-size-lefse" class="section level2 hasAnchor" number="8.2">
<h2><span class="header-section-number">8.2</span> Liner discriminant analysis (LDA) effect size (LEfSe)<a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="differential-analysis.html#cb66-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb66-2"><a href="differential-analysis.html#cb66-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">ps =</span> metaphlan2_ps_species_filter_trim,</span>
<span id="cb66-3"><a href="differential-analysis.html#cb66-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb66-4"><a href="differential-analysis.html#cb66-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb66-5"><a href="differential-analysis.html#cb66-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">norm =</span> <span class="st">&quot;CPM&quot;</span>,</span>
<span id="cb66-6"><a href="differential-analysis.html#cb66-6" aria-hidden="true" tabindex="-1"></a>                          <span class="at">Lda =</span> <span class="dv">2</span>)</span>
<span id="cb66-7"><a href="differential-analysis.html#cb66-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(metaphlan2_ps_lefse)</span></code></pre></div>
<pre><code>##                                   TaxaID         Block Enrichment LDA_Score EffectSize Log2FoldChange (Median)\nAA_vs_BB
## 1        s__Bacteroides_thetaiotaomicron 7_AA vs 15_BB         AA -4.408655   4.175519                          5.431276
## 2               s__Megamonas_rupellensis 7_AA vs 15_BB         AA -2.817746   2.691235                                NA
## 3                   s__Eggerthella_lenta 7_AA vs 15_BB         BB  2.892101   1.352461                                NA
## 4         s__Adlercreutzia_equolifaciens 7_AA vs 15_BB         BB  2.895174   1.731109                                NA
## 5 s__Lachnospiraceae_bacterium_8_1_57FAA 7_AA vs 15_BB         BB  2.922955   2.092586                                NA
## 6                  s__Ruminococcus_obeum 7_AA vs 15_BB         BB  3.670856   2.984520                         -5.013357
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All)
## 1                2857.052           46504.7952            1077.7570                        3.206709            23987.6919
## 2                   0.000             693.6793               0.0000                        5.351556             1074.4770
## 3                   0.000               0.0000             196.5372                       -3.050449              195.4140
## 4                   0.000               0.0000               0.0000                              NA              902.8222
## 5                   0.000               0.0000               0.0000                              NA              778.7779
## 6                8029.311             354.0913           11436.3153                       -2.582014             9772.4021
##   Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB
## 1          61188.062         6627.51904                    86.36                100.00                 80.00
## 2           3208.536           78.58289                    27.27                 57.14                 13.33
## 3             32.750          271.32387                    45.45                 14.29                 60.00
## 4              0.000         1324.13920                    31.82                  0.00                 46.67
## 5              0.000         1142.20756                    31.82                  0.00                 46.67
## 6           2220.627        13296.56382                    90.91                 71.43                100.00
##   Odds Ratio (95% CI)
## 1    0.061 (-5.4;5.5)
## 2      0.001 (-13;13)
## 3         15 (20;9.7)
## 4                &lt;NA&gt;
## 5                &lt;NA&gt;
## 6          19 (24;13)</code></pre>
<ul>
<li>Visualization</li>
</ul>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="differential-analysis.html#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="co"># # don&#39;t run this code when you do lefse in reality</span></span>
<span id="cb68-2"><a href="differential-analysis.html#cb68-2" aria-hidden="true" tabindex="-1"></a><span class="co"># metaphlan2_ps_lefse$LDA_Score &lt;- metaphlan2_ps_lefse$LDA_Score * 1000</span></span>
<span id="cb68-3"><a href="differential-analysis.html#cb68-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_lefse</span>(</span>
<span id="cb68-4"><a href="differential-analysis.html#cb68-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">da_res =</span> metaphlan2_ps_lefse,</span>
<span id="cb68-5"><a href="differential-analysis.html#cb68-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;LDA_Score&quot;</span>,</span>
<span id="cb68-6"><a href="differential-analysis.html#cb68-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="dv">1</span>,</span>
<span id="cb68-7"><a href="differential-analysis.html#cb68-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-63"></span>
<img src="MGS-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-63-1.png" alt="Lefse analysis" width="768" />
<p class="caption">
Figure 8.1: Lefse analysis
</p>
</div>
</div>
<div id="wilcoxon-rank-sum-test" class="section level2 hasAnchor" number="8.3">
<h2><span class="header-section-number">8.3</span> Wilcoxon Rank-Sum test<a href="differential-analysis.html#wilcoxon-rank-sum-test" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="differential-analysis.html#cb69-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb69-2"><a href="differential-analysis.html#cb69-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">ps =</span> metaphlan2_ps_species_filter_trim,</span>
<span id="cb69-3"><a href="differential-analysis.html#cb69-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb69-4"><a href="differential-analysis.html#cb69-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb69-5"><a href="differential-analysis.html#cb69-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-6"><a href="differential-analysis.html#cb69-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(metaphlan2_ps_wilcox)</span></code></pre></div>
<pre><code>##                           TaxaID         Block Enrichment EffectSize Statistic     Pvalue AdjustedPvalue
## 1  s__Acidaminococcus_fermentans 7_AA vs 15_BB  Nonsignif  0.1947240      49.0 0.75333110      0.9276557
## 2   s__Acidaminococcus_intestini 7_AA vs 15_BB  Nonsignif  0.1532475      54.0 0.90599343      0.9925957
## 3 s__Adlercreutzia_equolifaciens 7_AA vs 15_BB  Nonsignif  1.7460390      28.0 0.04076802      0.3973969
## 4        s__Alistipes_finegoldii 7_AA vs 15_BB  Nonsignif  0.2905269      75.0 0.08188173      0.5131384
## 5      s__Alistipes_indistinctus 7_AA vs 15_BB  Nonsignif  0.3410272      68.0 0.17792767      0.5131384
## 6       s__Alistipes_onderdonkii 7_AA vs 15_BB  Nonsignif  0.2965909      66.5 0.29869756      0.5931886
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1                                NA                       0            0.0000000                    0
## 2                                NA                       0            0.0000000                    0
## 3                                NA                       0            0.0000000                    0
## 4                                NA                       0            0.0027362                    0
## 5                                NA                       0            0.0000000                    0
## 6                                NA                       0            0.0002823                    0
##   Log2FoldChange (Rank)\nAA_vs_BB Mean Rank Abundance\nAA Mean Rank Abundance\nBB Occurrence (100%)\n(All)
## 1                     -0.09269949                   11.00                   11.73                    18.18
## 2                      0.03870725                   11.71                   11.40                    13.64
## 3                     -0.71479501                    8.00                   13.13                    31.82
## 4                      0.55679725                   14.71                   10.00                    40.91
## 5                      0.38896713                   13.71                   10.47                    27.27
## 6                      0.35298403                   13.50                   10.57                    45.45
##   Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                 14.29                 20.00      1.5 (2.4;0.69)
## 2                 14.29                 13.33    0.62 (-0.32;1.6)
## 3                  0.00                 46.67                &lt;NA&gt;
## 4                 57.14                 33.33     0.0025 (-12;12)
## 5                 42.86                 20.00   0.0085 (-9.3;9.4)
## 6                 57.14                 40.00     0.37 (-1.6;2.3)</code></pre>
<ul>
<li>Volcano</li>
</ul>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="differential-analysis.html#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_volcano</span>(</span>
<span id="cb71-2"><a href="differential-analysis.html#cb71-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">da_res =</span> metaphlan2_ps_wilcox,</span>
<span id="cb71-3"><a href="differential-analysis.html#cb71-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb71-4"><a href="differential-analysis.html#cb71-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;Log2FoldChange (Rank)</span><span class="sc">\n</span><span class="st">AA_vs_BB&quot;</span>,</span>
<span id="cb71-5"><a href="differential-analysis.html#cb71-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb71-6"><a href="differential-analysis.html#cb71-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb71-7"><a href="differential-analysis.html#cb71-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb71-8"><a href="differential-analysis.html#cb71-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb71-9"><a href="differential-analysis.html#cb71-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">topN =</span> <span class="dv">4</span>,</span>
<span id="cb71-10"><a href="differential-analysis.html#cb71-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">taxa_name =</span> <span class="st">&quot;s__Megamonas_rupellensis&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-65"></span>
<img src="MGS-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-65-1.png" alt="Wilcoxon Rank-Sum test" width="768" />
<p class="caption">
Figure 8.2: Wilcoxon Rank-Sum test
</p>
</div>
</div>
<div id="dominant-taxa" class="section level2 hasAnchor" number="8.4">
<h2><span class="header-section-number">8.4</span> Dominant taxa<a href="differential-analysis.html#dominant-taxa" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Display the significant taxa with selection using boxplot.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="differential-analysis.html#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_topN_boxplot</span>(</span>
<span id="cb72-2"><a href="differential-analysis.html#cb72-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">ps =</span> metaphlan2_ps_species_filter_trim,</span>
<span id="cb72-3"><a href="differential-analysis.html#cb72-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">da_res =</span> metaphlan2_ps_wilcox,</span>
<span id="cb72-4"><a href="differential-analysis.html#cb72-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;Log2FoldChange (Rank)</span><span class="sc">\n</span><span class="st">AA_vs_BB&quot;</span>,</span>
<span id="cb72-5"><a href="differential-analysis.html#cb72-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb72-6"><a href="differential-analysis.html#cb72-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb72-7"><a href="differential-analysis.html#cb72-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb72-8"><a href="differential-analysis.html#cb72-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">topN =</span> <span class="dv">4</span>,</span>
<span id="cb72-9"><a href="differential-analysis.html#cb72-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-66"></span>
<img src="MGS-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-66-1.png" alt="Dominant Taxa" width="1344" />
<p class="caption">
Figure 8.3: Dominant Taxa
</p>
</div>
</div>
<div id="multiple-differential-analysis-by-one-function" class="section level2 hasAnchor" number="8.5">
<h2><span class="header-section-number">8.5</span> Multiple differential analysis by one function<a href="differential-analysis.html#multiple-differential-analysis-by-one-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Here, we provide the <code>run_multiple_da</code> for obtaining the results list from multiple differential analysis methods.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="differential-analysis.html#cb73-1" aria-hidden="true" tabindex="-1"></a>multiple_res <span class="ot">&lt;-</span> <span class="fu">run_multiple_da</span>(</span>
<span id="cb73-2"><a href="differential-analysis.html#cb73-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">ps =</span> metaphlan2_ps_species_filter_trim,</span>
<span id="cb73-3"><a href="differential-analysis.html#cb73-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb73-4"><a href="differential-analysis.html#cb73-4" aria-hidden="true" tabindex="-1"></a>                   <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb73-5"><a href="differential-analysis.html#cb73-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">da_method =</span> <span class="fu">c</span>(<span class="st">&quot;wilcox&quot;</span>, <span class="st">&quot;limma_voom&quot;</span>, <span class="st">&quot;ttest&quot;</span>))</span>
<span id="cb73-6"><a href="differential-analysis.html#cb73-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-7"><a href="differential-analysis.html#cb73-7" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(multiple_res)</span></code></pre></div>
<pre><code>## [1] &quot;wilcox&quot;     &quot;limma_voom&quot; &quot;ttest&quot;</code></pre>
<ul>
<li>plot results</li>
</ul>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="differential-analysis.html#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_multiple_DA</span>(</span>
<span id="cb75-2"><a href="differential-analysis.html#cb75-2" aria-hidden="true" tabindex="-1"></a>      <span class="at">Multip_DA_res =</span> multiple_res,</span>
<span id="cb75-3"><a href="differential-analysis.html#cb75-3" aria-hidden="true" tabindex="-1"></a>      <span class="at">x_index_list =</span> <span class="fu">c</span>(<span class="st">&quot;Log2FoldChange (Rank)</span><span class="sc">\n</span><span class="st">AA_vs_BB&quot;</span>, </span>
<span id="cb75-4"><a href="differential-analysis.html#cb75-4" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&quot;logFC&quot;</span>, </span>
<span id="cb75-5"><a href="differential-analysis.html#cb75-5" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&quot;Log2FoldChange (geometricmean)</span><span class="sc">\n</span><span class="st">AA_vs_BB&quot;</span>),</span>
<span id="cb75-6"><a href="differential-analysis.html#cb75-6" aria-hidden="true" tabindex="-1"></a>      <span class="at">x_index_cutoff =</span> <span class="dv">0</span>,  </span>
<span id="cb75-7"><a href="differential-analysis.html#cb75-7" aria-hidden="true" tabindex="-1"></a>      <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb75-8"><a href="differential-analysis.html#cb75-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">y_index_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb75-9"><a href="differential-analysis.html#cb75-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">cellwidth =</span> <span class="dv">35</span>,</span>
<span id="cb75-10"><a href="differential-analysis.html#cb75-10" aria-hidden="true" tabindex="-1"></a>      <span class="at">cellheight =</span> <span class="dv">10</span>,</span>
<span id="cb75-11"><a href="differential-analysis.html#cb75-11" aria-hidden="true" tabindex="-1"></a>      <span class="at">fontsize_number =</span> <span class="dv">15</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-68"></span>
<img src="MGS-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-68-1.png" alt="Multiple DA results" width="960" />
<p class="caption">
Figure 8.4: Multiple DA results
</p>
</div>
</div>
<div id="systematic-information-6" class="section level2 hasAnchor" number="8.6">
<h2><span class="header-section-number">8.6</span> Systematic Information<a href="differential-analysis.html#systematic-information-6" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="differential-analysis.html#cb76-1" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ──────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.1.2 (2021-11-01)
##  os       macOS Monterey 12.2.1
##  system   x86_64, darwin17.0
##  ui       RStudio
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       Asia/Shanghai
##  date     2022-08-09
##  rstudio  2022.07.1+554 Spotted Wakerobin (desktop)
##  pandoc   2.18 @ /Applications/RStudio.app/Contents/MacOS/quarto/bin/tools/ (via rmarkdown)
## 
## ─ Packages ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version  date (UTC) lib source
##  abind                  1.4-5    2016-07-21 [1] CRAN (R 4.1.0)
##  ade4                   1.7-18   2021-09-16 [1] CRAN (R 4.1.0)
##  ALDEx2                 1.26.0   2021-10-26 [1] Bioconductor
##  annotate               1.72.0   2021-10-26 [1] Bioconductor
##  AnnotationDbi          1.56.2   2021-11-09 [1] Bioconductor
##  ape                    5.6-2    2022-03-02 [1] CRAN (R 4.1.2)
##  assertthat             0.2.1    2019-03-21 [1] CRAN (R 4.1.0)
##  backports              1.4.1    2021-12-13 [1] CRAN (R 4.1.0)
##  base64enc              0.1-3    2015-07-28 [1] CRAN (R 4.1.0)
##  bayesm                 3.1-4    2019-10-15 [1] CRAN (R 4.1.0)
##  Biobase                2.54.0   2021-10-26 [1] Bioconductor
##  BiocGenerics           0.40.0   2021-10-26 [1] Bioconductor
##  BiocParallel           1.28.3   2021-12-09 [1] Bioconductor
##  biomformat             1.22.0   2021-10-26 [1] Bioconductor
##  Biostrings             2.62.0   2021-10-26 [1] Bioconductor
##  bit                    4.0.4    2020-08-04 [1] CRAN (R 4.1.0)
##  bit64                  4.0.5    2020-08-30 [1] CRAN (R 4.1.0)
##  bitops                 1.0-7    2021-04-24 [1] CRAN (R 4.1.0)
##  blob                   1.2.2    2021-07-23 [1] CRAN (R 4.1.0)
##  bookdown               0.27     2022-06-14 [1] CRAN (R 4.1.2)
##  brio                   1.1.3    2021-11-30 [1] CRAN (R 4.1.0)
##  broom                  0.7.12   2022-01-28 [1] CRAN (R 4.1.2)
##  bslib                  0.3.1    2021-10-06 [1] CRAN (R 4.1.0)
##  cachem                 1.0.6    2021-08-19 [1] CRAN (R 4.1.0)
##  callr                  3.7.0    2021-04-20 [1] CRAN (R 4.1.0)
##  car                    3.0-12   2021-11-06 [1] CRAN (R 4.1.0)
##  carData                3.0-5    2022-01-06 [1] CRAN (R 4.1.2)
##  caTools                1.18.2   2021-03-28 [1] CRAN (R 4.1.0)
##  cccd                   1.6      2022-04-08 [1] CRAN (R 4.1.2)
##  cellranger             1.1.0    2016-07-27 [1] CRAN (R 4.1.0)
##  checkmate              2.0.0    2020-02-06 [1] CRAN (R 4.1.0)
##  cli                    3.3.0    2022-04-25 [1] CRAN (R 4.1.2)
##  cluster                2.1.2    2021-04-17 [1] CRAN (R 4.1.2)
##  codetools              0.2-18   2020-11-04 [1] CRAN (R 4.1.2)
##  coin                   1.4-2    2021-10-08 [1] CRAN (R 4.1.0)
##  colorspace             2.0-3    2022-02-21 [1] CRAN (R 4.1.2)
##  compositions           2.0-4    2022-01-05 [1] CRAN (R 4.1.2)
##  conflicted             1.1.0    2021-11-26 [1] CRAN (R 4.1.0)
##  corpcor                1.6.10   2021-09-16 [1] CRAN (R 4.1.0)
##  corrplot               0.92     2021-11-18 [1] CRAN (R 4.1.0)
##  cowplot                1.1.1    2020-12-30 [1] CRAN (R 4.1.0)
##  crayon                 1.5.0    2022-02-14 [1] CRAN (R 4.1.2)
##  crosstalk              1.2.0    2021-11-04 [1] CRAN (R 4.1.0)
##  data.table             1.14.2   2021-09-27 [1] CRAN (R 4.1.0)
##  DBI                    1.1.2    2021-12-20 [1] CRAN (R 4.1.0)
##  DelayedArray           0.20.0   2021-10-26 [1] Bioconductor
##  deldir                 1.0-6    2021-10-23 [1] CRAN (R 4.1.0)
##  DEoptimR               1.0-10   2022-01-03 [1] CRAN (R 4.1.2)
##  desc                   1.4.1    2022-03-06 [1] CRAN (R 4.1.2)
##  DESeq2                 1.34.0   2021-10-26 [1] Bioconductor
##  devtools               2.4.3    2021-11-30 [1] CRAN (R 4.1.0)
##  digest                 0.6.29   2021-12-01 [1] CRAN (R 4.1.0)
##  doParallel             1.0.17   2022-02-07 [1] CRAN (R 4.1.2)
##  doSNOW                 1.0.20   2022-02-04 [1] CRAN (R 4.1.2)
##  dplyr                * 1.0.8    2022-02-08 [1] CRAN (R 4.1.2)
##  DT                     0.21     2022-02-26 [1] CRAN (R 4.1.2)
##  dynamicTreeCut         1.63-1   2016-03-11 [1] CRAN (R 4.1.0)
##  edgeR                  3.36.0   2021-10-26 [1] Bioconductor
##  ellipsis               0.3.2    2021-04-29 [1] CRAN (R 4.1.0)
##  evaluate               0.15     2022-02-18 [1] CRAN (R 4.1.2)
##  fansi                  1.0.2    2022-01-14 [1] CRAN (R 4.1.2)
##  farver                 2.1.0    2021-02-28 [1] CRAN (R 4.1.0)
##  fastcluster            1.2.3    2021-05-24 [1] CRAN (R 4.1.0)
##  fastmap                1.1.0    2021-01-25 [1] CRAN (R 4.1.0)
##  fdrtool                1.2.17   2021-11-13 [1] CRAN (R 4.1.0)
##  filematrix             1.3      2018-02-27 [1] CRAN (R 4.1.0)
##  FNN                    1.1.3    2019-02-15 [1] CRAN (R 4.1.0)
##  foreach                1.5.2    2022-02-02 [1] CRAN (R 4.1.2)
##  foreign                0.8-82   2022-01-13 [1] CRAN (R 4.1.2)
##  forestplot             2.0.1    2021-09-03 [1] CRAN (R 4.1.0)
##  Formula                1.2-4    2020-10-16 [1] CRAN (R 4.1.0)
##  fs                     1.5.2    2021-12-08 [1] CRAN (R 4.1.0)
##  genefilter             1.76.0   2021-10-26 [1] Bioconductor
##  geneplotter            1.72.0   2021-10-26 [1] Bioconductor
##  generics               0.1.2    2022-01-31 [1] CRAN (R 4.1.2)
##  GenomeInfoDb           1.30.1   2022-01-30 [1] Bioconductor
##  GenomeInfoDbData       1.2.7    2022-03-09 [1] Bioconductor
##  GenomicRanges          1.46.1   2021-11-18 [1] Bioconductor
##  ggplot2              * 3.3.5    2021-06-25 [1] CRAN (R 4.1.0)
##  ggpubr               * 0.4.0    2020-06-27 [1] CRAN (R 4.1.0)
##  ggrepel                0.9.1    2021-01-15 [1] CRAN (R 4.1.0)
##  ggsignif               0.6.3    2021-09-09 [1] CRAN (R 4.1.0)
##  glasso                 1.11     2019-10-01 [1] CRAN (R 4.1.0)
##  glmnet                 4.1-3    2021-11-02 [1] CRAN (R 4.1.0)
##  glue                 * 1.6.2    2022-02-24 [1] CRAN (R 4.1.2)
##  Gmisc                * 3.0.0    2022-01-03 [1] CRAN (R 4.1.2)
##  GO.db                  3.14.0   2022-04-11 [1] Bioconductor
##  gplots                 3.1.1    2020-11-28 [1] CRAN (R 4.1.0)
##  gridExtra              2.3      2017-09-09 [1] CRAN (R 4.1.0)
##  gtable                 0.3.0    2019-03-25 [1] CRAN (R 4.1.0)
##  gtools                 3.9.2    2021-06-06 [1] CRAN (R 4.1.0)
##  highr                  0.9      2021-04-16 [1] CRAN (R 4.1.0)
##  Hmisc                  4.6-0    2021-10-07 [1] CRAN (R 4.1.0)
##  htmlTable            * 2.4.0    2022-01-04 [1] CRAN (R 4.1.2)
##  htmltools              0.5.2    2021-08-25 [1] CRAN (R 4.1.0)
##  htmlwidgets            1.5.4    2021-09-08 [1] CRAN (R 4.1.0)
##  httr                   1.4.2    2020-07-20 [1] CRAN (R 4.1.0)
##  huge                   1.3.5    2021-06-30 [1] CRAN (R 4.1.0)
##  igraph                 1.2.11   2022-01-04 [1] CRAN (R 4.1.2)
##  impute                 1.68.0   2021-10-26 [1] Bioconductor
##  IRanges                2.28.0   2021-10-26 [1] Bioconductor
##  irlba                  2.3.5    2021-12-06 [1] CRAN (R 4.1.0)
##  iterators              1.0.14   2022-02-05 [1] CRAN (R 4.1.2)
##  jpeg                   0.1-9    2021-07-24 [1] CRAN (R 4.1.0)
##  jquerylib              0.1.4    2021-04-26 [1] CRAN (R 4.1.0)
##  jsonlite               1.8.0    2022-02-22 [1] CRAN (R 4.1.2)
##  KEGGREST               1.34.0   2021-10-26 [1] Bioconductor
##  KernSmooth             2.23-20  2021-05-03 [1] CRAN (R 4.1.2)
##  knitr                  1.39     2022-04-26 [1] CRAN (R 4.1.2)
##  labeling               0.4.2    2020-10-20 [1] CRAN (R 4.1.0)
##  lattice                0.20-45  2021-09-22 [1] CRAN (R 4.1.2)
##  latticeExtra           0.6-29   2019-12-19 [1] CRAN (R 4.1.0)
##  lavaan                 0.6-11   2022-03-31 [1] CRAN (R 4.1.2)
##  libcoin                1.0-9    2021-09-27 [1] CRAN (R 4.1.0)
##  lifecycle              1.0.1    2021-09-24 [1] CRAN (R 4.1.0)
##  limma                  3.50.1   2022-02-17 [1] Bioconductor
##  locfit                 1.5-9.5  2022-03-03 [1] CRAN (R 4.1.2)
##  lubridate              1.8.0    2021-10-07 [1] CRAN (R 4.1.0)
##  magrittr             * 2.0.2    2022-01-26 [1] CRAN (R 4.1.2)
##  MASS                   7.3-55   2022-01-13 [1] CRAN (R 4.1.2)
##  Matrix                 1.4-0    2021-12-08 [1] CRAN (R 4.1.0)
##  MatrixGenerics         1.6.0    2021-10-26 [1] Bioconductor
##  matrixStats            0.61.0   2021-09-17 [1] CRAN (R 4.1.0)
##  mbzinb                 0.2      2022-03-16 [1] local
##  memoise                2.0.1    2021-11-26 [1] CRAN (R 4.1.0)
##  metagenomeSeq          1.36.0   2021-10-26 [1] Bioconductor
##  mgcv                   1.8-39   2022-02-24 [1] CRAN (R 4.1.2)
##  mixedCCA               1.5.2    2022-07-14 [1] Github (irinagain/mixedCCA@c6d41a3)
##  mnormt                 2.0.2    2020-09-01 [1] CRAN (R 4.1.0)
##  modeltools             0.2-23   2020-03-05 [1] CRAN (R 4.1.0)
##  multcomp               1.4-18   2022-01-04 [1] CRAN (R 4.1.2)
##  multtest               2.50.0   2021-10-26 [1] Bioconductor
##  munsell                0.5.0    2018-06-12 [1] CRAN (R 4.1.0)
##  mvtnorm                1.1-3    2021-10-08 [1] CRAN (R 4.1.0)
##  NADA                   1.6-1.1  2020-03-22 [1] CRAN (R 4.1.0)
##  NetCoMi              * 1.0.3    2022-07-14 [1] Github (stefpeschel/NetCoMi@d4d80d3)
##  nlme                   3.1-155  2022-01-13 [1] CRAN (R 4.1.2)
##  nnet                   7.3-17   2022-01-13 [1] CRAN (R 4.1.2)
##  pbapply                1.5-0    2021-09-16 [1] CRAN (R 4.1.0)
##  pbivnorm               0.6.0    2015-01-23 [1] CRAN (R 4.1.0)
##  pcaPP                  1.9-74   2021-04-23 [1] CRAN (R 4.1.0)
##  permute                0.9-7    2022-01-27 [1] CRAN (R 4.1.2)
##  pheatmap               1.0.12   2019-01-04 [1] CRAN (R 4.1.0)
##  phyloseq             * 1.38.0   2021-10-26 [1] Bioconductor
##  pillar                 1.7.0    2022-02-01 [1] CRAN (R 4.1.2)
##  pkgbuild               1.3.1    2021-12-20 [1] CRAN (R 4.1.0)
##  pkgconfig              2.0.3    2019-09-22 [1] CRAN (R 4.1.0)
##  pkgload                1.2.4    2021-11-30 [1] CRAN (R 4.1.0)
##  plyr                   1.8.6    2020-03-03 [1] CRAN (R 4.1.0)
##  png                    0.1-7    2013-12-03 [1] CRAN (R 4.1.0)
##  preprocessCore         1.56.0   2021-10-26 [1] Bioconductor
##  prettyunits            1.1.1    2020-01-24 [1] CRAN (R 4.1.0)
##  processx               3.5.2    2021-04-30 [1] CRAN (R 4.1.0)
##  proxy                  0.4-26   2021-06-07 [1] CRAN (R 4.1.0)
##  ps                     1.6.0    2021-02-28 [1] CRAN (R 4.1.0)
##  pscl                   1.5.5    2020-03-07 [1] CRAN (R 4.1.0)
##  psych                  2.2.5    2022-05-10 [1] CRAN (R 4.1.2)
##  pulsar                 0.3.7    2020-08-07 [1] CRAN (R 4.1.0)
##  purrr                  0.3.4    2020-04-17 [1] CRAN (R 4.1.0)
##  qgraph                 1.9.2    2022-03-04 [1] CRAN (R 4.1.2)
##  R6                     2.5.1    2021-08-19 [1] CRAN (R 4.1.0)
##  rbibutils              2.2.7    2021-12-07 [1] CRAN (R 4.1.0)
##  RColorBrewer           1.1-2    2014-12-07 [1] CRAN (R 4.1.0)
##  Rcpp                 * 1.0.8.2  2022-03-11 [1] CRAN (R 4.1.2)
##  RcppZiggurat           0.1.6    2020-10-20 [1] CRAN (R 4.1.0)
##  RCurl                  1.98-1.6 2022-02-08 [1] CRAN (R 4.1.2)
##  Rdpack                 2.2      2022-03-19 [1] CRAN (R 4.1.2)
##  readxl               * 1.4.0    2022-03-28 [1] CRAN (R 4.1.2)
##  remotes                2.4.2    2021-11-30 [1] CRAN (R 4.1.0)
##  reshape2               1.4.4    2020-04-09 [1] CRAN (R 4.1.0)
##  Rfast                  2.0.6    2022-02-16 [1] CRAN (R 4.1.2)
##  rhdf5                  2.38.1   2022-03-10 [1] Bioconductor
##  rhdf5filters           1.6.0    2021-10-26 [1] Bioconductor
##  Rhdf5lib               1.16.0   2021-10-26 [1] Bioconductor
##  rlang                  1.0.2    2022-03-04 [1] CRAN (R 4.1.2)
##  rmarkdown              2.14     2022-04-25 [1] CRAN (R 4.1.2)
##  robustbase             0.93-9   2021-09-27 [1] CRAN (R 4.1.0)
##  rootSolve              1.8.2.3  2021-09-29 [1] CRAN (R 4.1.0)
##  rpart                  4.1.16   2022-01-24 [1] CRAN (R 4.1.2)
##  rprojroot              2.0.2    2020-11-15 [1] CRAN (R 4.1.0)
##  RSQLite                2.2.10   2022-02-17 [1] CRAN (R 4.1.2)
##  rstatix                0.7.0    2021-02-13 [1] CRAN (R 4.1.0)
##  rstudioapi             0.13     2020-11-12 [1] CRAN (R 4.1.0)
##  S4Vectors              0.32.3   2021-11-21 [1] Bioconductor
##  sandwich               3.0-1    2021-05-18 [1] CRAN (R 4.1.0)
##  sass                   0.4.0    2021-05-12 [1] CRAN (R 4.1.0)
##  scales                 1.1.1    2020-05-11 [1] CRAN (R 4.1.0)
##  sessioninfo            1.2.2    2021-12-06 [1] CRAN (R 4.1.0)
##  shape                  1.4.6    2021-05-19 [1] CRAN (R 4.1.0)
##  snow                   0.4-4    2021-10-27 [1] CRAN (R 4.1.0)
##  SpiecEasi            * 1.1.2    2022-07-14 [1] Github (zdk123/SpiecEasi@c463727)
##  SPRING                 1.0.4    2022-08-03 [1] Github (GraceYoon/SPRING@3d641a4)
##  stringi                1.7.6    2021-11-29 [1] CRAN (R 4.1.0)
##  stringr                1.4.0    2019-02-10 [1] CRAN (R 4.1.0)
##  SummarizedExperiment   1.24.0   2021-10-26 [1] Bioconductor
##  survival               3.3-1    2022-03-03 [1] CRAN (R 4.1.2)
##  tensorA                0.36.2   2020-11-19 [1] CRAN (R 4.1.0)
##  testthat               3.1.2    2022-01-20 [1] CRAN (R 4.1.2)
##  TH.data                1.1-0    2021-09-27 [1] CRAN (R 4.1.0)
##  tibble               * 3.1.6    2021-11-07 [1] CRAN (R 4.1.0)
##  tidyr                  1.2.0    2022-02-01 [1] CRAN (R 4.1.2)
##  tidyselect             1.1.2    2022-02-21 [1] CRAN (R 4.1.2)
##  tmvnsim                1.0-2    2016-12-15 [1] CRAN (R 4.1.0)
##  truncnorm              1.0-8    2018-02-27 [1] CRAN (R 4.1.0)
##  usethis                2.1.5    2021-12-09 [1] CRAN (R 4.1.0)
##  utf8                   1.2.2    2021-07-24 [1] CRAN (R 4.1.0)
##  vctrs                  0.3.8    2021-04-29 [1] CRAN (R 4.1.0)
##  vegan                  2.5-7    2020-11-28 [1] CRAN (R 4.1.0)
##  VGAM                   1.1-6    2022-02-14 [1] CRAN (R 4.1.2)
##  WGCNA                  1.71     2022-04-22 [1] CRAN (R 4.1.2)
##  withr                  2.5.0    2022-03-03 [1] CRAN (R 4.1.2)
##  Wrench                 1.12.0   2021-10-26 [1] Bioconductor
##  xfun                   0.30     2022-03-02 [1] CRAN (R 4.1.2)
##  XMAS2                * 2.1.7.4  2022-08-09 [1] local
##  XML                    3.99-0.9 2022-02-24 [1] CRAN (R 4.1.2)
##  xtable                 1.8-4    2019-04-21 [1] CRAN (R 4.1.0)
##  XVector                0.34.0   2021-10-26 [1] Bioconductor
##  yaml                   2.3.5    2022-02-21 [1] CRAN (R 4.1.2)
##  zCompositions          1.4.0    2022-01-13 [1] CRAN (R 4.1.2)
##  zlibbioc               1.40.0   2021-10-26 [1] Bioconductor
##  zoo                    1.8-9    2021-03-09 [1] CRAN (R 4.1.0)
## 
##  [1] /Library/Frameworks/R.framework/Versions/4.1/Resources/library
## 
## ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-thingholm2019obese" class="csl-entry">
Thingholm, Louise B, Malte C Rühlemann, Manja Koch, Brie Fuqua, Guido Laucke, Ruwen Boehm, Corinna Bang, et al. 2019. <span>“Obese Individuals with and Without Type 2 Diabetes Show Different Gut Microbial Functional Capacity and Composition.”</span> <em>Cell Host &amp; Microbe</em> 26 (2): 252–64.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="microbial-composition.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="network-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/MGS_DataAnalysis_template/blob/main/07-Differential.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["MGS microbiota data analysis based on XMAS package.pdf", "MGS microbiota data analysis based on XMAS package.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection",
"scroll_highlight": true,
"toc_depth": 3
}
});
});
</script>

</body>

</html>
