<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>MGS microbiota data analysis based on XMAS package</title>
  <meta name="description" content="MGS microbiota data analysis based on XMAS package" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="MGS microbiota data analysis based on XMAS package" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="MGS microbiota data analysis based on XMAS package" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="MGS microbiota data analysis based on XMAS package" />
  
  <meta name="twitter:description" content="MGS microbiota data analysis based on XMAS package" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2022-05-18" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="convert-inputdata-into-phyloseq-object.html"/>
<script src="libs/header-attrs-2.13/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/viz-1.8.2/viz.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/grViz-binding-1.0.9/grViz.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.21/datatables.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">MGS microbiota data analysis based on XMAS package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-tutorial"><i class="fa fa-check"></i><b>1.1</b> Preparing for the tutorial<span></span></a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#workflow-description"><i class="fa fa-check"></i><b>1.2</b> Workflow description<span></span></a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#systematic-information"><i class="fa fa-check"></i><b>1.3</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html"><i class="fa fa-check"></i><b>2</b> Convert inputdata into phyloseq object<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metaphlan23"><i class="fa fa-check"></i><b>2.1</b> Metaphlan2/3<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#summarize-phyloseq-class-object"><i class="fa fa-check"></i><b>2.2</b> Summarize phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="2.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#systematic-information-1"><i class="fa fa-check"></i><b>2.3</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>3</b> Quality Control<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="quality-control.html"><a href="quality-control.html#spike-in-sample-brs"><i class="fa fa-check"></i><b>3.1</b> spike-in sample (BRS)<span></span></a></li>
<li class="chapter" data-level="3.2" data-path="quality-control.html"><a href="quality-control.html#spike-in-samples-remove"><i class="fa fa-check"></i><b>3.2</b> Spike-in sampleâ€™s remove<span></span></a></li>
<li class="chapter" data-level="3.3" data-path="quality-control.html"><a href="quality-control.html#systematic-information-2"><i class="fa fa-check"></i><b>3.3</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data-processing.html"><a href="data-processing.html"><i class="fa fa-check"></i><b>4</b> Data processing<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-processing.html"><a href="data-processing.html#extracting-specific-taxonomic-level"><i class="fa fa-check"></i><b>4.1</b> Extracting specific taxonomic level<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="data-processing.html"><a href="data-processing.html#filtering-the-low-relative-abundance-or-unclassified-taxa-by-the-threshold-total-counts-1e-4"><i class="fa fa-check"></i><b>4.2</b> Filtering the low relative abundance or unclassified taxa by the threshold (total counts &lt; 1e-4)<span></span></a></li>
<li class="chapter" data-level="4.3" data-path="data-processing.html"><a href="data-processing.html#trimming-the-taxa-with-low-occurrence-less-than-threshold"><i class="fa fa-check"></i><b>4.3</b> Trimming the taxa with low occurrence less than threshold<span></span></a></li>
<li class="chapter" data-level="4.4" data-path="data-processing.html"><a href="data-processing.html#systematic-information-3"><i class="fa fa-check"></i><b>4.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="diversity-analysis.html"><a href="diversity-analysis.html"><i class="fa fa-check"></i><b>5</b> Diversity analysis<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="diversity-analysis.html"><a href="diversity-analysis.html#alpha-diveristy"><i class="fa fa-check"></i><b>5.1</b> Alpha diveristy<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="diversity-analysis.html"><a href="diversity-analysis.html#beta-diversity"><i class="fa fa-check"></i><b>5.2</b> Beta diversity<span></span></a></li>
<li class="chapter" data-level="5.3" data-path="diversity-analysis.html"><a href="diversity-analysis.html#systematic-information-4"><i class="fa fa-check"></i><b>5.3</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html"><i class="fa fa-check"></i><b>6</b> PERMANOVA + beta dispersion + Ordination<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#permanova"><i class="fa fa-check"></i><b>6.1</b> PERMANOVA<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#beta-dispersion"><i class="fa fa-check"></i><b>6.2</b> beta dispersion<span></span></a></li>
<li class="chapter" data-level="6.3" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#ordination"><i class="fa fa-check"></i><b>6.3</b> Ordination<span></span></a></li>
<li class="chapter" data-level="6.4" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#systematic-information-5"><i class="fa fa-check"></i><b>6.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="microbial-composition.html"><a href="microbial-composition.html"><i class="fa fa-check"></i><b>7</b> Microbial composition<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="microbial-composition.html"><a href="microbial-composition.html#stacked-barplot"><i class="fa fa-check"></i><b>7.1</b> Stacked barplot<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="microbial-composition.html"><a href="microbial-composition.html#core-microbiota"><i class="fa fa-check"></i><b>7.2</b> Core microbiota<span></span></a></li>
<li class="chapter" data-level="7.3" data-path="microbial-composition.html"><a href="microbial-composition.html#systematic-information-6"><i class="fa fa-check"></i><b>7.3</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-analysis.html"><a href="differential-analysis.html"><i class="fa fa-check"></i><b>8</b> Differential Analysis<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse"><i class="fa fa-check"></i><b>8.1</b> Liner discriminant analysis (LDA) effect size (LEfSe)<span></span></a></li>
<li class="chapter" data-level="8.2" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-test"><i class="fa fa-check"></i><b>8.2</b> Wilcoxon Rank-Sum test<span></span></a></li>
<li class="chapter" data-level="8.3" data-path="differential-analysis.html"><a href="differential-analysis.html#dominant-taxa"><i class="fa fa-check"></i><b>8.3</b> Dominant taxa<span></span></a></li>
<li class="chapter" data-level="8.4" data-path="differential-analysis.html"><a href="differential-analysis.html#multiple-differential-analysis-by-one-function"><i class="fa fa-check"></i><b>8.4</b> Multiple differential analysis by one function<span></span></a></li>
<li class="chapter" data-level="8.5" data-path="differential-analysis.html"><a href="differential-analysis.html#systematic-information-7"><i class="fa fa-check"></i><b>8.5</b> Systematic Information<span></span></a></li>
</ul></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">MGS microbiota data analysis based on XMAS package</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">MGS microbiota data analysis based on XMAS package</h1>
<p class="author"><em>Hua Zou</em></p>
<p class="date"><em>2022-05-18</em></p>
</div>
<div id="prerequisites" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">Chapter 1</span> Prerequisites<a href="index.html#prerequisites" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Here, we give users one example to practice the MGS microbiota data analysis workflow by <strong>XMAS 2.0</strong>. By the way, we also recommend users handling your own microbiota data in a reasonable manner when you utilize this package or workflow. Pay attention to whether your data fit the methods this package provided.</p>
<div id="preparing-for-the-tutorial" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> Preparing for the tutorial<a href="index.html#preparing-for-the-tutorial" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This tutorial requires that the users should have basic knowledge on R language and traditional statistical background. If you are new to microbial research, we recommend the <strong>Statistical Analysis of Microbiome Data with R</strong> <span class="citation">(<a href="#ref-xia2018statistical" role="doc-biblioref">Xia et al. 2018</a>)</span> as guide book.</p>
<p>We recommend users using local laptop because our own server have not deployed <strong>XMAS 2.0</strong> package yet. In addition, installing the following software before using <strong>XMAS 2.0</strong>.</p>
<ul>
<li><p>R 3.6.3 or later release <a href="https://mirrors.tuna.tsinghua.edu.cn/CRAN/">Download link</a>.</p></li>
<li><p>Rstudio Desktop <a href="https://www.rstudio.com/products/rstudio/download/#download">Download link</a>.</p></li>
</ul>
</div>
<div id="workflow-description" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> Workflow description<a href="index.html#workflow-description" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The standard data analysis of 16S and MGS data by <strong>XMAS 2.0</strong>. The upstream process is performed by in-house pipeline. and this example just shows how to perform downstream data analysis. In briefly, the example comprises the following steps:</p>
<ol style="list-style-type: decimal">
<li><p>Converting inputs into phyloseq object;</p></li>
<li><p>Quality Evaluation;</p></li>
<li><p>Pre-Processing Data;</p></li>
<li><p>Diversity analysis;</p></li>
<li><p>Ordination analysis;</p></li>
<li><p>Composition analysis;</p></li>
<li><p>Differential analysis.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-1"></span>
</li>
</ol>
<div id="htmlwidget-2ad5e698f9e4febef105" style="width:672px;height:768px;" class="grViz html-widget"></div>
<script type="application/json" data-for="htmlwidget-2ad5e698f9e4febef105">{"x":{"diagram":"digraph flowchart { \n\n# initiate graph\ngraph [layout = dot, rankdir = TB, label = \"[dada2_res|metadata|tree]\n\n\", labelloc = t]\n\n\n# global node settings\nnode [shape = rectangle, style = filled, fillcolor = Linen, fontname = arial, width = 4]\n \ntab1 [label = \"Importing data: import_dada2()/summarize_phyloseq()\"]  \ntab2 [label = \"Quality Control(RefCheck/Remove): summarize_taxa()/run_RefCheck()/get_GroupPhyloseq()\"]  \ntab3 [label = \"Quality Control(Rarefaction curves): plot_RarefCure()\"]\ntab4 [label = \"Quality Control(Reads Track): plot_Dada2Track()\"]  \ntab5 [label = \"Quality Control(Variability): plot_taxa_cv()/plot_read_distribution()\"]  \ntab6 [label = \"Data processing(Rarefy): normalize()\"]\ntab7 [label = \"Data processing(Extract specific taxonomic level): summarize_taxa()\"]  \ntab8 [label = \"Data processing(Count2Relative): summarize_taxa(absolute = FALSE)\"]  \ntab9 [label = \"Data processing(Filter&trim): run_filter()/run_trim()\"]\ntab10 [label = \"GlobalView(Alpha diveristy): run_alpha_diversity()/plot_boxplot()\"]  \ntab11 [label = \"GlobalView(Beta diveristy): run_beta_diversity()/run_permanova()/run_ordination()/plot_Ordination()\"]  \ntab12 [label = \"GlobalView(microbial composition): plot_StackBarPlot()/plot_core_taxa()/core_members()\"]\ntab13 [label = \"DifferentialAnalysis: run_lefse()/plot_lefse()/run_wilcox()/plot_volcano()\"] \n \ntab1 -> tab2 -> tab3 -> tab4 -> tab5 -> tab6 -> tab7 -> tab8 -> tab9 -> tab10 -> tab11 -> tab12 -> tab13;  \n}  ","config":{"engine":"dot","options":null}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 1.1: Functions of XMAS 2.0 in 16s
</p>
</div>
</div>
<div id="systematic-information" class="section level2 hasAnchor" number="1.3">
<h2><span class="header-section-number">1.3</span> Systematic Information<a href="index.html#systematic-information" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="index.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.1.2 (2021-11-01)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Monterey 12.2.1
## 
## Matrix products: default
## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] ggpubr_0.4.0    ggplot2_3.3.5   phyloseq_1.38.0 tibble_3.1.6    dplyr_1.0.8     XMAS_2.1.0     
## 
## loaded via a namespace (and not attached):
##   [1] utf8_1.2.2                  tidyselect_1.1.2            RSQLite_2.2.10              AnnotationDbi_1.56.2       
##   [5] htmlwidgets_1.5.4           grid_4.1.2                  BiocParallel_1.28.3         munsell_0.5.0              
##   [9] codetools_0.2-18            withr_2.5.0                 colorspace_2.0-3            Biobase_2.54.0             
##  [13] highr_0.9                   knitr_1.37                  rstudioapi_0.13             pscl_1.5.5                 
##  [17] stats4_4.1.2                robustbase_0.93-9           ggsignif_0.6.3              MatrixGenerics_1.6.0       
##  [21] labeling_0.4.2              optparse_1.7.1              mbzinb_0.2                  GenomeInfoDbData_1.2.7     
##  [25] lpsymphony_1.22.0           pheatmap_1.0.12             bit64_4.0.5                 farver_2.1.0               
##  [29] rhdf5_2.38.1                vctrs_0.3.8                 generics_0.1.2              TH.data_1.1-0              
##  [33] xfun_0.30                   Maaslin2_1.8.0              R6_2.5.1                    GenomeInfoDb_1.30.1        
##  [37] locfit_1.5-9.5              RcppZiggurat_0.1.6          bitops_1.0-7                rhdf5filters_1.6.0         
##  [41] cachem_1.0.6                DelayedArray_0.20.0         assertthat_0.2.1            scales_1.1.1               
##  [45] multcomp_1.4-18             nnet_7.3-17                 gtable_0.3.0                sandwich_3.0-1             
##  [49] rlang_1.0.2                 genefilter_1.76.0           splines_4.1.2               protoclust_1.6.3           
##  [53] rstatix_0.7.0               broom_0.7.12                checkmate_2.0.0             yaml_2.3.5                 
##  [57] reshape2_1.4.4              abind_1.4-5                 backports_1.4.1             Rfast_2.0.6                
##  [61] qvalue_2.26.0               Hmisc_4.6-0                 DiagrammeR_1.0.9            tools_4.1.2                
##  [65] bookdown_0.24               zCompositions_1.4.0         ellipsis_0.3.2              gplots_3.1.1               
##  [69] jquerylib_0.1.4             biomformat_1.22.0           RColorBrewer_1.1-2          BiocGenerics_0.40.0        
##  [73] Rcpp_1.0.8.2                plyr_1.8.6                  base64enc_0.1-3             visNetwork_2.1.0           
##  [77] zlibbioc_1.40.0             purrr_0.3.4                 RCurl_1.98-1.6              rpart_4.1.16               
##  [81] Wrench_1.12.0               cowplot_1.1.1               S4Vectors_0.32.3            zoo_1.8-9                  
##  [85] SummarizedExperiment_1.24.0 ggrepel_0.9.1               cluster_2.1.2               magrittr_2.0.2             
##  [89] data.table_1.14.2           truncnorm_1.0-8             mvtnorm_1.1-3               matrixStats_0.61.0         
##  [93] ALDEx2_1.26.0               corncob_0.2.0               RAIDA_1.0                   evaluate_0.15              
##  [97] xtable_1.8-4                XML_3.99-0.9                jpeg_0.1-9                  IRanges_2.28.0             
## [101] gridExtra_2.3               shape_1.4.6                 compiler_4.1.2              KernSmooth_2.23-20         
## [105] crayon_1.5.0                htmltools_0.5.2             mgcv_1.8-39                 pcaPP_1.9-74               
## [109] Formula_1.2-4               tidyr_1.2.0                 geneplotter_1.72.0          libcoin_1.0-9              
## [113] DBI_1.1.2                   biglm_0.9-2.1               MASS_7.3-55                 Matrix_1.4-0               
## [117] ade4_1.7-18                 getopt_1.20.3               car_3.0-12                  permute_0.9-7              
## [121] cli_3.2.0                   parallel_4.1.2              igraph_1.2.11               GenomicRanges_1.46.1       
## [125] pkgconfig_2.0.3             coin_1.4-2                  foreign_0.8-82              foreach_1.5.2              
## [129] annotate_1.72.0             bslib_0.3.1                 multtest_2.50.0             XVector_0.34.0             
## [133] NADA_1.6-1.1                stringr_1.4.0               digest_0.6.29               vegan_2.5-7                
## [137] Biostrings_2.62.0           rmarkdown_2.13              htmlTable_2.4.0             edgeR_3.36.0               
## [141] gtools_3.9.2                modeltools_0.2-23           lifecycle_1.0.1             nlme_3.1-155               
## [145] jsonlite_1.8.0              Rhdf5lib_1.16.0             carData_3.0-5               limma_3.50.1               
## [149] fansi_1.0.2                 pillar_1.7.0                lattice_0.20-45             KEGGREST_1.34.0            
## [153] fastmap_1.1.0               httr_1.4.2                  DEoptimR_1.0-10             survival_3.3-1             
## [157] glue_1.6.2                  png_0.1-7                   iterators_1.0.14            glmnet_4.1-3               
## [161] bit_4.0.4                   stringi_1.7.6               sass_0.4.0                  metagenomeSeq_1.36.0       
## [165] blob_1.2.2                  DESeq2_1.34.0               latticeExtra_0.6-29         caTools_1.18.2             
## [169] memoise_2.0.1               ape_5.6-2</code></pre>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-xia2018statistical" class="csl-entry">
Xia, Yinglin, Jun Sun, Ding-Geng Chen, et al. 2018. <em>Statistical Analysis of Microbiome Data with r</em>. Vol. 847. Springer.
</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="convert-inputdata-into-phyloseq-object.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/MGS_DataAnalysis_template/blob/main/index.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["MGS microbiota data analysis based on XMAS package.pdf", "MGS microbiota data analysis based on XMAS package.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
