<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Network Analysis | MGS microbiota data analysis based on XMAS package</title>
  <meta name="description" content="Chapter 9 Network Analysis | MGS microbiota data analysis based on XMAS package" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Network Analysis | MGS microbiota data analysis based on XMAS package" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Chapter 9 Network Analysis | MGS microbiota data analysis based on XMAS package" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Network Analysis | MGS microbiota data analysis based on XMAS package" />
  
  <meta name="twitter:description" content="Chapter 9 Network Analysis | MGS microbiota data analysis based on XMAS package" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2023-10-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="differential-analysis.html"/>
<link rel="next" href="test-example.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.28/datatables.js"></script>
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">MGS microbiota data analysis based on XMAS package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-tutorial"><i class="fa fa-check"></i><b>1.1</b> Preparing for the tutorial</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#workflow-description"><i class="fa fa-check"></i><b>1.2</b> Workflow description</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html"><i class="fa fa-check"></i><b>2</b> Convert inputdata into phyloseq object</a>
<ul>
<li class="chapter" data-level="2.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metaphlan23"><i class="fa fa-check"></i><b>2.1</b> Metaphlan2/3</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#merging-output-of-metaphlan"><i class="fa fa-check"></i><b>2.1.1</b> Merging output of metaphlan</a></li>
<li class="chapter" data-level="2.1.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results"><i class="fa fa-check"></i><b>2.1.2</b> Importing results</a></li>
<li class="chapter" data-level="2.1.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table"><i class="fa fa-check"></i><b>2.1.3</b> taxa table</a></li>
<li class="chapter" data-level="2.1.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table"><i class="fa fa-check"></i><b>2.1.4</b> otu table</a></li>
<li class="chapter" data-level="2.1.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table"><i class="fa fa-check"></i><b>2.1.5</b> metadata table</a></li>
<li class="chapter" data-level="2.1.6" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object"><i class="fa fa-check"></i><b>2.1.6</b> phyloseq object</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#transform-the-abundance-of-taxa-whose-relative-abundance-under-limit-of-detection-lod-into-zeros"><i class="fa fa-check"></i><b>2.2</b> Transform the abundance of taxa whose relative abundance under Limit Of Detection (LOD) into Zeros</a></li>
<li class="chapter" data-level="2.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#systematic-information"><i class="fa fa-check"></i><b>2.3</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>3</b> Quality Control</a>
<ul>
<li class="chapter" data-level="3.1" data-path="quality-control.html"><a href="quality-control.html#spike-in-sample-brs"><i class="fa fa-check"></i><b>3.1</b> spike-in sample (BRS)</a></li>
<li class="chapter" data-level="3.2" data-path="quality-control.html"><a href="quality-control.html#spike-in-samples-brs-remove"><i class="fa fa-check"></i><b>3.2</b> Spike-in sample’s (BRS) remove</a></li>
<li class="chapter" data-level="3.3" data-path="quality-control.html"><a href="quality-control.html#systematic-information-1"><i class="fa fa-check"></i><b>3.3</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data-processing.html"><a href="data-processing.html"><i class="fa fa-check"></i><b>4</b> Data processing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-processing.html"><a href="data-processing.html#extracting-specific-taxonomic-level"><i class="fa fa-check"></i><b>4.1</b> Extracting specific taxonomic level</a></li>
<li class="chapter" data-level="4.2" data-path="data-processing.html"><a href="data-processing.html#systematic-information-2"><i class="fa fa-check"></i><b>4.2</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="diversity-analysis.html"><a href="diversity-analysis.html"><i class="fa fa-check"></i><b>5</b> Diversity analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="diversity-analysis.html"><a href="diversity-analysis.html#alpha-diveristy"><i class="fa fa-check"></i><b>5.1</b> Alpha diveristy</a></li>
<li class="chapter" data-level="5.2" data-path="diversity-analysis.html"><a href="diversity-analysis.html#beta-diversity"><i class="fa fa-check"></i><b>5.2</b> Beta diversity</a></li>
<li class="chapter" data-level="5.3" data-path="diversity-analysis.html"><a href="diversity-analysis.html#systematic-information-3"><i class="fa fa-check"></i><b>5.3</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html"><i class="fa fa-check"></i><b>6</b> PERMANOVA + beta dispersion + Ordination</a>
<ul>
<li class="chapter" data-level="6.1" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#permanova"><i class="fa fa-check"></i><b>6.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="6.2" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#beta-dispersion"><i class="fa fa-check"></i><b>6.2</b> beta dispersion</a></li>
<li class="chapter" data-level="6.3" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#ordination"><i class="fa fa-check"></i><b>6.3</b> Ordination</a></li>
<li class="chapter" data-level="6.4" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#systematic-information-4"><i class="fa fa-check"></i><b>6.4</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="microbial-composition.html"><a href="microbial-composition.html"><i class="fa fa-check"></i><b>7</b> Microbial composition</a>
<ul>
<li class="chapter" data-level="7.1" data-path="microbial-composition.html"><a href="microbial-composition.html#stacked-barplot"><i class="fa fa-check"></i><b>7.1</b> Stacked barplot</a></li>
<li class="chapter" data-level="7.2" data-path="microbial-composition.html"><a href="microbial-composition.html#core-microbiota"><i class="fa fa-check"></i><b>7.2</b> Core microbiota</a></li>
<li class="chapter" data-level="7.3" data-path="microbial-composition.html"><a href="microbial-composition.html#systematic-information-5"><i class="fa fa-check"></i><b>7.3</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-analysis.html"><a href="differential-analysis.html"><i class="fa fa-check"></i><b>8</b> Differential Analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="differential-analysis.html"><a href="differential-analysis.html#filtering-trimming"><i class="fa fa-check"></i><b>8.1</b> Filtering &amp; Trimming</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="differential-analysis.html"><a href="differential-analysis.html#filtering-the-low-relative-abundance-or-unclassified-taxa-by-the-threshold-total-counts-1e-4"><i class="fa fa-check"></i><b>8.1.1</b> Filtering the low relative abundance or unclassified taxa by the threshold (total counts &lt; 1e-4)</a></li>
<li class="chapter" data-level="8.1.2" data-path="differential-analysis.html"><a href="differential-analysis.html#trimming-the-taxa-with-low-occurrence-less-than-threshold"><i class="fa fa-check"></i><b>8.1.2</b> Trimming the taxa with low occurrence less than threshold</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse"><i class="fa fa-check"></i><b>8.2</b> Liner discriminant analysis (LDA) effect size (LEfSe)</a></li>
<li class="chapter" data-level="8.3" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-test"><i class="fa fa-check"></i><b>8.3</b> Wilcoxon Rank-Sum test</a></li>
<li class="chapter" data-level="8.4" data-path="differential-analysis.html"><a href="differential-analysis.html#dominant-taxa"><i class="fa fa-check"></i><b>8.4</b> Dominant taxa</a></li>
<li class="chapter" data-level="8.5" data-path="differential-analysis.html"><a href="differential-analysis.html#multiple-differential-analysis-by-one-function"><i class="fa fa-check"></i><b>8.5</b> Multiple differential analysis by one function</a></li>
<li class="chapter" data-level="8.6" data-path="differential-analysis.html"><a href="differential-analysis.html#systematic-information-6"><i class="fa fa-check"></i><b>8.6</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="network-analysis.html"><a href="network-analysis.html"><i class="fa fa-check"></i><b>9</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="network-analysis.html"><a href="network-analysis.html#importing-data"><i class="fa fa-check"></i><b>9.1</b> Importing data</a></li>
<li class="chapter" data-level="9.2" data-path="network-analysis.html"><a href="network-analysis.html#single-network-with-pearson-correlation-as-association-measure"><i class="fa fa-check"></i><b>9.2</b> Single network with Pearson correlation as association measure</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="network-analysis.html"><a href="network-analysis.html#building-network-model"><i class="fa fa-check"></i><b>9.2.1</b> Building network model</a></li>
<li class="chapter" data-level="9.2.2" data-path="network-analysis.html"><a href="network-analysis.html#analyzing-the-constructed-network"><i class="fa fa-check"></i><b>9.2.2</b> Analyzing the constructed network</a></li>
<li class="chapter" data-level="9.2.3" data-path="network-analysis.html"><a href="network-analysis.html#visualizing-the-network"><i class="fa fa-check"></i><b>9.2.3</b> Visualizing the network</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="network-analysis.html"><a href="network-analysis.html#dissimilarity-based-networks"><i class="fa fa-check"></i><b>9.3</b> Dissimilarity-based Networks</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="network-analysis.html"><a href="network-analysis.html#building-network-module"><i class="fa fa-check"></i><b>9.3.1</b> Building network module</a></li>
<li class="chapter" data-level="9.3.2" data-path="network-analysis.html"><a href="network-analysis.html#analyzing-the-constructed-network-1"><i class="fa fa-check"></i><b>9.3.2</b> Analyzing the constructed network</a></li>
<li class="chapter" data-level="9.3.3" data-path="network-analysis.html"><a href="network-analysis.html#visualizing-the-network-1"><i class="fa fa-check"></i><b>9.3.3</b> Visualizing the network</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="network-analysis.html"><a href="network-analysis.html#systematic-information-7"><i class="fa fa-check"></i><b>9.4</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="test-example.html"><a href="test-example.html"><i class="fa fa-check"></i><b>10</b> Test Example</a>
<ul>
<li class="chapter" data-level="10.1" data-path="test-example.html"><a href="test-example.html#loading-packages"><i class="fa fa-check"></i><b>10.1</b> Loading packages</a></li>
<li class="chapter" data-level="10.2" data-path="test-example.html"><a href="test-example.html#loading-data"><i class="fa fa-check"></i><b>10.2</b> Loading data</a></li>
<li class="chapter" data-level="10.3" data-path="test-example.html"><a href="test-example.html#step1-convert-inputs-into-phyloseq-data"><i class="fa fa-check"></i><b>10.3</b> Step1: Convert inputs into phyloseq data</a></li>
<li class="chapter" data-level="10.4" data-path="test-example.html"><a href="test-example.html#step2-transform-limit-of-detection-lod-into-zeros"><i class="fa fa-check"></i><b>10.4</b> Step2: Transform limit of detection (LOD) into Zeros</a></li>
<li class="chapter" data-level="10.5" data-path="test-example.html"><a href="test-example.html#step3-brs-checking"><i class="fa fa-check"></i><b>10.5</b> Step3: BRS checking</a></li>
<li class="chapter" data-level="10.6" data-path="test-example.html"><a href="test-example.html#step4-extracting-specific-taxonomic-level"><i class="fa fa-check"></i><b>10.6</b> Step4: Extracting specific taxonomic level</a></li>
<li class="chapter" data-level="10.7" data-path="test-example.html"><a href="test-example.html#step5-globalview"><i class="fa fa-check"></i><b>10.7</b> Step5: GlobalView</a></li>
<li class="chapter" data-level="10.8" data-path="test-example.html"><a href="test-example.html#step6-differential-analysis"><i class="fa fa-check"></i><b>10.8</b> Step6: Differential Analysis</a></li>
<li class="chapter" data-level="10.9" data-path="test-example.html"><a href="test-example.html#systematic-information-8"><i class="fa fa-check"></i><b>10.9</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">MGS microbiota data analysis based on XMAS package</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="network-analysis" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">Chapter 9</span> Network Analysis<a href="network-analysis.html#network-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<blockquote>
<p>Estimating microbial association networks from high-throughput sequencing data is a common exploratory data analysis approach aiming at understanding the complex interplay of microbial communities in their natural habitat. Statistical network estimation workflows comprise several analysis steps, including methods for zero handling, data normalization and computing microbial associations. Since microbial interactions are likely to change between conditions, e.g. between healthy individuals and patients, identifying network differences between groups is often an integral secondary analysis step.</p>
</blockquote>
<p>NetCoMi (Network Construction and Comparison for Microbiome Data) <span class="citation">(<a href="#ref-peschel2021netcomi">Peschel et al. 2021</a>)</span> provides functionality for constructing, analyzing, and comparing networks suitable for the application on microbial compositional data.</p>
<p>The following information is from <a href="https://github.com/stefpeschel/NetCoMi">NetCoMi github</a>.</p>
<p><strong>Association measures:</strong></p>
<ul>
<li>Pearson coefficient
(<a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/cor"><code>cor()</code></a>
from <code>stats</code> package)</li>
<li>Spearman coefficient
(<a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/cor"><code>cor()</code></a>
from <code>stats</code> package)</li>
<li>Biweight Midcorrelation
<a href="https://rdrr.io/cran/WGCNA/man/bicor.html"><code>bicor()</code></a> from <code>WGCNA</code>
package</li>
<li>SparCC
(<a href="https://rdrr.io/github/zdk123/SpiecEasi/man/sparcc.html"><code>sparcc()</code></a>
from <code>SpiecEasi</code> package)</li>
<li>CCLasso (<a href="https://github.com/huayingfang/CCLasso">R code on GitHub</a>)</li>
<li>CCREPE
(<a href="https://bioconductor.org/packages/release/bioc/html/ccrepe.html"><code>ccrepe</code></a>
package)</li>
<li>SpiecEasi (<a href="https://github.com/zdk123/SpiecEasi"><code>SpiecEasi</code></a>
package)</li>
<li>SPRING (<a href="https://github.com/GraceYoon/SPRING"><code>SPRING</code></a> package)</li>
<li>gCoda (<a href="https://github.com/huayingfang/gCoda">R code on GitHub</a>)</li>
<li>propr
(<a href="https://cran.r-project.org/web/packages/propr/index.html"><code>propr</code></a>
package)</li>
</ul>
<p><strong>Dissimilarity measures:</strong></p>
<ul>
<li>Euclidean distance
(<a href="https://www.rdocumentation.org/packages/vegan/versions/2.4-2/topics/vegdist"><code>vegdist()</code></a>
from <code>vegan</code> package)</li>
<li>Bray-Curtis dissimilarity
(<a href="https://www.rdocumentation.org/packages/vegan/versions/2.4-2/topics/vegdist"><code>vegdist()</code></a>
from <code>vegan</code> package)</li>
<li>Kullback-Leibler divergence (KLD)
(<a href="https://rdrr.io/cran/LaplacesDemon/man/KLD.html"><code>KLD()</code></a> from
<code>LaplacesDemon</code> package)</li>
<li>Jeffrey divergence (own code using
<a href="https://rdrr.io/cran/LaplacesDemon/man/KLD.html"><code>KLD()</code></a> from
<code>LaplacesDemon</code> package)</li>
<li>Jensen-Shannon divergence (own code using
<a href="https://rdrr.io/cran/LaplacesDemon/man/KLD.html"><code>KLD()</code></a> from
<code>LaplacesDemon</code> package)</li>
<li>Compositional KLD (own implementation following [Martín-Fernández
et al., 1999])</li>
<li>Aitchison distance
(<a href="https://www.rdocumentation.org/packages/vegan/versions/2.4-2/topics/vegdist"><code>vegdist()</code></a>
and <a href="https://rdrr.io/github/zdk123/SpiecEasi/man/clr.html"><code>clr()</code></a>
from <code>SpiecEasi</code> package)</li>
</ul>
<p><strong>Methods for zero replacement:</strong></p>
<ul>
<li>Adding a predefined pseudo count</li>
<li>Multiplicative replacement
(<a href="https://rdrr.io/cran/zCompositions/man/multRepl.html"><code>multRepl</code></a>
from <code>zCompositions</code> package)</li>
<li>Modified EM alr-algorithm
(<a href="https://rdrr.io/cran/zCompositions/man/lrEM.html"><code>lrEM</code></a> from
<code>zCompositions</code> package)</li>
<li>Bayesian-multiplicative replacement
(<a href="https://rdrr.io/cran/zCompositions/man/cmultRepl.html"><code>cmultRepl</code></a>
from <code>zCompositions</code> package)</li>
</ul>
<p><strong>Normalization methods:</strong></p>
<ul>
<li>Total Sum Scaling (TSS) (own implementation)</li>
<li>Cumulative Sum Scaling (CSS) (<a href=""><code>cumNormMat</code></a> from <code>metagenomeSeq</code>
package)</li>
<li>Common Sum Scaling (COM) (own implementation)</li>
<li>Rarefying (<a href=""><code>rrarefy</code></a> from <code>vegan</code> package)</li>
<li>Variance Stabilizing Transformation (VST)
(<a href=""><code>varianceStabilizingTransformation</code></a> from <code>DESeq2</code> package)</li>
<li>Centered log-ratio (clr) transformation
(<a href="https://rdrr.io/github/zdk123/SpiecEasi/man/clr.html"><code>clr()</code></a>
from <code>SpiecEasi</code> package))</li>
</ul>
<p>TSS, CSS, COM, VST, and the clr transformation are described in [Badri
et al., 2020].</p>
<p><strong>Loading packages</strong></p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="network-analysis.html#cb83-1" tabindex="-1"></a><span class="fu">library</span>(XMAS2)</span>
<span id="cb83-2"><a href="network-analysis.html#cb83-2" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb83-3"><a href="network-analysis.html#cb83-3" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb83-4"><a href="network-analysis.html#cb83-4" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb83-5"><a href="network-analysis.html#cb83-5" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb83-6"><a href="network-analysis.html#cb83-6" tabindex="-1"></a><span class="fu">library</span>(NetCoMi)</span></code></pre></div>
<div id="importing-data" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> Importing data<a href="network-analysis.html#importing-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="network-analysis.html#cb84-1" tabindex="-1"></a>metaphlan2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb84-2"><a href="network-analysis.html#cb84-2" tabindex="-1"></a>                         <span class="at">ps =</span> metaphlan2_ps,</span>
<span id="cb84-3"><a href="network-analysis.html#cb84-3" tabindex="-1"></a>                         <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb84-4"><a href="network-analysis.html#cb84-4" tabindex="-1"></a>                         <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>,</span>
<span id="cb84-5"><a href="network-analysis.html#cb84-5" tabindex="-1"></a>                         <span class="at">discard =</span> <span class="cn">TRUE</span>)</span>
<span id="cb84-6"><a href="network-analysis.html#cb84-6" tabindex="-1"></a></span>
<span id="cb84-7"><a href="network-analysis.html#cb84-7" tabindex="-1"></a>metaphlan2_ps_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(<span class="at">ps =</span> metaphlan2_ps_remove_BRS, </span>
<span id="cb84-8"><a href="network-analysis.html#cb84-8" tabindex="-1"></a>                                 <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>)</span>
<span id="cb84-9"><a href="network-analysis.html#cb84-9" tabindex="-1"></a></span>
<span id="cb84-10"><a href="network-analysis.html#cb84-10" tabindex="-1"></a>metaphlan2_ps_genus_filter <span class="ot">&lt;-</span> <span class="fu">run_filter</span>(<span class="at">ps =</span> metaphlan2_ps_genus, </span>
<span id="cb84-11"><a href="network-analysis.html#cb84-11" tabindex="-1"></a>                                         <span class="at">cutoff =</span> <span class="fl">1e-4</span>, </span>
<span id="cb84-12"><a href="network-analysis.html#cb84-12" tabindex="-1"></a>                                         <span class="at">unclass =</span> <span class="cn">TRUE</span>)</span>
<span id="cb84-13"><a href="network-analysis.html#cb84-13" tabindex="-1"></a></span>
<span id="cb84-14"><a href="network-analysis.html#cb84-14" tabindex="-1"></a>metaphlan2_ps_genus_filter</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 51 taxa and 22 samples ]
## sample_data() Sample Data:       [ 22 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 51 taxa by 6 taxonomic ranks ]</code></pre>
</div>
<div id="single-network-with-pearson-correlation-as-association-measure" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> Single network with Pearson correlation as association measure<a href="network-analysis.html#single-network-with-pearson-correlation-as-association-measure" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Since Pearson correlations may lead to compositional effects when applied to sequencing data, we use the clr transformation as normalization method. Zero treatment is necessary in this case.</p>
<p>A threshold of 0.3 is used as sparsification method, so that only OTUs with an absolute correlation greater than or equal to 0.3 are connected.</p>
<div id="building-network-model" class="section level3 hasAnchor" number="9.2.1">
<h3><span class="header-section-number">9.2.1</span> Building network model<a href="network-analysis.html#building-network-model" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="network-analysis.html#cb86-1" tabindex="-1"></a>net_single <span class="ot">&lt;-</span> <span class="fu">netConstruct</span>(metaphlan2_ps_genus_filter,  </span>
<span id="cb86-2"><a href="network-analysis.html#cb86-2" tabindex="-1"></a>                           <span class="at">measure =</span> <span class="st">&quot;pearson&quot;</span>,</span>
<span id="cb86-3"><a href="network-analysis.html#cb86-3" tabindex="-1"></a>                           <span class="at">normMethod =</span> <span class="st">&quot;none&quot;</span>, </span>
<span id="cb86-4"><a href="network-analysis.html#cb86-4" tabindex="-1"></a>                           <span class="at">zeroMethod =</span> <span class="st">&quot;multRepl&quot;</span>,</span>
<span id="cb86-5"><a href="network-analysis.html#cb86-5" tabindex="-1"></a>                           <span class="at">sparsMethod =</span> <span class="st">&quot;threshold&quot;</span>, </span>
<span id="cb86-6"><a href="network-analysis.html#cb86-6" tabindex="-1"></a>                           <span class="at">thresh =</span> <span class="fl">0.7</span>,</span>
<span id="cb86-7"><a href="network-analysis.html#cb86-7" tabindex="-1"></a>                           <span class="at">verbose =</span> <span class="dv">3</span>,</span>
<span id="cb86-8"><a href="network-analysis.html#cb86-8" tabindex="-1"></a>                           <span class="at">seed =</span> <span class="dv">123</span>)</span></code></pre></div>
</div>
<div id="analyzing-the-constructed-network" class="section level3 hasAnchor" number="9.2.2">
<h3><span class="header-section-number">9.2.2</span> Analyzing the constructed network<a href="network-analysis.html#analyzing-the-constructed-network" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>NetCoMi’s <code>netAnalyze()</code> function is used for analyzing the constructed network(s).</p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="network-analysis.html#cb87-1" tabindex="-1"></a>props_single <span class="ot">&lt;-</span> <span class="fu">netAnalyze</span>(net_single, <span class="at">clustMethod =</span> <span class="st">&quot;cluster_fast_greedy&quot;</span>)</span>
<span id="cb87-2"><a href="network-analysis.html#cb87-2" tabindex="-1"></a></span>
<span id="cb87-3"><a href="network-analysis.html#cb87-3" tabindex="-1"></a><span class="fu">summary</span>(props_single, <span class="at">numbNodes =</span> 5L)</span></code></pre></div>
<pre><code>## 
## Component sizes
## ```````````````                
## size: 6 4 3 2  1
##    #: 1 1 2 6 23
## ______________________________
## Global network properties
## `````````````````````````
## Largest connected component (LCC):
##                                   
## Relative LCC size          0.11765
## Clustering coefficient     0.59409
## Modularity                 0.12245
## Positive edge percentage 100.00000
## Edge density               0.46667
## Natural connectivity       0.38192
## Vertex connectivity        1.00000
## Edge connectivity          1.00000
## Average dissimilarity*     0.67983
## Average path length**      0.76641
## 
## Whole network:
##                                   
## Number of components      33.00000
## Clustering coefficient     0.69803
## Modularity                 0.83058
## Positive edge percentage 100.00000
## Edge density               0.01725
## Natural connectivity       0.02661
## 
##  *Dissimilarity = 1 - edge weight
## **Path length: Units with average dissimilarity
## 
## ______________________________
## Clusters
## - In the whole network
## - Algorithm: cluster_fast_greedy
## ```````````````````````````````` 
##                              
## name:  0 1 2 3 4 5 6 7 8 9 10
##    #: 23 6 4 2 2 2 3 2 2 2  3
## 
## ______________________________
## Hubs
## - In alphabetical/numerical order
## - Based on empirical quantiles of centralities
## ```````````````````````````````````````````````              
##  g__Alistipes 
##  g__Holdemania
##  g__Proteus   
## 
## ______________________________
## Centrality measures
## - In decreasing order
## - Centrality of disconnected components is zero
## ````````````````````````````````````````````````
## Degree (normalized):
##                       
## g__Proteus        0.08
## g__Alistipes      0.06
## g__Holdemania     0.06
## g__Flavonifractor 0.04
## g__Enterococcus   0.02
## 
## Betweenness centrality (normalized):
##                       
## g__Proteus         0.6
## g__Alistipes       0.4
## g__Acidaminococcus 0.0
## g__Adlercreutzia   0.0
## g__Akkermansia     0.0
## 
## Closeness centrality (normalized):
##                          
## g__Proteus        2.21621
## g__Holdemania     1.91300
## g__Alistipes      1.71565
## g__Flavonifractor 1.35153
## g__Enterococcus   1.19617
## 
## Eigenvector centrality (normalized):
##                          
## g__Proteus        1.00000
## g__Holdemania     0.88889
## g__Alistipes      0.76367
## g__Flavonifractor 0.64535
## g__Enterococcus   0.33917</code></pre>
</div>
<div id="visualizing-the-network" class="section level3 hasAnchor" number="9.2.3">
<h3><span class="header-section-number">9.2.3</span> Visualizing the network<a href="network-analysis.html#visualizing-the-network" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>primary plot</li>
</ul>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="network-analysis.html#cb89-1" tabindex="-1"></a><span class="fu">plot</span>(props_single, </span>
<span id="cb89-2"><a href="network-analysis.html#cb89-2" tabindex="-1"></a>     <span class="at">nodeColor =</span> <span class="st">&quot;cluster&quot;</span>, </span>
<span id="cb89-3"><a href="network-analysis.html#cb89-3" tabindex="-1"></a>     <span class="at">nodeSize =</span> <span class="st">&quot;eigenvector&quot;</span>,</span>
<span id="cb89-4"><a href="network-analysis.html#cb89-4" tabindex="-1"></a>     <span class="at">title1 =</span> <span class="st">&quot;Network on 16s with Pearson correlations&quot;</span>, </span>
<span id="cb89-5"><a href="network-analysis.html#cb89-5" tabindex="-1"></a>     <span class="at">showTitle =</span> <span class="cn">TRUE</span>,</span>
<span id="cb89-6"><a href="network-analysis.html#cb89-6" tabindex="-1"></a>     <span class="at">labelLength =</span> <span class="dv">10</span>,</span>
<span id="cb89-7"><a href="network-analysis.html#cb89-7" tabindex="-1"></a>     <span class="at">cexTitle =</span> <span class="fl">1.5</span>)</span>
<span id="cb89-8"><a href="network-analysis.html#cb89-8" tabindex="-1"></a></span>
<span id="cb89-9"><a href="network-analysis.html#cb89-9" tabindex="-1"></a><span class="fu">legend</span>(<span class="fl">0.7</span>, <span class="fl">1.1</span>, <span class="at">cex =</span> <span class="dv">1</span>, <span class="at">title =</span> <span class="st">&quot;estimated correlation:&quot;</span>, </span>
<span id="cb89-10"><a href="network-analysis.html#cb89-10" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span>,<span class="st">&quot;-&quot;</span>), <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;#009900&quot;</span>,<span class="st">&quot;red&quot;</span>), </span>
<span id="cb89-11"><a href="network-analysis.html#cb89-11" tabindex="-1"></a>       <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">horiz =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-53"></span>
<img src="MGS-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-53-1.png" alt="Network on 16s with Pearson correlations" width="768" />
<p class="caption">
Figure 9.1: Network on 16s with Pearson correlations
</p>
</div>
<ul>
<li><p>improve the visualization by changing the following arguments:</p>
<ul>
<li><p><code>repulsion = 0.8</code>: Place the nodes further apart</p></li>
<li><p><code>rmSingles = TRUE</code>: Single nodes are removed</p></li>
<li><p><code>labelScale = FALSE</code> and <code>cexLabels = 1.6</code>: All labels have equal size and are enlarged to improve readability of small node’s labels</p></li>
<li><p><code>nodeSizeSpread = 3</code> (default is 4): Node sizes are more similar if the value is decreased. This argument (in combination with <code>cexNodes</code>) is useful to enlarge small nodes while keeping the size of big nodes.</p></li>
</ul></li>
</ul>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="network-analysis.html#cb90-1" tabindex="-1"></a><span class="fu">plot</span>(props_single, </span>
<span id="cb90-2"><a href="network-analysis.html#cb90-2" tabindex="-1"></a>     <span class="at">nodeColor =</span> <span class="st">&quot;cluster&quot;</span>, </span>
<span id="cb90-3"><a href="network-analysis.html#cb90-3" tabindex="-1"></a>     <span class="at">nodeSize =</span> <span class="st">&quot;eigenvector&quot;</span>,</span>
<span id="cb90-4"><a href="network-analysis.html#cb90-4" tabindex="-1"></a>     <span class="at">labelLength =</span> <span class="dv">40</span>,</span>
<span id="cb90-5"><a href="network-analysis.html#cb90-5" tabindex="-1"></a>     <span class="at">repulsion =</span> <span class="fl">0.8</span>,</span>
<span id="cb90-6"><a href="network-analysis.html#cb90-6" tabindex="-1"></a>     <span class="at">rmSingles =</span> <span class="cn">TRUE</span>,</span>
<span id="cb90-7"><a href="network-analysis.html#cb90-7" tabindex="-1"></a>     <span class="at">labelScale =</span> <span class="cn">FALSE</span>,</span>
<span id="cb90-8"><a href="network-analysis.html#cb90-8" tabindex="-1"></a>     <span class="at">cexLabels =</span> <span class="dv">1</span>,</span>
<span id="cb90-9"><a href="network-analysis.html#cb90-9" tabindex="-1"></a>     <span class="at">nodeSizeSpread =</span> <span class="dv">3</span>,</span>
<span id="cb90-10"><a href="network-analysis.html#cb90-10" tabindex="-1"></a>     <span class="at">cexNodes =</span> <span class="dv">2</span>,</span>
<span id="cb90-11"><a href="network-analysis.html#cb90-11" tabindex="-1"></a>     <span class="at">title1 =</span> <span class="st">&quot;Network on metabolomics with Pearson correlations&quot;</span>, </span>
<span id="cb90-12"><a href="network-analysis.html#cb90-12" tabindex="-1"></a>     <span class="at">showTitle =</span> <span class="cn">TRUE</span>,</span>
<span id="cb90-13"><a href="network-analysis.html#cb90-13" tabindex="-1"></a>     <span class="at">cexTitle =</span> <span class="fl">1.5</span>)</span>
<span id="cb90-14"><a href="network-analysis.html#cb90-14" tabindex="-1"></a></span>
<span id="cb90-15"><a href="network-analysis.html#cb90-15" tabindex="-1"></a><span class="fu">legend</span>(<span class="fl">0.7</span>, <span class="fl">1.1</span>, <span class="at">cex =</span> <span class="fl">1.2</span>, <span class="at">title =</span> <span class="st">&quot;estimated correlation:&quot;</span>,</span>
<span id="cb90-16"><a href="network-analysis.html#cb90-16" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span>,<span class="st">&quot;-&quot;</span>), <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;#009900&quot;</span>,<span class="st">&quot;red&quot;</span>),</span>
<span id="cb90-17"><a href="network-analysis.html#cb90-17" tabindex="-1"></a>       <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">horiz =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-54"></span>
<img src="MGS-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-54-1.png" alt="Network on 16s with Pearson correlations" width="1152" />
<p class="caption">
Figure 9.2: Network on 16s with Pearson correlations
</p>
</div>
<ul>
<li>Network on genus level with Pearson correlations</li>
</ul>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="network-analysis.html#cb91-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb91-2"><a href="network-analysis.html#cb91-2" tabindex="-1"></a></span>
<span id="cb91-3"><a href="network-analysis.html#cb91-3" tabindex="-1"></a>graph <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">graph_from_adjacency_matrix</span>(net_single<span class="sc">$</span>adjaMat1, <span class="at">weighted =</span> <span class="cn">TRUE</span>)</span>
<span id="cb91-4"><a href="network-analysis.html#cb91-4" tabindex="-1"></a>lay_fr <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">layout_with_fr</span>(graph)</span>
<span id="cb91-5"><a href="network-analysis.html#cb91-5" tabindex="-1"></a><span class="fu">rownames</span>(lay_fr) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(net_single<span class="sc">$</span>adjaMat1)</span>
<span id="cb91-6"><a href="network-analysis.html#cb91-6" tabindex="-1"></a></span>
<span id="cb91-7"><a href="network-analysis.html#cb91-7" tabindex="-1"></a><span class="fu">plot</span>(props_single,</span>
<span id="cb91-8"><a href="network-analysis.html#cb91-8" tabindex="-1"></a>     <span class="at">layout =</span> <span class="st">&quot;layout_with_fr&quot;</span>,</span>
<span id="cb91-9"><a href="network-analysis.html#cb91-9" tabindex="-1"></a>     <span class="at">shortenLabels =</span> <span class="st">&quot;simple&quot;</span>,</span>
<span id="cb91-10"><a href="network-analysis.html#cb91-10" tabindex="-1"></a>     <span class="at">labelLength =</span> <span class="dv">30</span>,</span>
<span id="cb91-11"><a href="network-analysis.html#cb91-11" tabindex="-1"></a>     <span class="at">charToRm =</span> <span class="st">&quot;g__&quot;</span>,</span>
<span id="cb91-12"><a href="network-analysis.html#cb91-12" tabindex="-1"></a>     <span class="at">labelScale =</span> <span class="cn">FALSE</span>,</span>
<span id="cb91-13"><a href="network-analysis.html#cb91-13" tabindex="-1"></a>     <span class="at">rmSingles =</span> <span class="cn">TRUE</span>,</span>
<span id="cb91-14"><a href="network-analysis.html#cb91-14" tabindex="-1"></a>     <span class="at">nodeSize =</span> <span class="st">&quot;clr&quot;</span>,</span>
<span id="cb91-15"><a href="network-analysis.html#cb91-15" tabindex="-1"></a>     <span class="at">nodeColor =</span> <span class="st">&quot;cluster&quot;</span>,</span>
<span id="cb91-16"><a href="network-analysis.html#cb91-16" tabindex="-1"></a>     <span class="at">hubBorderCol =</span> <span class="st">&quot;darkgray&quot;</span>,</span>
<span id="cb91-17"><a href="network-analysis.html#cb91-17" tabindex="-1"></a>     <span class="at">cexNodes =</span> <span class="dv">2</span>,</span>
<span id="cb91-18"><a href="network-analysis.html#cb91-18" tabindex="-1"></a>     <span class="at">cexLabels =</span> <span class="fl">1.5</span>,</span>
<span id="cb91-19"><a href="network-analysis.html#cb91-19" tabindex="-1"></a>     <span class="at">cexHubLabels =</span> <span class="dv">2</span>,</span>
<span id="cb91-20"><a href="network-analysis.html#cb91-20" tabindex="-1"></a>     <span class="at">title1 =</span> <span class="st">&quot;Network on genus level with Pearson correlations&quot;</span>, </span>
<span id="cb91-21"><a href="network-analysis.html#cb91-21" tabindex="-1"></a>     <span class="at">showTitle =</span> <span class="cn">TRUE</span>,</span>
<span id="cb91-22"><a href="network-analysis.html#cb91-22" tabindex="-1"></a>     <span class="at">cexTitle =</span> <span class="fl">2.3</span>)</span>
<span id="cb91-23"><a href="network-analysis.html#cb91-23" tabindex="-1"></a></span>
<span id="cb91-24"><a href="network-analysis.html#cb91-24" tabindex="-1"></a><span class="fu">legend</span>(<span class="fl">0.7</span>, <span class="fl">1.1</span>, <span class="at">cex =</span> <span class="fl">2.2</span>, <span class="at">title =</span> <span class="st">&quot;estimated correlation:&quot;</span>,</span>
<span id="cb91-25"><a href="network-analysis.html#cb91-25" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span>,<span class="st">&quot;-&quot;</span>), <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;#009900&quot;</span>,<span class="st">&quot;red&quot;</span>), </span>
<span id="cb91-26"><a href="network-analysis.html#cb91-26" tabindex="-1"></a>       <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">horiz =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-55"></span>
<img src="MGS-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-55-1.png" alt="Network on 16s with Pearson correlations (Genus)" width="1152" />
<p class="caption">
Figure 9.3: Network on 16s with Pearson correlations (Genus)
</p>
</div>
<p>Let’s check whether the largest nodes are actually those with highest column sums in the matrix with normalized counts returned by netConstruct().</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="network-analysis.html#cb92-1" tabindex="-1"></a><span class="fu">sort</span>(<span class="fu">colSums</span>(net_single<span class="sc">$</span>normCounts1), <span class="at">decreasing =</span> <span class="cn">TRUE</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span></code></pre></div>
<pre><code>##      g__Bacteroides  g__Bifidobacterium       g__Prevotella      g__Eubacterium g__Faecalibacterium    g__Streptococcus     g__Ruminococcus 
##           6.2673586           2.3606139           2.0131275           1.2915922           1.2683941           0.8139481           0.7899226 
##        g__Megamonas          g__Blautia    g__Lactobacillus 
##           0.7171101           0.5348218           0.4701121</code></pre>
<ul>
<li>Network on genus level with Pearson correlations with phylum annotation</li>
</ul>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="network-analysis.html#cb94-1" tabindex="-1"></a><span class="cf">if</span> (<span class="cn">FALSE</span>) {</span>
<span id="cb94-2"><a href="network-analysis.html#cb94-2" tabindex="-1"></a><span class="co"># Get phyla names from the taxonomic table created before</span></span>
<span id="cb94-3"><a href="network-analysis.html#cb94-3" tabindex="-1"></a>taxatab <span class="ot">&lt;-</span> metaphlan2_ps_genus_filter<span class="sc">@</span>tax_table <span class="sc">%&gt;%</span></span>
<span id="cb94-4"><a href="network-analysis.html#cb94-4" tabindex="-1"></a>  <span class="fu">data.frame</span>()</span>
<span id="cb94-5"><a href="network-analysis.html#cb94-5" tabindex="-1"></a>phyla <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">gsub</span>(<span class="st">&quot;p__&quot;</span>, <span class="st">&quot;&quot;</span>, taxatab[, <span class="st">&quot;Phylum&quot;</span>]))</span>
<span id="cb94-6"><a href="network-analysis.html#cb94-6" tabindex="-1"></a><span class="fu">names</span>(phyla) <span class="ot">&lt;-</span> taxatab<span class="sc">$</span>Genus</span>
<span id="cb94-7"><a href="network-analysis.html#cb94-7" tabindex="-1"></a></span>
<span id="cb94-8"><a href="network-analysis.html#cb94-8" tabindex="-1"></a><span class="co"># Define phylum colors</span></span>
<span id="cb94-9"><a href="network-analysis.html#cb94-9" tabindex="-1"></a>phylcol <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;cyan&quot;</span>, <span class="st">&quot;blue3&quot;</span>, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;lawngreen&quot;</span>, <span class="st">&quot;yellow&quot;</span>, <span class="st">&quot;deeppink&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;black&quot;</span>, <span class="st">&quot;green&quot;</span>)</span>
<span id="cb94-10"><a href="network-analysis.html#cb94-10" tabindex="-1"></a></span>
<span id="cb94-11"><a href="network-analysis.html#cb94-11" tabindex="-1"></a><span class="fu">plot</span>(props_single,</span>
<span id="cb94-12"><a href="network-analysis.html#cb94-12" tabindex="-1"></a>     <span class="at">layout =</span> <span class="st">&quot;spring&quot;</span>,</span>
<span id="cb94-13"><a href="network-analysis.html#cb94-13" tabindex="-1"></a>     <span class="at">repulsion =</span> <span class="fl">0.84</span>,</span>
<span id="cb94-14"><a href="network-analysis.html#cb94-14" tabindex="-1"></a>     <span class="at">shortenLabels =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb94-15"><a href="network-analysis.html#cb94-15" tabindex="-1"></a>     <span class="at">charToRm =</span> <span class="st">&quot;g__&quot;</span>,</span>
<span id="cb94-16"><a href="network-analysis.html#cb94-16" tabindex="-1"></a>     <span class="at">labelLength =</span> <span class="dv">30</span>,</span>
<span id="cb94-17"><a href="network-analysis.html#cb94-17" tabindex="-1"></a>     <span class="at">labelScale =</span> <span class="cn">FALSE</span>,</span>
<span id="cb94-18"><a href="network-analysis.html#cb94-18" tabindex="-1"></a>     <span class="at">rmSingles =</span> <span class="cn">TRUE</span>,</span>
<span id="cb94-19"><a href="network-analysis.html#cb94-19" tabindex="-1"></a>     <span class="at">nodeSize =</span> <span class="st">&quot;clr&quot;</span>,</span>
<span id="cb94-20"><a href="network-analysis.html#cb94-20" tabindex="-1"></a>     <span class="at">nodeSizeSpread =</span> <span class="dv">4</span>,</span>
<span id="cb94-21"><a href="network-analysis.html#cb94-21" tabindex="-1"></a>     <span class="at">nodeColor =</span> <span class="st">&quot;feature&quot;</span>, </span>
<span id="cb94-22"><a href="network-analysis.html#cb94-22" tabindex="-1"></a>     <span class="at">featVecCol =</span> phyla, </span>
<span id="cb94-23"><a href="network-analysis.html#cb94-23" tabindex="-1"></a>     <span class="at">colorVec =</span>  phylcol,</span>
<span id="cb94-24"><a href="network-analysis.html#cb94-24" tabindex="-1"></a>     <span class="at">posCol =</span> <span class="st">&quot;darkturquoise&quot;</span>, </span>
<span id="cb94-25"><a href="network-analysis.html#cb94-25" tabindex="-1"></a>     <span class="at">negCol =</span> <span class="st">&quot;orange&quot;</span>,</span>
<span id="cb94-26"><a href="network-analysis.html#cb94-26" tabindex="-1"></a>     <span class="at">edgeTranspLow =</span> <span class="dv">0</span>,</span>
<span id="cb94-27"><a href="network-analysis.html#cb94-27" tabindex="-1"></a>     <span class="at">edgeTranspHigh =</span> <span class="dv">40</span>,</span>
<span id="cb94-28"><a href="network-analysis.html#cb94-28" tabindex="-1"></a>     <span class="at">cexNodes =</span> <span class="dv">2</span>,</span>
<span id="cb94-29"><a href="network-analysis.html#cb94-29" tabindex="-1"></a>     <span class="at">cexLabels =</span> <span class="dv">2</span>,</span>
<span id="cb94-30"><a href="network-analysis.html#cb94-30" tabindex="-1"></a>     <span class="at">cexHubLabels =</span> <span class="fl">2.5</span>,</span>
<span id="cb94-31"><a href="network-analysis.html#cb94-31" tabindex="-1"></a>     <span class="at">title1 =</span> <span class="st">&quot;Network on genus level with Pearson correlations&quot;</span>, </span>
<span id="cb94-32"><a href="network-analysis.html#cb94-32" tabindex="-1"></a>     <span class="at">showTitle =</span> <span class="cn">TRUE</span>,</span>
<span id="cb94-33"><a href="network-analysis.html#cb94-33" tabindex="-1"></a>     <span class="at">cexTitle =</span> <span class="fl">2.3</span>)</span>
<span id="cb94-34"><a href="network-analysis.html#cb94-34" tabindex="-1"></a></span>
<span id="cb94-35"><a href="network-analysis.html#cb94-35" tabindex="-1"></a><span class="co"># Colors used in the legend should be equally transparent as in the plot</span></span>
<span id="cb94-36"><a href="network-analysis.html#cb94-36" tabindex="-1"></a>phylcol_transp <span class="ot">&lt;-</span> NetCoMi<span class="sc">:::</span><span class="fu">colToTransp</span>(phylcol, <span class="dv">60</span>)</span>
<span id="cb94-37"><a href="network-analysis.html#cb94-37" tabindex="-1"></a></span>
<span id="cb94-38"><a href="network-analysis.html#cb94-38" tabindex="-1"></a><span class="fu">legend</span>(<span class="sc">-</span><span class="fl">1.2</span>, <span class="fl">1.2</span>, <span class="at">cex =</span> <span class="dv">2</span>, <span class="at">pt.cex =</span> <span class="fl">2.5</span>, <span class="at">title =</span> <span class="st">&quot;Phylum:&quot;</span>, </span>
<span id="cb94-39"><a href="network-analysis.html#cb94-39" tabindex="-1"></a>       <span class="at">legend=</span><span class="fu">levels</span>(phyla), <span class="at">col =</span> phylcol_transp, <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">pch =</span> <span class="dv">16</span>) </span>
<span id="cb94-40"><a href="network-analysis.html#cb94-40" tabindex="-1"></a></span>
<span id="cb94-41"><a href="network-analysis.html#cb94-41" tabindex="-1"></a><span class="fu">legend</span>(<span class="fl">0.7</span>, <span class="fl">1.1</span>, <span class="at">cex =</span> <span class="fl">2.2</span>, <span class="at">title =</span> <span class="st">&quot;estimated correlation:&quot;</span>,</span>
<span id="cb94-42"><a href="network-analysis.html#cb94-42" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span>,<span class="st">&quot;-&quot;</span>), <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;darkturquoise&quot;</span>,<span class="st">&quot;orange&quot;</span>), </span>
<span id="cb94-43"><a href="network-analysis.html#cb94-43" tabindex="-1"></a>       <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">horiz =</span> <span class="cn">TRUE</span>)  </span>
<span id="cb94-44"><a href="network-analysis.html#cb94-44" tabindex="-1"></a>}</span></code></pre></div>
</div>
</div>
<div id="dissimilarity-based-networks" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> Dissimilarity-based Networks<a href="network-analysis.html#dissimilarity-based-networks" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<blockquote>
<p>If a dissimilarity measure is used for network construction, nodes are subjects instead of OTUs. The estimated dissimilarities are transformed into similarities, which are used as edge weights so that subjects with a similar microbial composition are placed close together in the network plot.</p>
</blockquote>
<blockquote>
<p>We construct a single network using Aitchison’s distance being suitable for the application on compositional data.</p>
</blockquote>
<blockquote>
<p>Since the Aitchison distance is based on the clr-transformation, zeros in the data need to be replaced.</p>
</blockquote>
<blockquote>
<p>The network is sparsified using the k-nearest neighbor (knn) algorithm.</p>
</blockquote>
<div id="building-network-module" class="section level3 hasAnchor" number="9.3.1">
<h3><span class="header-section-number">9.3.1</span> Building network module<a href="network-analysis.html#building-network-module" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="network-analysis.html#cb95-1" tabindex="-1"></a>net_aitchison <span class="ot">&lt;-</span> <span class="fu">netConstruct</span>(metaphlan2_ps_genus_filter,</span>
<span id="cb95-2"><a href="network-analysis.html#cb95-2" tabindex="-1"></a>                              <span class="at">measure =</span> <span class="st">&quot;aitchison&quot;</span>,</span>
<span id="cb95-3"><a href="network-analysis.html#cb95-3" tabindex="-1"></a>                              <span class="at">zeroMethod =</span> <span class="st">&quot;multRepl&quot;</span>,</span>
<span id="cb95-4"><a href="network-analysis.html#cb95-4" tabindex="-1"></a>                              <span class="at">sparsMethod =</span> <span class="st">&quot;knn&quot;</span>, </span>
<span id="cb95-5"><a href="network-analysis.html#cb95-5" tabindex="-1"></a>                              <span class="at">kNeighbor =</span> <span class="dv">3</span>,</span>
<span id="cb95-6"><a href="network-analysis.html#cb95-6" tabindex="-1"></a>                              <span class="at">verbose =</span> <span class="dv">3</span>)</span></code></pre></div>
<pre><code>## Infos about changed arguments:</code></pre>
<pre><code>## Counts normalized to fractions for measure &#39;aitchison&#39;.</code></pre>
<pre><code>## 51 taxa and 22 samples remaining.</code></pre>
<pre><code>## 
## Zero treatment:</code></pre>
<pre><code>## Execute multRepl() ...</code></pre>
<pre><code>## Warning in (function (X, label = NULL, dl = NULL, frac = 0.65, imp.missing = FALSE, : Column(s) containing more than 80% zeros/unobserved values were found (check it out using zPatterns).
##                     (You can use the z.warning argument to modify the warning threshold).</code></pre>
<pre><code>## Done.
## 
## Normalization:
## Counts normalized by total sum scaling.
## 
## Calculate &#39;aitchison&#39; dissimilarities ... Done.
## 
## Sparsify dissimilarities via &#39;knn&#39; ... Done.</code></pre>
</div>
<div id="analyzing-the-constructed-network-1" class="section level3 hasAnchor" number="9.3.2">
<h3><span class="header-section-number">9.3.2</span> Analyzing the constructed network<a href="network-analysis.html#analyzing-the-constructed-network-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>NetCoMi’s <code>netAnalyze()</code> function is used for analyzing the constructed network(s).</p>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="network-analysis.html#cb103-1" tabindex="-1"></a>props_aitchison <span class="ot">&lt;-</span> <span class="fu">netAnalyze</span>(net_aitchison,</span>
<span id="cb103-2"><a href="network-analysis.html#cb103-2" tabindex="-1"></a>                              <span class="at">clustMethod =</span> <span class="st">&quot;hierarchical&quot;</span>,</span>
<span id="cb103-3"><a href="network-analysis.html#cb103-3" tabindex="-1"></a>                              <span class="at">clustPar =</span> <span class="fu">list</span>(<span class="at">method =</span> <span class="st">&quot;average&quot;</span>, <span class="at">k =</span> <span class="dv">3</span>),</span>
<span id="cb103-4"><a href="network-analysis.html#cb103-4" tabindex="-1"></a>                              <span class="at">hubPar =</span> <span class="st">&quot;eigenvector&quot;</span>)</span>
<span id="cb103-5"><a href="network-analysis.html#cb103-5" tabindex="-1"></a></span>
<span id="cb103-6"><a href="network-analysis.html#cb103-6" tabindex="-1"></a><span class="fu">summary</span>(props_aitchison, <span class="at">numbNodes =</span> 5L)</span></code></pre></div>
<pre><code>## 
## Component sizes
## ```````````````        
## size: 22
##    #:  1
## ______________________________
## Global network properties
## `````````````````````````
##                                   
## Number of components       1.00000
## Clustering coefficient     0.32105
## Modularity                 0.27275
## Positive edge percentage 100.00000
## Edge density               0.20346
## Natural connectivity       0.13451
## Vertex connectivity        2.00000
## Edge connectivity          3.00000
## Average dissimilarity*     0.84117
## Average path length**      0.53694
## 
##  *Dissimilarity = 1 - edge weight
## **Path length: Units with average dissimilarity
## 
## ______________________________
## Clusters
## - In the whole network
## - Algorithm: hierarchical
## ````````````````````````` 
##             
## name:  1 2 3
##    #: 13 5 4
## 
## ______________________________
## Hubs
## - In alphabetical/numerical order
## - Based on empirical quantiles of centralities
## ```````````````````````````````````````````````    
##  s10
##  s7 
## 
## ______________________________
## Centrality measures
## - In decreasing order
## - Computed for the complete network
## ````````````````````````````````````
## Degree (normalized):
##            
## s10 0.47619
## s7  0.42857
## s8  0.33333
## s12 0.28571
## s16 0.23810
## 
## Betweenness centrality (normalized):
##            
## s10 0.37143
## s7  0.23333
## s8  0.19524
## s20 0.15238
## s6  0.09524
## 
## Closeness centrality (normalized):
##            
## s20 3.78684
## s10 3.70033
## s8  3.36495
## s14 3.34778
## s7  3.30724
## 
## Eigenvector centrality (normalized):
##            
## s7  1.00000
## s10 0.93516
## s8  0.74209
## s12 0.68679
## s22 0.65236</code></pre>
</div>
<div id="visualizing-the-network-1" class="section level3 hasAnchor" number="9.3.3">
<h3><span class="header-section-number">9.3.3</span> Visualizing the network<a href="network-analysis.html#visualizing-the-network-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>primary plot</li>
</ul>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="network-analysis.html#cb105-1" tabindex="-1"></a><span class="fu">plot</span>(props_aitchison, </span>
<span id="cb105-2"><a href="network-analysis.html#cb105-2" tabindex="-1"></a>     <span class="at">nodeColor =</span> <span class="st">&quot;cluster&quot;</span>, </span>
<span id="cb105-3"><a href="network-analysis.html#cb105-3" tabindex="-1"></a>     <span class="at">nodeSize =</span> <span class="st">&quot;eigenvector&quot;</span>,</span>
<span id="cb105-4"><a href="network-analysis.html#cb105-4" tabindex="-1"></a>     <span class="at">repulsion =</span> <span class="fl">0.8</span>,</span>
<span id="cb105-5"><a href="network-analysis.html#cb105-5" tabindex="-1"></a>     <span class="at">rmSingles =</span> <span class="cn">TRUE</span>,</span>
<span id="cb105-6"><a href="network-analysis.html#cb105-6" tabindex="-1"></a>     <span class="at">labelScale =</span> <span class="cn">FALSE</span>,</span>
<span id="cb105-7"><a href="network-analysis.html#cb105-7" tabindex="-1"></a>     <span class="at">cexLabels =</span> <span class="fl">1.6</span>,</span>
<span id="cb105-8"><a href="network-analysis.html#cb105-8" tabindex="-1"></a>     <span class="at">nodeSizeSpread =</span> <span class="dv">3</span>,</span>
<span id="cb105-9"><a href="network-analysis.html#cb105-9" tabindex="-1"></a>     <span class="at">cexNodes =</span> <span class="fl">1.5</span>,</span>
<span id="cb105-10"><a href="network-analysis.html#cb105-10" tabindex="-1"></a>     <span class="at">title1 =</span> <span class="st">&quot;Network on metagenomics with Aitchison distance&quot;</span>, </span>
<span id="cb105-11"><a href="network-analysis.html#cb105-11" tabindex="-1"></a>     <span class="at">showTitle =</span> <span class="cn">TRUE</span>,</span>
<span id="cb105-12"><a href="network-analysis.html#cb105-12" tabindex="-1"></a>     <span class="at">cexTitle =</span> <span class="fl">1.5</span>,</span>
<span id="cb105-13"><a href="network-analysis.html#cb105-13" tabindex="-1"></a>     <span class="at">hubTransp =</span> <span class="dv">50</span>,</span>
<span id="cb105-14"><a href="network-analysis.html#cb105-14" tabindex="-1"></a>     <span class="at">edgeTranspLow =</span> <span class="dv">60</span>,</span>
<span id="cb105-15"><a href="network-analysis.html#cb105-15" tabindex="-1"></a>     <span class="at">charToRm =</span> <span class="st">&quot;00000&quot;</span>,</span>
<span id="cb105-16"><a href="network-analysis.html#cb105-16" tabindex="-1"></a>     <span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">5</span>))</span>
<span id="cb105-17"><a href="network-analysis.html#cb105-17" tabindex="-1"></a></span>
<span id="cb105-18"><a href="network-analysis.html#cb105-18" tabindex="-1"></a><span class="co"># get green color with 50% transparency</span></span>
<span id="cb105-19"><a href="network-analysis.html#cb105-19" tabindex="-1"></a>green2 <span class="ot">&lt;-</span> <span class="fu">colToTransp</span>(<span class="st">&quot;#009900&quot;</span>, <span class="dv">40</span>)</span>
<span id="cb105-20"><a href="network-analysis.html#cb105-20" tabindex="-1"></a></span>
<span id="cb105-21"><a href="network-analysis.html#cb105-21" tabindex="-1"></a><span class="fu">legend</span>(<span class="fl">0.4</span>, <span class="fl">1.1</span>,</span>
<span id="cb105-22"><a href="network-analysis.html#cb105-22" tabindex="-1"></a>       <span class="at">cex =</span> <span class="fl">1.5</span>,</span>
<span id="cb105-23"><a href="network-analysis.html#cb105-23" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;high similarity (low Aitchison distance)&quot;</span>,</span>
<span id="cb105-24"><a href="network-analysis.html#cb105-24" tabindex="-1"></a>                  <span class="st">&quot;low similarity (high Aitchison distance)&quot;</span>), </span>
<span id="cb105-25"><a href="network-analysis.html#cb105-25" tabindex="-1"></a>       <span class="at">lty =</span> <span class="dv">1</span>, </span>
<span id="cb105-26"><a href="network-analysis.html#cb105-26" tabindex="-1"></a>       <span class="at">lwd =</span> <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">1</span>),</span>
<span id="cb105-27"><a href="network-analysis.html#cb105-27" tabindex="-1"></a>       <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;darkgreen&quot;</span>, green2),</span>
<span id="cb105-28"><a href="network-analysis.html#cb105-28" tabindex="-1"></a>       <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-60"></span>
<img src="MGS-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-60-1.png" alt="Dissimilarity-based Networks" width="1152" />
<p class="caption">
Figure 9.4: Dissimilarity-based Networks
</p>
</div>
<ul>
<li>plot samples with group information</li>
</ul>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="network-analysis.html#cb106-1" tabindex="-1"></a><span class="co"># Get gorup names from the sample table created before</span></span>
<span id="cb106-2"><a href="network-analysis.html#cb106-2" tabindex="-1"></a>sampletab <span class="ot">&lt;-</span> metaphlan2_ps_genus_filter<span class="sc">@</span>sam_data <span class="sc">%&gt;%</span></span>
<span id="cb106-3"><a href="network-analysis.html#cb106-3" tabindex="-1"></a>  <span class="fu">data.frame</span>()</span>
<span id="cb106-4"><a href="network-analysis.html#cb106-4" tabindex="-1"></a>group_labels <span class="ot">&lt;-</span> <span class="fu">factor</span>(sampletab<span class="sc">$</span>Group)</span>
<span id="cb106-5"><a href="network-analysis.html#cb106-5" tabindex="-1"></a><span class="fu">names</span>(group_labels) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(sampletab)</span>
<span id="cb106-6"><a href="network-analysis.html#cb106-6" tabindex="-1"></a></span>
<span id="cb106-7"><a href="network-analysis.html#cb106-7" tabindex="-1"></a><span class="co"># Define group colors</span></span>
<span id="cb106-8"><a href="network-analysis.html#cb106-8" tabindex="-1"></a>group_lcol <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;blue3&quot;</span>, <span class="st">&quot;red&quot;</span>)</span>
<span id="cb106-9"><a href="network-analysis.html#cb106-9" tabindex="-1"></a></span>
<span id="cb106-10"><a href="network-analysis.html#cb106-10" tabindex="-1"></a><span class="fu">plot</span>(props_aitchison,</span>
<span id="cb106-11"><a href="network-analysis.html#cb106-11" tabindex="-1"></a>     <span class="at">layout =</span> <span class="st">&quot;spring&quot;</span>,</span>
<span id="cb106-12"><a href="network-analysis.html#cb106-12" tabindex="-1"></a>     <span class="at">repulsion =</span> <span class="fl">0.84</span>,</span>
<span id="cb106-13"><a href="network-analysis.html#cb106-13" tabindex="-1"></a>     <span class="at">shortenLabels =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb106-14"><a href="network-analysis.html#cb106-14" tabindex="-1"></a>     <span class="at">labelScale =</span> <span class="cn">FALSE</span>,</span>
<span id="cb106-15"><a href="network-analysis.html#cb106-15" tabindex="-1"></a>     <span class="at">rmSingles =</span> <span class="cn">TRUE</span>,</span>
<span id="cb106-16"><a href="network-analysis.html#cb106-16" tabindex="-1"></a>     <span class="at">nodeSize =</span> <span class="st">&quot;eigenvector&quot;</span>,</span>
<span id="cb106-17"><a href="network-analysis.html#cb106-17" tabindex="-1"></a>     <span class="at">nodeSizeSpread =</span> <span class="dv">4</span>,</span>
<span id="cb106-18"><a href="network-analysis.html#cb106-18" tabindex="-1"></a>     <span class="at">nodeColor =</span> <span class="st">&quot;feature&quot;</span>, </span>
<span id="cb106-19"><a href="network-analysis.html#cb106-19" tabindex="-1"></a>     <span class="at">featVecCol =</span> group_labels, </span>
<span id="cb106-20"><a href="network-analysis.html#cb106-20" tabindex="-1"></a>     <span class="at">colorVec =</span>  group_lcol,</span>
<span id="cb106-21"><a href="network-analysis.html#cb106-21" tabindex="-1"></a>     <span class="at">posCol =</span> <span class="st">&quot;darkturquoise&quot;</span>, </span>
<span id="cb106-22"><a href="network-analysis.html#cb106-22" tabindex="-1"></a>     <span class="at">negCol =</span> <span class="st">&quot;orange&quot;</span>,</span>
<span id="cb106-23"><a href="network-analysis.html#cb106-23" tabindex="-1"></a>     <span class="at">edgeTranspLow =</span> <span class="dv">0</span>,</span>
<span id="cb106-24"><a href="network-analysis.html#cb106-24" tabindex="-1"></a>     <span class="at">edgeTranspHigh =</span> <span class="dv">40</span>,</span>
<span id="cb106-25"><a href="network-analysis.html#cb106-25" tabindex="-1"></a>     <span class="at">cexNodes =</span> <span class="dv">2</span>,</span>
<span id="cb106-26"><a href="network-analysis.html#cb106-26" tabindex="-1"></a>     <span class="at">cexLabels =</span> <span class="dv">2</span>,</span>
<span id="cb106-27"><a href="network-analysis.html#cb106-27" tabindex="-1"></a>     <span class="at">cexHubLabels =</span> <span class="fl">2.5</span>,</span>
<span id="cb106-28"><a href="network-analysis.html#cb106-28" tabindex="-1"></a>     <span class="at">title1 =</span> <span class="st">&quot;Network on genus level with Aitchison distance&quot;</span>, </span>
<span id="cb106-29"><a href="network-analysis.html#cb106-29" tabindex="-1"></a>     <span class="at">showTitle =</span> <span class="cn">TRUE</span>,</span>
<span id="cb106-30"><a href="network-analysis.html#cb106-30" tabindex="-1"></a>     <span class="at">cexTitle =</span> <span class="fl">2.3</span>)</span>
<span id="cb106-31"><a href="network-analysis.html#cb106-31" tabindex="-1"></a></span>
<span id="cb106-32"><a href="network-analysis.html#cb106-32" tabindex="-1"></a><span class="co"># Colors used in the legend should be equally transparent as in the plot</span></span>
<span id="cb106-33"><a href="network-analysis.html#cb106-33" tabindex="-1"></a>group_transp <span class="ot">&lt;-</span> NetCoMi<span class="sc">:::</span><span class="fu">colToTransp</span>(group_lcol, <span class="dv">60</span>)</span>
<span id="cb106-34"><a href="network-analysis.html#cb106-34" tabindex="-1"></a></span>
<span id="cb106-35"><a href="network-analysis.html#cb106-35" tabindex="-1"></a><span class="fu">legend</span>(<span class="sc">-</span><span class="fl">1.2</span>, <span class="fl">1.2</span>, <span class="at">cex =</span> <span class="dv">2</span>, <span class="at">pt.cex =</span> <span class="fl">2.5</span>, <span class="at">title =</span> <span class="st">&quot;Group:&quot;</span>, </span>
<span id="cb106-36"><a href="network-analysis.html#cb106-36" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">levels</span>(group_labels), <span class="at">col =</span> group_transp, <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">pch =</span> <span class="dv">16</span>) </span>
<span id="cb106-37"><a href="network-analysis.html#cb106-37" tabindex="-1"></a></span>
<span id="cb106-38"><a href="network-analysis.html#cb106-38" tabindex="-1"></a><span class="co"># get green color with 50% transparency</span></span>
<span id="cb106-39"><a href="network-analysis.html#cb106-39" tabindex="-1"></a>green2 <span class="ot">&lt;-</span> <span class="fu">colToTransp</span>(<span class="st">&quot;#009900&quot;</span>, <span class="dv">40</span>)</span>
<span id="cb106-40"><a href="network-analysis.html#cb106-40" tabindex="-1"></a></span>
<span id="cb106-41"><a href="network-analysis.html#cb106-41" tabindex="-1"></a><span class="fu">legend</span>(<span class="fl">0.4</span>, <span class="fl">1.1</span>,</span>
<span id="cb106-42"><a href="network-analysis.html#cb106-42" tabindex="-1"></a>       <span class="at">cex =</span> <span class="fl">1.5</span>,</span>
<span id="cb106-43"><a href="network-analysis.html#cb106-43" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;high similarity (low Aitchison distance)&quot;</span>,</span>
<span id="cb106-44"><a href="network-analysis.html#cb106-44" tabindex="-1"></a>                  <span class="st">&quot;low similarity (high Aitchison distance)&quot;</span>), </span>
<span id="cb106-45"><a href="network-analysis.html#cb106-45" tabindex="-1"></a>       <span class="at">lty =</span> <span class="dv">1</span>, </span>
<span id="cb106-46"><a href="network-analysis.html#cb106-46" tabindex="-1"></a>       <span class="at">lwd =</span> <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">1</span>),</span>
<span id="cb106-47"><a href="network-analysis.html#cb106-47" tabindex="-1"></a>       <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;darkgreen&quot;</span>, green2),</span>
<span id="cb106-48"><a href="network-analysis.html#cb106-48" tabindex="-1"></a>       <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-61"></span>
<img src="MGS-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-61-1.png" alt="Dissimilarity-based Networks (group information)" width="1920" />
<p class="caption">
Figure 9.5: Dissimilarity-based Networks (group information)
</p>
</div>
</div>
</div>
<div id="systematic-information-7" class="section level2 hasAnchor" number="9.4">
<h2><span class="header-section-number">9.4</span> Systematic Information<a href="network-analysis.html#systematic-information-7" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="network-analysis.html#cb107-1" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.1.3 (2022-03-10)
##  os       macOS Monterey 12.2.1
##  system   x86_64, darwin17.0
##  ui       RStudio
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       Asia/Shanghai
##  date     2023-10-27
##  rstudio  2023.09.0+463 Desert Sunflower (desktop)
##  pandoc   3.1.1 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/ (via rmarkdown)
## 
## ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version   date (UTC) lib source
##  abind                  1.4-5     2016-07-21 [2] CRAN (R 4.1.0)
##  ade4                   1.7-22    2023-02-06 [2] CRAN (R 4.1.2)
##  ALDEx2                 1.30.0    2022-11-01 [2] Bioconductor
##  annotate               1.72.0    2021-10-26 [2] Bioconductor
##  AnnotationDbi          1.60.2    2023-03-10 [2] Bioconductor
##  ape                  * 5.7-1     2023-03-13 [2] CRAN (R 4.1.2)
##  askpass                1.1       2019-01-13 [2] CRAN (R 4.1.0)
##  backports              1.4.1     2021-12-13 [2] CRAN (R 4.1.0)
##  base64enc              0.1-3     2015-07-28 [2] CRAN (R 4.1.0)
##  bayesm                 3.1-5     2022-12-02 [2] CRAN (R 4.1.2)
##  Biobase              * 2.54.0    2021-10-26 [2] Bioconductor
##  BiocGenerics         * 0.40.0    2021-10-26 [2] Bioconductor
##  BiocParallel           1.28.3    2021-12-09 [2] Bioconductor
##  biomformat             1.22.0    2021-10-26 [2] Bioconductor
##  Biostrings             2.62.0    2021-10-26 [2] Bioconductor
##  bit                    4.0.5     2022-11-15 [2] CRAN (R 4.1.2)
##  bit64                  4.0.5     2020-08-30 [2] CRAN (R 4.1.0)
##  bitops                 1.0-7     2021-04-24 [2] CRAN (R 4.1.0)
##  blob                   1.2.4     2023-03-17 [2] CRAN (R 4.1.2)
##  bookdown               0.34      2023-05-09 [2] CRAN (R 4.1.2)
##  broom                  1.0.5     2023-06-09 [2] CRAN (R 4.1.3)
##  bslib                  0.5.0     2023-06-09 [2] CRAN (R 4.1.3)
##  cachem                 1.0.8     2023-05-01 [2] CRAN (R 4.1.2)
##  callr                  3.7.3     2022-11-02 [2] CRAN (R 4.1.2)
##  car                    3.1-2     2023-03-30 [2] CRAN (R 4.1.2)
##  carData                3.0-5     2022-01-06 [2] CRAN (R 4.1.2)
##  caTools                1.18.2    2021-03-28 [2] CRAN (R 4.1.0)
##  cccd                   1.6       2022-04-08 [2] CRAN (R 4.1.2)
##  cellranger             1.1.0     2016-07-27 [2] CRAN (R 4.1.0)
##  checkmate              2.2.0     2023-04-27 [2] CRAN (R 4.1.2)
##  class                  7.3-22    2023-05-03 [2] CRAN (R 4.1.2)
##  classInt               0.4-9     2023-02-28 [2] CRAN (R 4.1.2)
##  cli                    3.6.1     2023-03-23 [2] CRAN (R 4.1.2)
##  cluster                2.1.4     2022-08-22 [2] CRAN (R 4.1.2)
##  coda                   0.19-4    2020-09-30 [2] CRAN (R 4.1.0)
##  codetools              0.2-19    2023-02-01 [2] CRAN (R 4.1.2)
##  coin                   1.4-2     2021-10-08 [2] CRAN (R 4.1.0)
##  colorspace             2.1-0     2023-01-23 [2] CRAN (R 4.1.2)
##  compositions           2.0-6     2023-04-13 [2] CRAN (R 4.1.2)
##  conflicted           * 1.2.0     2023-02-01 [2] CRAN (R 4.1.2)
##  corpcor                1.6.10    2021-09-16 [2] CRAN (R 4.1.0)
##  corrplot               0.92      2021-11-18 [2] CRAN (R 4.1.0)
##  cowplot                1.1.1     2020-12-30 [2] CRAN (R 4.1.0)
##  crayon                 1.5.2     2022-09-29 [2] CRAN (R 4.1.2)
##  crosstalk              1.2.0     2021-11-04 [2] CRAN (R 4.1.0)
##  data.table             1.14.8    2023-02-17 [2] CRAN (R 4.1.2)
##  DBI                    1.1.3     2022-06-18 [2] CRAN (R 4.1.2)
##  DelayedArray           0.20.0    2021-10-26 [2] Bioconductor
##  deldir                 1.0-9     2023-05-17 [2] CRAN (R 4.1.3)
##  DEoptimR               1.0-14    2023-06-09 [2] CRAN (R 4.1.3)
##  DESeq2                 1.34.0    2021-10-26 [2] Bioconductor
##  devtools             * 2.4.5     2022-10-11 [2] CRAN (R 4.1.2)
##  digest                 0.6.33    2023-07-07 [1] CRAN (R 4.1.3)
##  doParallel             1.0.17    2022-02-07 [2] CRAN (R 4.1.2)
##  doSNOW                 1.0.20    2022-02-04 [2] CRAN (R 4.1.2)
##  dplyr                * 1.1.2     2023-04-20 [2] CRAN (R 4.1.2)
##  DT                     0.28      2023-05-18 [2] CRAN (R 4.1.3)
##  dynamicTreeCut         1.63-1    2016-03-11 [2] CRAN (R 4.1.0)
##  e1071                  1.7-13    2023-02-01 [2] CRAN (R 4.1.2)
##  edgeR                  3.36.0    2021-10-26 [2] Bioconductor
##  ellipsis               0.3.2     2021-04-29 [2] CRAN (R 4.1.0)
##  emmeans                1.8.7     2023-06-23 [1] CRAN (R 4.1.3)
##  estimability           1.4.1     2022-08-05 [2] CRAN (R 4.1.2)
##  evaluate               0.21      2023-05-05 [2] CRAN (R 4.1.2)
##  FactoMineR             2.8       2023-03-27 [2] CRAN (R 4.1.2)
##  fansi                  1.0.4     2023-01-22 [2] CRAN (R 4.1.2)
##  farver                 2.1.1     2022-07-06 [2] CRAN (R 4.1.2)
##  fastcluster            1.2.3     2021-05-24 [2] CRAN (R 4.1.0)
##  fastmap                1.1.1     2023-02-24 [2] CRAN (R 4.1.2)
##  fdrtool                1.2.17    2021-11-13 [2] CRAN (R 4.1.0)
##  filematrix             1.3       2018-02-27 [2] CRAN (R 4.1.0)
##  flashClust             1.01-2    2012-08-21 [2] CRAN (R 4.1.0)
##  FNN                    1.1.3.2   2023-03-20 [2] CRAN (R 4.1.2)
##  foreach                1.5.2     2022-02-02 [2] CRAN (R 4.1.2)
##  foreign                0.8-84    2022-12-06 [2] CRAN (R 4.1.2)
##  forestplot             3.1.1     2022-12-06 [2] CRAN (R 4.1.2)
##  formatR                1.14      2023-01-17 [2] CRAN (R 4.1.2)
##  Formula                1.2-5     2023-02-24 [2] CRAN (R 4.1.2)
##  fs                     1.6.2     2023-04-25 [2] CRAN (R 4.1.2)
##  futile.logger        * 1.4.3     2016-07-10 [2] CRAN (R 4.1.0)
##  futile.options         1.0.1     2018-04-20 [2] CRAN (R 4.1.0)
##  genefilter             1.76.0    2021-10-26 [2] Bioconductor
##  geneplotter            1.72.0    2021-10-26 [2] Bioconductor
##  generics               0.1.3     2022-07-05 [2] CRAN (R 4.1.2)
##  GenomeInfoDb         * 1.30.1    2022-01-30 [2] Bioconductor
##  GenomeInfoDbData       1.2.7     2022-03-09 [2] Bioconductor
##  GenomicRanges        * 1.46.1    2021-11-18 [2] Bioconductor
##  ggiraph                0.8.7     2023-03-17 [2] CRAN (R 4.1.2)
##  ggiraphExtra           0.3.0     2020-10-06 [2] CRAN (R 4.1.2)
##  ggplot2              * 3.4.2     2023-04-03 [2] CRAN (R 4.1.2)
##  ggpubr               * 0.6.0     2023-02-10 [2] CRAN (R 4.1.2)
##  ggrepel                0.9.3     2023-02-03 [2] CRAN (R 4.1.2)
##  ggsci                  3.0.0     2023-03-08 [2] CRAN (R 4.1.2)
##  ggsignif               0.6.4     2022-10-13 [2] CRAN (R 4.1.2)
##  ggVennDiagram          1.2.2     2022-09-08 [2] CRAN (R 4.1.2)
##  glasso                 1.11      2019-10-01 [2] CRAN (R 4.1.0)
##  glmnet                 4.1-7     2023-03-23 [2] CRAN (R 4.1.2)
##  glue                 * 1.6.2     2022-02-24 [2] CRAN (R 4.1.2)
##  Gmisc                * 3.0.2     2023-03-13 [2] CRAN (R 4.1.2)
##  GO.db                  3.14.0    2022-04-11 [2] Bioconductor
##  gplots                 3.1.3     2022-04-25 [2] CRAN (R 4.1.2)
##  gridExtra              2.3       2017-09-09 [2] CRAN (R 4.1.0)
##  gtable                 0.3.3     2023-03-21 [2] CRAN (R 4.1.2)
##  gtools                 3.9.4     2022-11-27 [2] CRAN (R 4.1.2)
##  highr                  0.10      2022-12-22 [2] CRAN (R 4.1.2)
##  Hmisc                  5.1-0     2023-05-08 [2] CRAN (R 4.1.2)
##  hms                    1.1.3     2023-03-21 [2] CRAN (R 4.1.2)
##  htmlTable            * 2.4.1     2022-07-07 [2] CRAN (R 4.1.2)
##  htmltools              0.5.5     2023-03-23 [2] CRAN (R 4.1.2)
##  htmlwidgets            1.6.2     2023-03-17 [2] CRAN (R 4.1.2)
##  httpuv                 1.6.11    2023-05-11 [2] CRAN (R 4.1.3)
##  httr                   1.4.6     2023-05-08 [2] CRAN (R 4.1.2)
##  huge                   1.3.5     2021-06-30 [2] CRAN (R 4.1.0)
##  igraph                 1.5.0     2023-06-16 [1] CRAN (R 4.1.3)
##  impute                 1.68.0    2021-10-26 [2] Bioconductor
##  insight                0.19.3    2023-06-29 [2] CRAN (R 4.1.3)
##  IRanges              * 2.28.0    2021-10-26 [2] Bioconductor
##  irlba                  2.3.5.1   2022-10-03 [2] CRAN (R 4.1.2)
##  iterators              1.0.14    2022-02-05 [2] CRAN (R 4.1.2)
##  jpeg                   0.1-10    2022-11-29 [2] CRAN (R 4.1.2)
##  jquerylib              0.1.4     2021-04-26 [2] CRAN (R 4.1.0)
##  jsonlite               1.8.7     2023-06-29 [2] CRAN (R 4.1.3)
##  kableExtra             1.3.4     2021-02-20 [2] CRAN (R 4.1.2)
##  KEGGREST               1.34.0    2021-10-26 [2] Bioconductor
##  KernSmooth             2.23-22   2023-07-10 [2] CRAN (R 4.1.3)
##  knitr                  1.43      2023-05-25 [2] CRAN (R 4.1.3)
##  labeling               0.4.2     2020-10-20 [2] CRAN (R 4.1.0)
##  lambda.r               1.2.4     2019-09-18 [2] CRAN (R 4.1.0)
##  later                  1.3.1     2023-05-02 [2] CRAN (R 4.1.2)
##  lattice              * 0.21-8    2023-04-05 [2] CRAN (R 4.1.2)
##  lavaan                 0.6-15    2023-03-14 [2] CRAN (R 4.1.2)
##  leaps                  3.1       2020-01-16 [2] CRAN (R 4.1.0)
##  libcoin                1.0-9     2021-09-27 [2] CRAN (R 4.1.0)
##  lifecycle              1.0.3     2022-10-07 [2] CRAN (R 4.1.2)
##  limma                  3.50.3    2022-04-07 [2] Bioconductor
##  locfit                 1.5-9.8   2023-06-11 [2] CRAN (R 4.1.3)
##  LOCOM                  1.1       2022-08-05 [2] Github (yijuanhu/LOCOM@c181e0f)
##  lubridate              1.9.2     2023-02-10 [2] CRAN (R 4.1.2)
##  magrittr             * 2.0.3     2022-03-30 [2] CRAN (R 4.1.2)
##  MASS                   7.3-60    2023-05-04 [2] CRAN (R 4.1.2)
##  Matrix                 1.6-0     2023-07-08 [2] CRAN (R 4.1.3)
##  MatrixGenerics       * 1.6.0     2021-10-26 [2] Bioconductor
##  matrixStats          * 1.0.0     2023-06-02 [2] CRAN (R 4.1.3)
##  mbzinb                 0.2       2022-03-16 [2] local
##  memoise                2.0.1     2021-11-26 [2] CRAN (R 4.1.0)
##  metagenomeSeq          1.36.0    2021-10-26 [2] Bioconductor
##  mgcv                   1.8-42    2023-03-02 [2] CRAN (R 4.1.2)
##  microbiome             1.16.0    2021-10-26 [2] Bioconductor
##  mime                   0.12      2021-09-28 [2] CRAN (R 4.1.0)
##  miniUI                 0.1.1.1   2018-05-18 [2] CRAN (R 4.1.0)
##  mixedCCA               1.6.2     2022-09-09 [2] CRAN (R 4.1.2)
##  mnormt                 2.1.1     2022-09-26 [2] CRAN (R 4.1.2)
##  modeltools             0.2-23    2020-03-05 [2] CRAN (R 4.1.0)
##  multcomp               1.4-25    2023-06-20 [2] CRAN (R 4.1.3)
##  multcompView           0.1-9     2023-04-09 [2] CRAN (R 4.1.2)
##  multtest               2.50.0    2021-10-26 [2] Bioconductor
##  munsell                0.5.0     2018-06-12 [2] CRAN (R 4.1.0)
##  mvtnorm                1.2-2     2023-06-08 [2] CRAN (R 4.1.3)
##  mycor                  0.1.1     2018-04-10 [2] CRAN (R 4.1.0)
##  NADA                   1.6-1.1   2020-03-22 [2] CRAN (R 4.1.0)
##  NetCoMi              * 1.0.3     2022-07-14 [2] Github (stefpeschel/NetCoMi@d4d80d3)
##  nlme                 * 3.1-162   2023-01-31 [2] CRAN (R 4.1.2)
##  nnet                   7.3-19    2023-05-03 [2] CRAN (R 4.1.2)
##  openssl                2.0.6     2023-03-09 [2] CRAN (R 4.1.2)
##  pbapply                1.7-2     2023-06-27 [2] CRAN (R 4.1.3)
##  pbivnorm               0.6.0     2015-01-23 [2] CRAN (R 4.1.0)
##  pcaPP                  2.0-3     2022-10-24 [2] CRAN (R 4.1.2)
##  permute              * 0.9-7     2022-01-27 [2] CRAN (R 4.1.2)
##  pheatmap             * 1.0.12    2019-01-04 [2] CRAN (R 4.1.0)
##  phyloseq             * 1.38.0    2021-10-26 [2] Bioconductor
##  picante              * 1.8.2     2020-06-10 [2] CRAN (R 4.1.0)
##  pillar                 1.9.0     2023-03-22 [2] CRAN (R 4.1.2)
##  pkgbuild               1.4.2     2023-06-26 [2] CRAN (R 4.1.3)
##  pkgconfig              2.0.3     2019-09-22 [2] CRAN (R 4.1.0)
##  pkgload                1.3.2.1   2023-07-08 [2] CRAN (R 4.1.3)
##  plyr                   1.8.8     2022-11-11 [2] CRAN (R 4.1.2)
##  png                    0.1-8     2022-11-29 [2] CRAN (R 4.1.2)
##  ppcor                  1.1       2015-12-03 [2] CRAN (R 4.1.0)
##  preprocessCore         1.56.0    2021-10-26 [2] Bioconductor
##  prettyunits            1.1.1     2020-01-24 [2] CRAN (R 4.1.0)
##  processx               3.8.2     2023-06-30 [2] CRAN (R 4.1.3)
##  profvis                0.3.8     2023-05-02 [2] CRAN (R 4.1.2)
##  promises               1.2.0.1   2021-02-11 [2] CRAN (R 4.1.0)
##  protoclust             1.6.4     2022-04-01 [2] CRAN (R 4.1.2)
##  proxy                  0.4-27    2022-06-09 [2] CRAN (R 4.1.2)
##  ps                     1.7.5     2023-04-18 [2] CRAN (R 4.1.2)
##  pscl                   1.5.5.1   2023-05-10 [2] CRAN (R 4.1.2)
##  psych                  2.3.6     2023-06-21 [2] CRAN (R 4.1.3)
##  pulsar                 0.3.10    2023-01-26 [2] CRAN (R 4.1.2)
##  purrr                * 1.0.1     2023-01-10 [2] CRAN (R 4.1.2)
##  qgraph                 1.9.5     2023-05-16 [2] CRAN (R 4.1.3)
##  quadprog               1.5-8     2019-11-20 [2] CRAN (R 4.1.0)
##  qvalue                 2.26.0    2021-10-26 [2] Bioconductor
##  R6                     2.5.1     2021-08-19 [2] CRAN (R 4.1.0)
##  RAIDA                  1.0       2022-03-14 [2] local
##  rbibutils              2.2.13    2023-01-13 [2] CRAN (R 4.1.2)
##  RColorBrewer         * 1.1-3     2022-04-03 [2] CRAN (R 4.1.2)
##  Rcpp                 * 1.0.11    2023-07-06 [1] CRAN (R 4.1.3)
##  RcppZiggurat           0.1.6     2020-10-20 [2] CRAN (R 4.1.0)
##  RCurl                  1.98-1.12 2023-03-27 [2] CRAN (R 4.1.2)
##  Rdpack                 2.4       2022-07-20 [2] CRAN (R 4.1.2)
##  readr                * 2.1.4     2023-02-10 [2] CRAN (R 4.1.2)
##  readxl               * 1.4.3     2023-07-06 [2] CRAN (R 4.1.3)
##  remotes                2.4.2     2021-11-30 [2] CRAN (R 4.1.0)
##  reshape2               1.4.4     2020-04-09 [2] CRAN (R 4.1.0)
##  reticulate             1.30      2023-06-09 [2] CRAN (R 4.1.3)
##  Rfast                  2.0.8     2023-07-03 [2] CRAN (R 4.1.3)
##  rhdf5                  2.38.1    2022-03-10 [2] Bioconductor
##  rhdf5filters           1.6.0     2021-10-26 [2] Bioconductor
##  Rhdf5lib               1.16.0    2021-10-26 [2] Bioconductor
##  rlang                  1.1.1     2023-04-28 [1] CRAN (R 4.1.2)
##  rmarkdown              2.23      2023-07-01 [2] CRAN (R 4.1.3)
##  robustbase             0.99-0    2023-06-16 [2] CRAN (R 4.1.3)
##  rootSolve              1.8.2.3   2021-09-29 [2] CRAN (R 4.1.0)
##  rpart                  4.1.19    2022-10-21 [2] CRAN (R 4.1.2)
##  RSpectra               0.16-1    2022-04-24 [2] CRAN (R 4.1.2)
##  RSQLite                2.3.1     2023-04-03 [2] CRAN (R 4.1.2)
##  rstatix                0.7.2     2023-02-01 [2] CRAN (R 4.1.2)
##  rstudioapi             0.15.0    2023-07-07 [2] CRAN (R 4.1.3)
##  Rtsne                  0.16      2022-04-17 [2] CRAN (R 4.1.2)
##  RVenn                  1.1.0     2019-07-18 [2] CRAN (R 4.1.0)
##  rvest                  1.0.3     2022-08-19 [2] CRAN (R 4.1.2)
##  S4Vectors            * 0.32.4    2022-03-29 [2] Bioconductor
##  sandwich               3.0-2     2022-06-15 [2] CRAN (R 4.1.2)
##  sass                   0.4.6     2023-05-03 [2] CRAN (R 4.1.2)
##  scales                 1.2.1     2022-08-20 [2] CRAN (R 4.1.2)
##  scatterplot3d          0.3-44    2023-05-05 [2] CRAN (R 4.1.2)
##  sessioninfo            1.2.2     2021-12-06 [2] CRAN (R 4.1.0)
##  sf                     1.0-7     2022-03-07 [2] CRAN (R 4.1.2)
##  shape                  1.4.6     2021-05-19 [2] CRAN (R 4.1.0)
##  shiny                  1.7.4.1   2023-07-06 [2] CRAN (R 4.1.3)
##  sjlabelled             1.2.0     2022-04-10 [2] CRAN (R 4.1.2)
##  sjmisc                 2.8.9     2021-12-03 [2] CRAN (R 4.1.0)
##  snow                   0.4-4     2021-10-27 [2] CRAN (R 4.1.0)
##  SpiecEasi            * 1.1.2     2022-07-14 [2] Github (zdk123/SpiecEasi@c463727)
##  SPRING                 1.0.4     2022-08-03 [2] Github (GraceYoon/SPRING@3d641a4)
##  stringi                1.7.12    2023-01-11 [2] CRAN (R 4.1.2)
##  stringr                1.5.0     2022-12-02 [2] CRAN (R 4.1.2)
##  SummarizedExperiment * 1.24.0    2021-10-26 [2] Bioconductor
##  survival               3.5-5     2023-03-12 [2] CRAN (R 4.1.2)
##  svglite                2.1.1     2023-01-10 [2] CRAN (R 4.1.2)
##  systemfonts            1.0.4     2022-02-11 [2] CRAN (R 4.1.2)
##  tensorA                0.36.2    2020-11-19 [2] CRAN (R 4.1.0)
##  TH.data                1.1-2     2023-04-17 [2] CRAN (R 4.1.2)
##  tibble               * 3.2.1     2023-03-20 [2] CRAN (R 4.1.2)
##  tidyr                * 1.3.0     2023-01-24 [2] CRAN (R 4.1.2)
##  tidyselect             1.2.0     2022-10-10 [2] CRAN (R 4.1.2)
##  timechange             0.2.0     2023-01-11 [2] CRAN (R 4.1.2)
##  truncnorm              1.0-9     2023-03-20 [2] CRAN (R 4.1.2)
##  tzdb                   0.4.0     2023-05-12 [2] CRAN (R 4.1.3)
##  umap                   0.2.10.0  2023-02-01 [2] CRAN (R 4.1.2)
##  units                  0.8-2     2023-04-27 [2] CRAN (R 4.1.2)
##  urlchecker             1.0.1     2021-11-30 [2] CRAN (R 4.1.0)
##  usethis              * 2.2.2     2023-07-06 [2] CRAN (R 4.1.3)
##  utf8                   1.2.3     2023-01-31 [2] CRAN (R 4.1.2)
##  uuid                   1.1-0     2022-04-19 [2] CRAN (R 4.1.2)
##  vctrs                  0.6.3     2023-06-14 [1] CRAN (R 4.1.3)
##  vegan                * 2.6-4     2022-10-11 [2] CRAN (R 4.1.2)
##  VennDiagram          * 1.7.3     2022-04-12 [2] CRAN (R 4.1.2)
##  VGAM                   1.1-8     2023-03-09 [2] CRAN (R 4.1.2)
##  viridis              * 0.6.3     2023-05-03 [2] CRAN (R 4.1.2)
##  viridisLite          * 0.4.2     2023-05-02 [2] CRAN (R 4.1.2)
##  vroom                  1.6.3     2023-04-28 [2] CRAN (R 4.1.2)
##  webshot                0.5.5     2023-06-26 [2] CRAN (R 4.1.3)
##  WGCNA                  1.72-1    2023-01-18 [2] CRAN (R 4.1.2)
##  withr                  2.5.0     2022-03-03 [2] CRAN (R 4.1.2)
##  Wrench                 1.12.0    2021-10-26 [2] Bioconductor
##  xfun                   0.40      2023-08-09 [1] CRAN (R 4.1.3)
##  XMAS2                * 2.2.0     2023-10-27 [1] local
##  XML                    3.99-0.14 2023-03-19 [2] CRAN (R 4.1.2)
##  xml2                   1.3.5     2023-07-06 [2] CRAN (R 4.1.3)
##  xtable                 1.8-4     2019-04-21 [2] CRAN (R 4.1.0)
##  XVector                0.34.0    2021-10-26 [2] Bioconductor
##  yaml                   2.3.7     2023-01-23 [2] CRAN (R 4.1.2)
##  zCompositions          1.4.0-1   2022-03-26 [2] CRAN (R 4.1.2)
##  zlibbioc               1.40.0    2021-10-26 [2] Bioconductor
##  zoo                    1.8-12    2023-04-13 [2] CRAN (R 4.1.2)
## 
##  [1] /Users/zouhua/Library/R/x86_64/4.1/library
##  [2] /Library/Frameworks/R.framework/Versions/4.1/Resources/library
## 
## ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-peschel2021netcomi" class="csl-entry">
Peschel, Stefanie, Christian L Müller, Erika von Mutius, Anne-Laure Boulesteix, and Martin Depner. 2021. <span>“NetCoMi: Network Construction and Comparison for Microbiome Data in r.”</span> <em>Briefings in Bioinformatics</em> 22 (4): bbaa290.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="differential-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="test-example.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/MGS_DataAnalysis_template/blob/main/08-Network_Analysis.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["MGS microbiota data analysis based on XMAS package.pdf", "MGS microbiota data analysis based on XMAS package.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true,
"toc_depth": 3
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
