<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Quality Control | MGS microbiota data analysis based on XMAS package</title>
  <meta name="description" content="Chapter 3 Quality Control | MGS microbiota data analysis based on XMAS package" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Quality Control | MGS microbiota data analysis based on XMAS package" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Chapter 3 Quality Control | MGS microbiota data analysis based on XMAS package" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Quality Control | MGS microbiota data analysis based on XMAS package" />
  
  <meta name="twitter:description" content="Chapter 3 Quality Control | MGS microbiota data analysis based on XMAS package" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2022-05-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="convert-inputdata-into-phyloseq-object.html"/>
<link rel="next" href="data-processing.html"/>
<script src="libs/header-attrs-2.13/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/viz-1.8.2/viz.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/grViz-binding-1.0.9/grViz.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.21/datatables.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">MGS microbiota data analysis based on XMAS package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-tutorial"><i class="fa fa-check"></i><b>1.1</b> Preparing for the tutorial<span></span></a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#workflow-description"><i class="fa fa-check"></i><b>1.2</b> Workflow description<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html"><i class="fa fa-check"></i><b>2</b> Convert inputdata into phyloseq object<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metaphlan23"><i class="fa fa-check"></i><b>2.1</b> Metaphlan2/3<span></span></a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results"><i class="fa fa-check"></i><b>2.1.1</b> Importing results<span></span></a></li>
<li class="chapter" data-level="2.1.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table"><i class="fa fa-check"></i><b>2.1.2</b> taxa table<span></span></a></li>
<li class="chapter" data-level="2.1.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table"><i class="fa fa-check"></i><b>2.1.3</b> otu table<span></span></a></li>
<li class="chapter" data-level="2.1.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table"><i class="fa fa-check"></i><b>2.1.4</b> metadata table<span></span></a></li>
<li class="chapter" data-level="2.1.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object"><i class="fa fa-check"></i><b>2.1.5</b> phyloseq object<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#summarize-phyloseq-class-object"><i class="fa fa-check"></i><b>2.2</b> Summarize phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="2.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#systematic-information"><i class="fa fa-check"></i><b>2.3</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>3</b> Quality Control<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="quality-control.html"><a href="quality-control.html#spike-in-sample-brs"><i class="fa fa-check"></i><b>3.1</b> spike-in sample (BRS)<span></span></a></li>
<li class="chapter" data-level="3.2" data-path="quality-control.html"><a href="quality-control.html#spike-in-samples-brs-remove"><i class="fa fa-check"></i><b>3.2</b> Spike-in sample’s (BRS) remove<span></span></a></li>
<li class="chapter" data-level="3.3" data-path="quality-control.html"><a href="quality-control.html#systematic-information-1"><i class="fa fa-check"></i><b>3.3</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data-processing.html"><a href="data-processing.html"><i class="fa fa-check"></i><b>4</b> Data processing<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-processing.html"><a href="data-processing.html#extracting-specific-taxonomic-level"><i class="fa fa-check"></i><b>4.1</b> Extracting specific taxonomic level<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="data-processing.html"><a href="data-processing.html#filtering-the-low-relative-abundance-or-unclassified-taxa-by-the-threshold-total-counts-1e-4"><i class="fa fa-check"></i><b>4.2</b> Filtering the low relative abundance or unclassified taxa by the threshold (total counts &lt; 1e-4)<span></span></a></li>
<li class="chapter" data-level="4.3" data-path="data-processing.html"><a href="data-processing.html#trimming-the-taxa-with-low-occurrence-less-than-threshold"><i class="fa fa-check"></i><b>4.3</b> Trimming the taxa with low occurrence less than threshold<span></span></a></li>
<li class="chapter" data-level="4.4" data-path="data-processing.html"><a href="data-processing.html#systematic-information-2"><i class="fa fa-check"></i><b>4.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="diversity-analysis.html"><a href="diversity-analysis.html"><i class="fa fa-check"></i><b>5</b> Diversity analysis<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="diversity-analysis.html"><a href="diversity-analysis.html#alpha-diveristy"><i class="fa fa-check"></i><b>5.1</b> Alpha diveristy<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="diversity-analysis.html"><a href="diversity-analysis.html#beta-diversity"><i class="fa fa-check"></i><b>5.2</b> Beta diversity<span></span></a></li>
<li class="chapter" data-level="5.3" data-path="diversity-analysis.html"><a href="diversity-analysis.html#systematic-information-3"><i class="fa fa-check"></i><b>5.3</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html"><i class="fa fa-check"></i><b>6</b> PERMANOVA + beta dispersion + Ordination<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#permanova"><i class="fa fa-check"></i><b>6.1</b> PERMANOVA<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#beta-dispersion"><i class="fa fa-check"></i><b>6.2</b> beta dispersion<span></span></a></li>
<li class="chapter" data-level="6.3" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#ordination"><i class="fa fa-check"></i><b>6.3</b> Ordination<span></span></a></li>
<li class="chapter" data-level="6.4" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#systematic-information-4"><i class="fa fa-check"></i><b>6.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="microbial-composition.html"><a href="microbial-composition.html"><i class="fa fa-check"></i><b>7</b> Microbial composition<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="microbial-composition.html"><a href="microbial-composition.html#stacked-barplot"><i class="fa fa-check"></i><b>7.1</b> Stacked barplot<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="microbial-composition.html"><a href="microbial-composition.html#core-microbiota"><i class="fa fa-check"></i><b>7.2</b> Core microbiota<span></span></a></li>
<li class="chapter" data-level="7.3" data-path="microbial-composition.html"><a href="microbial-composition.html#systematic-information-5"><i class="fa fa-check"></i><b>7.3</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-analysis.html"><a href="differential-analysis.html"><i class="fa fa-check"></i><b>8</b> Differential Analysis<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse"><i class="fa fa-check"></i><b>8.1</b> Liner discriminant analysis (LDA) effect size (LEfSe)<span></span></a></li>
<li class="chapter" data-level="8.2" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-test"><i class="fa fa-check"></i><b>8.2</b> Wilcoxon Rank-Sum test<span></span></a></li>
<li class="chapter" data-level="8.3" data-path="differential-analysis.html"><a href="differential-analysis.html#dominant-taxa"><i class="fa fa-check"></i><b>8.3</b> Dominant taxa<span></span></a></li>
<li class="chapter" data-level="8.4" data-path="differential-analysis.html"><a href="differential-analysis.html#multiple-differential-analysis-by-one-function"><i class="fa fa-check"></i><b>8.4</b> Multiple differential analysis by one function<span></span></a></li>
<li class="chapter" data-level="8.5" data-path="differential-analysis.html"><a href="differential-analysis.html#systematic-information-6"><i class="fa fa-check"></i><b>8.5</b> Systematic Information<span></span></a></li>
</ul></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">MGS microbiota data analysis based on XMAS package</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="quality-control" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> Quality Control<a href="quality-control.html#quality-control" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><strong>Loading packages</strong></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="quality-control.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(XMAS)</span>
<span id="cb14-2"><a href="quality-control.html#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb14-3"><a href="quality-control.html#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb14-4"><a href="quality-control.html#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span></code></pre></div>
<p>The taxonomic levels of spike-in sample’s bacteria is species. Firstly, using the <code>summarize_taxa</code> to get the genus level phyloseq object and then do <code>run_RefCheck</code> under the optimal parameters.</p>
<div id="spike-in-sample-brs" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> spike-in sample (BRS)<a href="quality-control.html#spike-in-sample-brs" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>get the BRS_ID</li>
</ul>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="quality-control.html#cb15-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(metaphlan2_ps, <span class="at">taxa_level =</span> <span class="st">&quot;Species&quot;</span>)</span>
<span id="cb15-2"><a href="quality-control.html#cb15-2" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species<span class="sc">@</span>sam_data</span></code></pre></div>
<pre><code>##      Group phynotype
## s1      BB      0.00
## s2      AA      2.50
## s3      BB      0.00
## s4      AA      1.25
## s5      AA     30.00
## s6      AA     15.00
## s7      BB      8.75
## s8      BB      0.00
## s9      BB      3.75
## s10     BB      2.50
## s11     BB     15.00
## s12     BB      2.50
## s13     BB      2.50
## s14     BB      0.00
## s15     BB      1.07
## s16     BB      2.50
## s17     AA      5.00
## s18     BB     35.00
## s19     BB      7.50
## s20     BB     15.00
## s21     AA      3.75
## s22     AA      3.75
## refE    QC        NA</code></pre>
<ul>
<li>run <code>run_RefCheck</code></li>
</ul>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="quality-control.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">run_RefCheck</span>(</span>
<span id="cb17-2"><a href="quality-control.html#cb17-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">ps=</span>metaphlan2_ps_species,</span>
<span id="cb17-3"><a href="quality-control.html#cb17-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">BRS_ID=</span><span class="st">&quot;refE&quot;</span>,</span>
<span id="cb17-4"><a href="quality-control.html#cb17-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">Reference=</span><span class="cn">NULL</span>,</span>
<span id="cb17-5"><a href="quality-control.html#cb17-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">Ref_type=</span><span class="st">&quot;MGS&quot;</span>,</span>
<span id="cb17-6"><a href="quality-control.html#cb17-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">Save=</span><span class="cn">NULL</span>)</span>
<span id="cb17-7"><a href="quality-control.html#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="do">##                                         7682    7683    7684    7685     7842     7843     7844</span></span>
<span id="cb17-8"><a href="quality-control.html#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="do">## Bifidobacterium_longum              10.31563 9.25812 9.69184 7.76031 11.03311 11.61484 12.29030</span></span>
<span id="cb17-9"><a href="quality-control.html#cb17-9" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_uniformis                2.24195 2.24035 1.92015 2.18435  2.43230  2.38180  2.13830</span></span>
<span id="cb17-10"><a href="quality-control.html#cb17-10" aria-hidden="true" tabindex="-1"></a><span class="do">## Faecalibacterium_prausnitzii         0.65615 0.60153 0.60112 0.62079  0.54147  0.55383  0.58806</span></span>
<span id="cb17-11"><a href="quality-control.html#cb17-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Bifidobacterium_adolescentis         7.05426 6.28460 6.57297 6.25448  4.69357  4.80628  4.94943</span></span>
<span id="cb17-12"><a href="quality-control.html#cb17-12" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_thetaiotaomicron         3.25076 3.31897 3.22418 3.43809  3.35611  3.38323  3.29098</span></span>
<span id="cb17-13"><a href="quality-control.html#cb17-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Collinsella_aerofaciens              0.56220 0.53249 0.60476 0.47934  0.65513  0.68833  0.76063</span></span>
<span id="cb17-14"><a href="quality-control.html#cb17-14" aria-hidden="true" tabindex="-1"></a><span class="do">## Coprococcus_comes                    2.28581 2.42978 2.25227 2.82904  1.21527  1.16485  1.07160</span></span>
<span id="cb17-15"><a href="quality-control.html#cb17-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Dorea_formicigenerans                4.83509 5.02149 5.18268 5.56891  3.34720  3.09877  3.08677</span></span>
<span id="cb17-16"><a href="quality-control.html#cb17-16" aria-hidden="true" tabindex="-1"></a><span class="do">## Streptococcus_salivarius             3.54266 3.74119 3.62036 4.01546  2.90216  2.70193  2.61760</span></span>
<span id="cb17-17"><a href="quality-control.html#cb17-17" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_xylanisolvens            1.55648 1.84824 1.91166 1.85273  1.75220  1.74002  1.69811</span></span>
<span id="cb17-18"><a href="quality-control.html#cb17-18" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_ovatus                   3.08489 3.27226 3.10904 3.24565  3.51376  3.50063  3.37872</span></span>
<span id="cb17-19"><a href="quality-control.html#cb17-19" aria-hidden="true" tabindex="-1"></a><span class="do">## Roseburia_hominis                    0.04383 0.04183 0.04107 0.02304  0.03853  0.03464  0.03532</span></span>
<span id="cb17-20"><a href="quality-control.html#cb17-20" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_vulgatus                 3.06713 3.20369 3.14979 3.15352  3.24822  3.09280  3.13038</span></span>
<span id="cb17-21"><a href="quality-control.html#cb17-21" aria-hidden="true" tabindex="-1"></a><span class="do">## Prevotella_copri                     2.03128 1.99619 1.92504 2.15422  1.57638  1.60584  1.57913</span></span>
<span id="cb17-22"><a href="quality-control.html#cb17-22" aria-hidden="true" tabindex="-1"></a><span class="do">## Bifidobacterium_pseudocatenulatum    6.89310 6.36177 7.39605 5.76804  6.19464  6.47094  7.41615</span></span>
<span id="cb17-23"><a href="quality-control.html#cb17-23" aria-hidden="true" tabindex="-1"></a><span class="do">## Lachnospiraceae_bacterium_5_1_63FAA  0.22002 0.26062 0.28058 0.33329  0.07093  0.06778  0.05229</span></span>
<span id="cb17-24"><a href="quality-control.html#cb17-24" aria-hidden="true" tabindex="-1"></a><span class="do">## Impurity_level                       7.33876 7.34799 7.14891 6.47250  8.88190  9.72198  9.72162</span></span>
<span id="cb17-25"><a href="quality-control.html#cb17-25" aria-hidden="true" tabindex="-1"></a><span class="do">##                                         7845     refE        mean</span></span>
<span id="cb17-26"><a href="quality-control.html#cb17-26" aria-hidden="true" tabindex="-1"></a><span class="do">## Bifidobacterium_longum              11.69019  0.01646  9.29675556</span></span>
<span id="cb17-27"><a href="quality-control.html#cb17-27" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_uniformis                2.41437  0.21061  2.01824222</span></span>
<span id="cb17-28"><a href="quality-control.html#cb17-28" aria-hidden="true" tabindex="-1"></a><span class="do">## Faecalibacterium_prausnitzii         0.54655  1.61939  0.70321000</span></span>
<span id="cb17-29"><a href="quality-control.html#cb17-29" aria-hidden="true" tabindex="-1"></a><span class="do">## Bifidobacterium_adolescentis         4.84278  0.04649  5.05609556</span></span>
<span id="cb17-30"><a href="quality-control.html#cb17-30" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_thetaiotaomicron         3.30355  1.47422  3.11556556</span></span>
<span id="cb17-31"><a href="quality-control.html#cb17-31" aria-hidden="true" tabindex="-1"></a><span class="do">## Collinsella_aerofaciens              0.66896  0.07251  0.55826111</span></span>
<span id="cb17-32"><a href="quality-control.html#cb17-32" aria-hidden="true" tabindex="-1"></a><span class="do">## Coprococcus_comes                    1.07834  0.01444  1.59348889</span></span>
<span id="cb17-33"><a href="quality-control.html#cb17-33" aria-hidden="true" tabindex="-1"></a><span class="do">## Dorea_formicigenerans                2.70732  0.02775  3.65288667</span></span>
<span id="cb17-34"><a href="quality-control.html#cb17-34" aria-hidden="true" tabindex="-1"></a><span class="do">## Streptococcus_salivarius             2.57348  0.02846  2.86036667</span></span>
<span id="cb17-35"><a href="quality-control.html#cb17-35" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_xylanisolvens            1.67676  0.32466  1.59565111</span></span>
<span id="cb17-36"><a href="quality-control.html#cb17-36" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_ovatus                   3.42300  0.25782  2.97619667</span></span>
<span id="cb17-37"><a href="quality-control.html#cb17-37" aria-hidden="true" tabindex="-1"></a><span class="do">## Roseburia_hominis                    0.03597  0.01307  0.03414444</span></span>
<span id="cb17-38"><a href="quality-control.html#cb17-38" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_vulgatus                 3.06113  2.14684  3.02816667</span></span>
<span id="cb17-39"><a href="quality-control.html#cb17-39" aria-hidden="true" tabindex="-1"></a><span class="do">## Prevotella_copri                     1.59224 60.84109  8.36682333</span></span>
<span id="cb17-40"><a href="quality-control.html#cb17-40" aria-hidden="true" tabindex="-1"></a><span class="do">## Bifidobacterium_pseudocatenulatum    6.27837  0.15023  5.88103222</span></span>
<span id="cb17-41"><a href="quality-control.html#cb17-41" aria-hidden="true" tabindex="-1"></a><span class="do">## Lachnospiraceae_bacterium_5_1_63FAA  0.04781  0.06825  0.15573000</span></span>
<span id="cb17-42"><a href="quality-control.html#cb17-42" aria-hidden="true" tabindex="-1"></a><span class="do">## Impurity_level                      10.57600 32.69000 11.09996222</span></span>
<span id="cb17-43"><a href="quality-control.html#cb17-43" aria-hidden="true" tabindex="-1"></a><span class="do">##                                                                                Evaluation</span></span>
<span id="cb17-44"><a href="quality-control.html#cb17-44" aria-hidden="true" tabindex="-1"></a><span class="do">## Bifidobacterium_longum              refE didn&#39;t pass the threshold (2022-05-27 11:45:59).</span></span>
<span id="cb17-45"><a href="quality-control.html#cb17-45" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_uniformis               refE didn&#39;t pass the threshold (2022-05-27 11:45:59).</span></span>
<span id="cb17-46"><a href="quality-control.html#cb17-46" aria-hidden="true" tabindex="-1"></a><span class="do">## Faecalibacterium_prausnitzii        refE didn&#39;t pass the threshold (2022-05-27 11:45:59).</span></span>
<span id="cb17-47"><a href="quality-control.html#cb17-47" aria-hidden="true" tabindex="-1"></a><span class="do">## Bifidobacterium_adolescentis        refE didn&#39;t pass the threshold (2022-05-27 11:45:59).</span></span>
<span id="cb17-48"><a href="quality-control.html#cb17-48" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_thetaiotaomicron        refE didn&#39;t pass the threshold (2022-05-27 11:45:59).</span></span>
<span id="cb17-49"><a href="quality-control.html#cb17-49" aria-hidden="true" tabindex="-1"></a><span class="do">## Collinsella_aerofaciens             refE didn&#39;t pass the threshold (2022-05-27 11:45:59).</span></span>
<span id="cb17-50"><a href="quality-control.html#cb17-50" aria-hidden="true" tabindex="-1"></a><span class="do">## Coprococcus_comes                   refE didn&#39;t pass the threshold (2022-05-27 11:45:59).</span></span>
<span id="cb17-51"><a href="quality-control.html#cb17-51" aria-hidden="true" tabindex="-1"></a><span class="do">## Dorea_formicigenerans               refE didn&#39;t pass the threshold (2022-05-27 11:45:59).</span></span>
<span id="cb17-52"><a href="quality-control.html#cb17-52" aria-hidden="true" tabindex="-1"></a><span class="do">## Streptococcus_salivarius            refE didn&#39;t pass the threshold (2022-05-27 11:45:59).</span></span>
<span id="cb17-53"><a href="quality-control.html#cb17-53" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_xylanisolvens           refE didn&#39;t pass the threshold (2022-05-27 11:45:59).</span></span>
<span id="cb17-54"><a href="quality-control.html#cb17-54" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_ovatus                  refE didn&#39;t pass the threshold (2022-05-27 11:45:59).</span></span>
<span id="cb17-55"><a href="quality-control.html#cb17-55" aria-hidden="true" tabindex="-1"></a><span class="do">## Roseburia_hominis                   refE didn&#39;t pass the threshold (2022-05-27 11:45:59).</span></span>
<span id="cb17-56"><a href="quality-control.html#cb17-56" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_vulgatus                refE didn&#39;t pass the threshold (2022-05-27 11:45:59).</span></span>
<span id="cb17-57"><a href="quality-control.html#cb17-57" aria-hidden="true" tabindex="-1"></a><span class="do">## Prevotella_copri                    refE didn&#39;t pass the threshold (2022-05-27 11:45:59).</span></span>
<span id="cb17-58"><a href="quality-control.html#cb17-58" aria-hidden="true" tabindex="-1"></a><span class="do">## Bifidobacterium_pseudocatenulatum   refE didn&#39;t pass the threshold (2022-05-27 11:45:59).</span></span>
<span id="cb17-59"><a href="quality-control.html#cb17-59" aria-hidden="true" tabindex="-1"></a><span class="do">## Lachnospiraceae_bacterium_5_1_63FAA refE didn&#39;t pass the threshold (2022-05-27 11:45:59).</span></span>
<span id="cb17-60"><a href="quality-control.html#cb17-60" aria-hidden="true" tabindex="-1"></a><span class="do">## Impurity_level                      refE didn&#39;t pass the threshold (2022-05-27 11:45:59).</span></span></code></pre></div>
</div>
<div id="spike-in-samples-brs-remove" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Spike-in sample’s (BRS) remove<a href="quality-control.html#spike-in-samples-brs-remove" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="quality-control.html#cb18-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb18-2"><a href="quality-control.html#cb18-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ps =</span> metaphlan2_ps,</span>
<span id="cb18-3"><a href="quality-control.html#cb18-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb18-4"><a href="quality-control.html#cb18-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>)</span>
<span id="cb18-5"><a href="quality-control.html#cb18-5" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_remove_BRS</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 326 taxa and 22 samples ]
## sample_data() Sample Data:       [ 22 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 326 taxa by 7 taxonomic ranks ]</code></pre>
</div>
<div id="systematic-information-1" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Systematic Information<a href="quality-control.html#systematic-information-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="quality-control.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.1.2 (2021-11-01)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Monterey 12.2.1
## 
## Matrix products: default
## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] SummarizedExperiment_1.24.0 Biobase_2.54.0              GenomicRanges_1.46.1       
##  [4] GenomeInfoDb_1.30.1         IRanges_2.28.0              S4Vectors_0.32.3           
##  [7] BiocGenerics_0.40.0         MatrixGenerics_1.6.0        matrixStats_0.61.0         
## [10] ggpubr_0.4.0                ggplot2_3.3.5               phyloseq_1.38.0            
## [13] tibble_3.1.6                dplyr_1.0.8                 XMAS_2.1.2                 
## 
## loaded via a namespace (and not attached):
##   [1] backports_1.4.1        Hmisc_4.6-0            plyr_1.8.6             igraph_1.2.11         
##   [5] splines_4.1.2          BiocParallel_1.28.3    TH.data_1.1-0          digest_0.6.29         
##   [9] foreach_1.5.2          htmltools_0.5.2        fansi_1.0.2            checkmate_2.0.0       
##  [13] magrittr_2.0.2         memoise_2.0.1          cluster_2.1.2          limma_3.50.1          
##  [17] Biostrings_2.62.0      annotate_1.72.0        sandwich_3.0-1         metagenomeSeq_1.36.0  
##  [21] jpeg_0.1-9             colorspace_2.0-3       ggrepel_0.9.1          blob_1.2.2            
##  [25] xfun_0.30              crayon_1.5.0           RCurl_1.98-1.6         jsonlite_1.8.0        
##  [29] libcoin_1.0-9          genefilter_1.76.0      survival_3.3-1         zoo_1.8-9             
##  [33] iterators_1.0.14       ape_5.6-2              glue_1.6.2             gtable_0.3.0          
##  [37] zlibbioc_1.40.0        XVector_0.34.0         DelayedArray_0.20.0    car_3.0-12            
##  [41] RcppZiggurat_0.1.6     Rhdf5lib_1.16.0        shape_1.4.6            pscl_1.5.5            
##  [45] abind_1.4-5            scales_1.1.1           pheatmap_1.0.12        mvtnorm_1.1-3         
##  [49] DBI_1.1.2              edgeR_3.36.0           rstatix_0.7.0          Rcpp_1.0.8.2          
##  [53] htmlTable_2.4.0        xtable_1.8-4           foreign_0.8-82         bit_4.0.4             
##  [57] Formula_1.2-4          DT_0.21                truncnorm_1.0-8        glmnet_4.1-3          
##  [61] htmlwidgets_1.5.4      httr_1.4.2             DiagrammeR_1.0.9       gplots_3.1.1          
##  [65] RColorBrewer_1.1-2     modeltools_0.2-23      ellipsis_0.3.2         NADA_1.6-1.1          
##  [69] pkgconfig_2.0.3        XML_3.99-0.9           farver_2.1.0           nnet_7.3-17           
##  [73] sass_0.4.0             locfit_1.5-9.5         utf8_1.2.2             tidyselect_1.1.2      
##  [77] labeling_0.4.2         rlang_1.0.2            reshape2_1.4.4         AnnotationDbi_1.56.2  
##  [81] munsell_0.5.0          tools_4.1.2            visNetwork_2.1.0       cachem_1.0.6          
##  [85] cli_3.2.0              generics_0.1.2         RSQLite_2.2.10         ade4_1.7-18           
##  [89] broom_0.7.12           evaluate_0.15          biomformat_1.22.0      stringr_1.4.0         
##  [93] fastmap_1.1.0          yaml_2.3.5             knitr_1.37             bit64_4.0.5           
##  [97] caTools_1.18.2         purrr_0.3.4            KEGGREST_1.34.0        coin_1.4-2            
## [101] nlme_3.1-155           mbzinb_0.2             compiler_4.1.2         rstudioapi_0.13       
## [105] png_0.1-7              ggsignif_0.6.3         zCompositions_1.4.0    geneplotter_1.72.0    
## [109] bslib_0.3.1            stringi_1.7.6          highr_0.9              lattice_0.20-45       
## [113] Matrix_1.4-0           vegan_2.5-7            permute_0.9-7          multtest_2.50.0       
## [117] vctrs_0.3.8            pillar_1.7.0           lifecycle_1.0.1        rhdf5filters_1.6.0    
## [121] jquerylib_0.1.4        cowplot_1.1.1          data.table_1.14.2      bitops_1.0-7          
## [125] latticeExtra_0.6-29    R6_2.5.1               bookdown_0.24          gridExtra_2.3         
## [129] KernSmooth_2.23-20     codetools_0.2-18       MASS_7.3-55            gtools_3.9.2          
## [133] assertthat_0.2.1       Wrench_1.12.0          rhdf5_2.38.1           DESeq2_1.34.0         
## [137] withr_2.5.0            multcomp_1.4-18        GenomeInfoDbData_1.2.7 mgcv_1.8-39           
## [141] parallel_4.1.2         rpart_4.1.16           grid_4.1.2             tidyr_1.2.0           
## [145] Rfast_2.0.6            rmarkdown_2.13         carData_3.0-5          ALDEx2_1.26.0         
## [149] base64enc_0.1-3</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="convert-inputdata-into-phyloseq-object.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="data-processing.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/MGS_DataAnalysis_template/blob/main/02-Quality-evaluation.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["MGS microbiota data analysis based on XMAS package.pdf", "MGS microbiota data analysis based on XMAS package.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
