<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 Test Example | MGS microbiota data analysis based on XMAS package</title>
  <meta name="description" content="Chapter 10 Test Example | MGS microbiota data analysis based on XMAS package" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 Test Example | MGS microbiota data analysis based on XMAS package" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Chapter 10 Test Example | MGS microbiota data analysis based on XMAS package" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 Test Example | MGS microbiota data analysis based on XMAS package" />
  
  <meta name="twitter:description" content="Chapter 10 Test Example | MGS microbiota data analysis based on XMAS package" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2023-10-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="network-analysis.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.28/datatables.js"></script>
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">MGS microbiota data analysis based on XMAS package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-tutorial"><i class="fa fa-check"></i><b>1.1</b> Preparing for the tutorial</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#workflow-description"><i class="fa fa-check"></i><b>1.2</b> Workflow description</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html"><i class="fa fa-check"></i><b>2</b> Convert inputdata into phyloseq object</a>
<ul>
<li class="chapter" data-level="2.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metaphlan23"><i class="fa fa-check"></i><b>2.1</b> Metaphlan2/3</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#merging-output-of-metaphlan"><i class="fa fa-check"></i><b>2.1.1</b> Merging output of metaphlan</a></li>
<li class="chapter" data-level="2.1.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results"><i class="fa fa-check"></i><b>2.1.2</b> Importing results</a></li>
<li class="chapter" data-level="2.1.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table"><i class="fa fa-check"></i><b>2.1.3</b> taxa table</a></li>
<li class="chapter" data-level="2.1.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table"><i class="fa fa-check"></i><b>2.1.4</b> otu table</a></li>
<li class="chapter" data-level="2.1.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table"><i class="fa fa-check"></i><b>2.1.5</b> metadata table</a></li>
<li class="chapter" data-level="2.1.6" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object"><i class="fa fa-check"></i><b>2.1.6</b> phyloseq object</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#transform-the-abundance-of-taxa-whose-relative-abundance-under-limit-of-detection-lod-into-zeros"><i class="fa fa-check"></i><b>2.2</b> Transform the abundance of taxa whose relative abundance under Limit Of Detection (LOD) into Zeros</a></li>
<li class="chapter" data-level="2.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#systematic-information"><i class="fa fa-check"></i><b>2.3</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>3</b> Quality Control</a>
<ul>
<li class="chapter" data-level="3.1" data-path="quality-control.html"><a href="quality-control.html#spike-in-sample-brs"><i class="fa fa-check"></i><b>3.1</b> spike-in sample (BRS)</a></li>
<li class="chapter" data-level="3.2" data-path="quality-control.html"><a href="quality-control.html#spike-in-samples-brs-remove"><i class="fa fa-check"></i><b>3.2</b> Spike-in sample’s (BRS) remove</a></li>
<li class="chapter" data-level="3.3" data-path="quality-control.html"><a href="quality-control.html#systematic-information-1"><i class="fa fa-check"></i><b>3.3</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data-processing.html"><a href="data-processing.html"><i class="fa fa-check"></i><b>4</b> Data processing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-processing.html"><a href="data-processing.html#extracting-specific-taxonomic-level"><i class="fa fa-check"></i><b>4.1</b> Extracting specific taxonomic level</a></li>
<li class="chapter" data-level="4.2" data-path="data-processing.html"><a href="data-processing.html#systematic-information-2"><i class="fa fa-check"></i><b>4.2</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="diversity-analysis.html"><a href="diversity-analysis.html"><i class="fa fa-check"></i><b>5</b> Diversity analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="diversity-analysis.html"><a href="diversity-analysis.html#alpha-diveristy"><i class="fa fa-check"></i><b>5.1</b> Alpha diveristy</a></li>
<li class="chapter" data-level="5.2" data-path="diversity-analysis.html"><a href="diversity-analysis.html#beta-diversity"><i class="fa fa-check"></i><b>5.2</b> Beta diversity</a></li>
<li class="chapter" data-level="5.3" data-path="diversity-analysis.html"><a href="diversity-analysis.html#systematic-information-3"><i class="fa fa-check"></i><b>5.3</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html"><i class="fa fa-check"></i><b>6</b> PERMANOVA + beta dispersion + Ordination</a>
<ul>
<li class="chapter" data-level="6.1" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#permanova"><i class="fa fa-check"></i><b>6.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="6.2" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#beta-dispersion"><i class="fa fa-check"></i><b>6.2</b> beta dispersion</a></li>
<li class="chapter" data-level="6.3" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#ordination"><i class="fa fa-check"></i><b>6.3</b> Ordination</a></li>
<li class="chapter" data-level="6.4" data-path="permanova-beta-dispersion-ordination.html"><a href="permanova-beta-dispersion-ordination.html#systematic-information-4"><i class="fa fa-check"></i><b>6.4</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="microbial-composition.html"><a href="microbial-composition.html"><i class="fa fa-check"></i><b>7</b> Microbial composition</a>
<ul>
<li class="chapter" data-level="7.1" data-path="microbial-composition.html"><a href="microbial-composition.html#stacked-barplot"><i class="fa fa-check"></i><b>7.1</b> Stacked barplot</a></li>
<li class="chapter" data-level="7.2" data-path="microbial-composition.html"><a href="microbial-composition.html#core-microbiota"><i class="fa fa-check"></i><b>7.2</b> Core microbiota</a></li>
<li class="chapter" data-level="7.3" data-path="microbial-composition.html"><a href="microbial-composition.html#systematic-information-5"><i class="fa fa-check"></i><b>7.3</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-analysis.html"><a href="differential-analysis.html"><i class="fa fa-check"></i><b>8</b> Differential Analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="differential-analysis.html"><a href="differential-analysis.html#filtering-trimming"><i class="fa fa-check"></i><b>8.1</b> Filtering &amp; Trimming</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="differential-analysis.html"><a href="differential-analysis.html#filtering-the-low-relative-abundance-or-unclassified-taxa-by-the-threshold-total-counts-1e-4"><i class="fa fa-check"></i><b>8.1.1</b> Filtering the low relative abundance or unclassified taxa by the threshold (total counts &lt; 1e-4)</a></li>
<li class="chapter" data-level="8.1.2" data-path="differential-analysis.html"><a href="differential-analysis.html#trimming-the-taxa-with-low-occurrence-less-than-threshold"><i class="fa fa-check"></i><b>8.1.2</b> Trimming the taxa with low occurrence less than threshold</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse"><i class="fa fa-check"></i><b>8.2</b> Liner discriminant analysis (LDA) effect size (LEfSe)</a></li>
<li class="chapter" data-level="8.3" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-test"><i class="fa fa-check"></i><b>8.3</b> Wilcoxon Rank-Sum test</a></li>
<li class="chapter" data-level="8.4" data-path="differential-analysis.html"><a href="differential-analysis.html#dominant-taxa"><i class="fa fa-check"></i><b>8.4</b> Dominant taxa</a></li>
<li class="chapter" data-level="8.5" data-path="differential-analysis.html"><a href="differential-analysis.html#multiple-differential-analysis-by-one-function"><i class="fa fa-check"></i><b>8.5</b> Multiple differential analysis by one function</a></li>
<li class="chapter" data-level="8.6" data-path="differential-analysis.html"><a href="differential-analysis.html#systematic-information-6"><i class="fa fa-check"></i><b>8.6</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="network-analysis.html"><a href="network-analysis.html"><i class="fa fa-check"></i><b>9</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="network-analysis.html"><a href="network-analysis.html#importing-data"><i class="fa fa-check"></i><b>9.1</b> Importing data</a></li>
<li class="chapter" data-level="9.2" data-path="network-analysis.html"><a href="network-analysis.html#single-network-with-pearson-correlation-as-association-measure"><i class="fa fa-check"></i><b>9.2</b> Single network with Pearson correlation as association measure</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="network-analysis.html"><a href="network-analysis.html#building-network-model"><i class="fa fa-check"></i><b>9.2.1</b> Building network model</a></li>
<li class="chapter" data-level="9.2.2" data-path="network-analysis.html"><a href="network-analysis.html#analyzing-the-constructed-network"><i class="fa fa-check"></i><b>9.2.2</b> Analyzing the constructed network</a></li>
<li class="chapter" data-level="9.2.3" data-path="network-analysis.html"><a href="network-analysis.html#visualizing-the-network"><i class="fa fa-check"></i><b>9.2.3</b> Visualizing the network</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="network-analysis.html"><a href="network-analysis.html#dissimilarity-based-networks"><i class="fa fa-check"></i><b>9.3</b> Dissimilarity-based Networks</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="network-analysis.html"><a href="network-analysis.html#building-network-module"><i class="fa fa-check"></i><b>9.3.1</b> Building network module</a></li>
<li class="chapter" data-level="9.3.2" data-path="network-analysis.html"><a href="network-analysis.html#analyzing-the-constructed-network-1"><i class="fa fa-check"></i><b>9.3.2</b> Analyzing the constructed network</a></li>
<li class="chapter" data-level="9.3.3" data-path="network-analysis.html"><a href="network-analysis.html#visualizing-the-network-1"><i class="fa fa-check"></i><b>9.3.3</b> Visualizing the network</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="network-analysis.html"><a href="network-analysis.html#systematic-information-7"><i class="fa fa-check"></i><b>9.4</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="test-example.html"><a href="test-example.html"><i class="fa fa-check"></i><b>10</b> Test Example</a>
<ul>
<li class="chapter" data-level="10.1" data-path="test-example.html"><a href="test-example.html#loading-packages"><i class="fa fa-check"></i><b>10.1</b> Loading packages</a></li>
<li class="chapter" data-level="10.2" data-path="test-example.html"><a href="test-example.html#loading-data"><i class="fa fa-check"></i><b>10.2</b> Loading data</a></li>
<li class="chapter" data-level="10.3" data-path="test-example.html"><a href="test-example.html#step1-convert-inputs-into-phyloseq-data"><i class="fa fa-check"></i><b>10.3</b> Step1: Convert inputs into phyloseq data</a></li>
<li class="chapter" data-level="10.4" data-path="test-example.html"><a href="test-example.html#step2-transform-limit-of-detection-lod-into-zeros"><i class="fa fa-check"></i><b>10.4</b> Step2: Transform limit of detection (LOD) into Zeros</a></li>
<li class="chapter" data-level="10.5" data-path="test-example.html"><a href="test-example.html#step3-brs-checking"><i class="fa fa-check"></i><b>10.5</b> Step3: BRS checking</a></li>
<li class="chapter" data-level="10.6" data-path="test-example.html"><a href="test-example.html#step4-extracting-specific-taxonomic-level"><i class="fa fa-check"></i><b>10.6</b> Step4: Extracting specific taxonomic level</a></li>
<li class="chapter" data-level="10.7" data-path="test-example.html"><a href="test-example.html#step5-globalview"><i class="fa fa-check"></i><b>10.7</b> Step5: GlobalView</a></li>
<li class="chapter" data-level="10.8" data-path="test-example.html"><a href="test-example.html#step6-differential-analysis"><i class="fa fa-check"></i><b>10.8</b> Step6: Differential Analysis</a></li>
<li class="chapter" data-level="10.9" data-path="test-example.html"><a href="test-example.html#systematic-information-8"><i class="fa fa-check"></i><b>10.9</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">MGS microbiota data analysis based on XMAS package</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="test-example" class="section level1 hasAnchor" number="10">
<h1><span class="header-section-number">Chapter 10</span> Test Example<a href="test-example.html#test-example" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="loading-packages" class="section level2 hasAnchor" number="10.1">
<h2><span class="header-section-number">10.1</span> Loading packages<a href="test-example.html#loading-packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="test-example.html#cb109-1" tabindex="-1"></a><span class="fu">library</span>(XMAS2)</span>
<span id="cb109-2"><a href="test-example.html#cb109-2" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb109-3"><a href="test-example.html#cb109-3" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb109-4"><a href="test-example.html#cb109-4" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb109-5"><a href="test-example.html#cb109-5" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb109-6"><a href="test-example.html#cb109-6" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb109-7"><a href="test-example.html#cb109-7" tabindex="-1"></a><span class="fu">library</span>(readxl)</span></code></pre></div>
</div>
<div id="loading-data" class="section level2 hasAnchor" number="10.2">
<h2><span class="header-section-number">10.2</span> Loading data<a href="test-example.html#loading-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="test-example.html#cb110-1" tabindex="-1"></a>metaphlan2_res <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;DataSet/RawData/merged_metaphlan2.tsv&quot;</span>,</span>
<span id="cb110-2"><a href="test-example.html#cb110-2" tabindex="-1"></a>                             <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb110-3"><a href="test-example.html#cb110-3" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;ID&quot;</span>)</span>
<span id="cb110-4"><a href="test-example.html#cb110-4" tabindex="-1"></a></span>
<span id="cb110-5"><a href="test-example.html#cb110-5" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">&quot;DataSet/RawData/诺禾宏基因组678月-ZH.xlsx&quot;</span>, <span class="at">sheet =</span> <span class="dv">3</span>)</span></code></pre></div>
</div>
<div id="step1-convert-inputs-into-phyloseq-data" class="section level2 hasAnchor" number="10.3">
<h2><span class="header-section-number">10.3</span> Step1: Convert inputs into phyloseq data<a href="test-example.html#step1-convert-inputs-into-phyloseq-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="test-example.html#cb111-1" tabindex="-1"></a>metaphlan2_res_list <span class="ot">&lt;-</span> <span class="fu">import_metaphlan_taxa</span>(<span class="at">data_metaphlan2 =</span> metaphlan2_res, </span>
<span id="cb111-2"><a href="test-example.html#cb111-2" tabindex="-1"></a>                                             <span class="at">taxa_level =</span> <span class="st">&quot;Species&quot;</span>)</span>
<span id="cb111-3"><a href="test-example.html#cb111-3" tabindex="-1"></a>tax_tab <span class="ot">&lt;-</span> metaphlan2_res_list<span class="sc">$</span>tax_tab</span>
<span id="cb111-4"><a href="test-example.html#cb111-4" tabindex="-1"></a></span>
<span id="cb111-5"><a href="test-example.html#cb111-5" tabindex="-1"></a>otu_tab <span class="ot">&lt;-</span> metaphlan2_res_list<span class="sc">$</span>abu_tab</span>
<span id="cb111-6"><a href="test-example.html#cb111-6" tabindex="-1"></a><span class="fu">colnames</span>(otu_tab) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;X&quot;</span>, <span class="st">&quot;S_&quot;</span>, <span class="fu">colnames</span>(otu_tab))</span>
<span id="cb111-7"><a href="test-example.html#cb111-7" tabindex="-1"></a></span>
<span id="cb111-8"><a href="test-example.html#cb111-8" tabindex="-1"></a>sam_tab <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb111-9"><a href="test-example.html#cb111-9" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Group=</span><span class="fu">ifelse</span>(SampleType <span class="sc">==</span> <span class="st">&quot;粪便&quot;</span>, <span class="st">&quot;Stool&quot;</span>, </span>
<span id="cb111-10"><a href="test-example.html#cb111-10" tabindex="-1"></a>                             <span class="fu">ifelse</span>(SampleType <span class="sc">==</span> <span class="st">&quot;QC&quot;</span>, <span class="st">&quot;QC&quot;</span>, <span class="st">&quot;Product&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb111-11"><a href="test-example.html#cb111-11" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(SampleTubeID, Group, <span class="fu">everything</span>())</span>
<span id="cb111-12"><a href="test-example.html#cb111-12" tabindex="-1"></a><span class="fu">rownames</span>(sam_tab) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;S_&quot;</span>, sam_tab<span class="sc">$</span>SeqID_MGS)</span>
<span id="cb111-13"><a href="test-example.html#cb111-13" tabindex="-1"></a></span>
<span id="cb111-14"><a href="test-example.html#cb111-14" tabindex="-1"></a>overlap_samples <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">intersect</span>(<span class="fu">rownames</span>(sam_tab), <span class="fu">colnames</span>(otu_tab))</span>
<span id="cb111-15"><a href="test-example.html#cb111-15" tabindex="-1"></a></span>
<span id="cb111-16"><a href="test-example.html#cb111-16" tabindex="-1"></a>otu_tab_cln <span class="ot">&lt;-</span> otu_tab[, <span class="fu">match</span>(overlap_samples, <span class="fu">colnames</span>(otu_tab))]</span>
<span id="cb111-17"><a href="test-example.html#cb111-17" tabindex="-1"></a>sam_tab_cln <span class="ot">&lt;-</span> sam_tab[<span class="fu">match</span>(overlap_samples, <span class="fu">rownames</span>(sam_tab)), ]</span>
<span id="cb111-18"><a href="test-example.html#cb111-18" tabindex="-1"></a><span class="fu">rownames</span>(sam_tab_cln) <span class="ot">&lt;-</span> overlap_samples</span>
<span id="cb111-19"><a href="test-example.html#cb111-19" tabindex="-1"></a></span>
<span id="cb111-20"><a href="test-example.html#cb111-20" tabindex="-1"></a>metaphlan2_ps <span class="ot">&lt;-</span> <span class="fu">get_metaphlan_phyloseq</span>(</span>
<span id="cb111-21"><a href="test-example.html#cb111-21" tabindex="-1"></a>                    <span class="at">otu_tab =</span> otu_tab_cln, </span>
<span id="cb111-22"><a href="test-example.html#cb111-22" tabindex="-1"></a>                    <span class="at">sam_tab =</span> sam_tab_cln,</span>
<span id="cb111-23"><a href="test-example.html#cb111-23" tabindex="-1"></a>                    <span class="at">tax_tab =</span> tax_tab)</span>
<span id="cb111-24"><a href="test-example.html#cb111-24" tabindex="-1"></a>metaphlan2_ps</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 315 taxa and 145 samples ]
## sample_data() Sample Data:       [ 145 samples by 12 sample variables ]
## tax_table()   Taxonomy Table:    [ 315 taxa by 7 taxonomic ranks ]</code></pre>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="test-example.html#cb113-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="st">&quot;DataSet/Step1/&quot;</span>)) {</span>
<span id="cb113-2"><a href="test-example.html#cb113-2" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="st">&quot;DataSet/Step1/&quot;</span>)</span>
<span id="cb113-3"><a href="test-example.html#cb113-3" tabindex="-1"></a>}</span>
<span id="cb113-4"><a href="test-example.html#cb113-4" tabindex="-1"></a><span class="fu">saveRDS</span>(metaphlan2_ps, <span class="st">&quot;DataSet/Step1/Donor_MGS_phyloseq.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="step2-transform-limit-of-detection-lod-into-zeros" class="section level2 hasAnchor" number="10.4">
<h2><span class="header-section-number">10.4</span> Step2: Transform limit of detection (LOD) into Zeros<a href="test-example.html#step2-transform-limit-of-detection-lod-into-zeros" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="test-example.html#cb114-1" tabindex="-1"></a><span class="co"># species</span></span>
<span id="cb114-2"><a href="test-example.html#cb114-2" tabindex="-1"></a>metaphlan2_ps_LOD_species <span class="ot">&lt;-</span> <span class="fu">aggregate_LOD_taxa</span>(<span class="at">ps =</span> metaphlan2_ps, </span>
<span id="cb114-3"><a href="test-example.html#cb114-3" tabindex="-1"></a>                                                <span class="at">taxa_level =</span> <span class="st">&quot;Species&quot;</span>, </span>
<span id="cb114-4"><a href="test-example.html#cb114-4" tabindex="-1"></a>                                                <span class="at">cutoff =</span> <span class="fl">1e-04</span>)</span>
<span id="cb114-5"><a href="test-example.html#cb114-5" tabindex="-1"></a></span>
<span id="cb114-6"><a href="test-example.html#cb114-6" tabindex="-1"></a><span class="co"># genus</span></span>
<span id="cb114-7"><a href="test-example.html#cb114-7" tabindex="-1"></a>metaphlan2_ps_LOD_genus <span class="ot">&lt;-</span> <span class="fu">aggregate_LOD_taxa</span>(<span class="at">ps =</span> metaphlan2_ps, </span>
<span id="cb114-8"><a href="test-example.html#cb114-8" tabindex="-1"></a>                                              <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>, </span>
<span id="cb114-9"><a href="test-example.html#cb114-9" tabindex="-1"></a>                                              <span class="at">cutoff =</span> <span class="fl">1e-04</span>)</span>
<span id="cb114-10"><a href="test-example.html#cb114-10" tabindex="-1"></a></span>
<span id="cb114-11"><a href="test-example.html#cb114-11" tabindex="-1"></a><span class="co"># order</span></span>
<span id="cb114-12"><a href="test-example.html#cb114-12" tabindex="-1"></a>metaphlan2_ps_LOD_order <span class="ot">&lt;-</span> <span class="fu">aggregate_LOD_taxa</span>(<span class="at">ps =</span> metaphlan2_ps, </span>
<span id="cb114-13"><a href="test-example.html#cb114-13" tabindex="-1"></a>                                              <span class="at">taxa_level =</span> <span class="st">&quot;Order&quot;</span>, </span>
<span id="cb114-14"><a href="test-example.html#cb114-14" tabindex="-1"></a>                                              <span class="at">cutoff =</span> <span class="fl">1e-04</span>)</span>
<span id="cb114-15"><a href="test-example.html#cb114-15" tabindex="-1"></a></span>
<span id="cb114-16"><a href="test-example.html#cb114-16" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="st">&quot;DataSet/Step2/&quot;</span>)) {</span>
<span id="cb114-17"><a href="test-example.html#cb114-17" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="st">&quot;DataSet/Step2/&quot;</span>)</span>
<span id="cb114-18"><a href="test-example.html#cb114-18" tabindex="-1"></a>}</span>
<span id="cb114-19"><a href="test-example.html#cb114-19" tabindex="-1"></a><span class="fu">saveRDS</span>(metaphlan2_ps_LOD_species, <span class="st">&quot;DataSet/Step2/Donor_MGS_phyloseq_LOD_species.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb114-20"><a href="test-example.html#cb114-20" tabindex="-1"></a><span class="fu">saveRDS</span>(metaphlan2_ps_LOD_genus, <span class="st">&quot;DataSet/Step2/Donor_MGS_phyloseq_LOD_genus.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb114-21"><a href="test-example.html#cb114-21" tabindex="-1"></a><span class="fu">saveRDS</span>(metaphlan2_ps_LOD_order, <span class="st">&quot;DataSet/Step2/Donor_MGS_phyloseq_LOD_order.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="step3-brs-checking" class="section level2 hasAnchor" number="10.5">
<h2><span class="header-section-number">10.5</span> Step3: BRS checking<a href="test-example.html#step3-brs-checking" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="test-example.html#cb115-1" tabindex="-1"></a>metaphlan2_ps <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;DataSet/Step1/Donor_MGS_phyloseq.RDS&quot;</span>)</span>
<span id="cb115-2"><a href="test-example.html#cb115-2" tabindex="-1"></a>metaphlan2_ps_LOD_species <span class="ot">&lt;-</span> <span class="fu">aggregate_LOD_taxa</span>(<span class="at">ps =</span> metaphlan2_ps, </span>
<span id="cb115-3"><a href="test-example.html#cb115-3" tabindex="-1"></a>                                                <span class="at">taxa_level =</span> <span class="st">&quot;Species&quot;</span>, </span>
<span id="cb115-4"><a href="test-example.html#cb115-4" tabindex="-1"></a>                                                <span class="at">cutoff =</span> <span class="fl">1e-04</span>)</span>
<span id="cb115-5"><a href="test-example.html#cb115-5" tabindex="-1"></a><span class="fu">tail</span>(metaphlan2_ps_LOD_species<span class="sc">@</span>sam_data <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>())</span></code></pre></div>
<pre><code>##             SampleTubeID   Group Date_Sequencing ProductID SampleType ProductBatch Date_Sampling Date_Receiving SeqID_MGS SeqID_16s
## S_7769      GGM50-210730   Stool      2021-08-03       M50       粪便 CYM50-210735    2021.07.30     2021-08-06      7769      7929
## S_7770 CYM50-210735-0727 Product      2021-08-03       M50   肠菌胶囊 CYM50-210735    2021.07.27     2021-08-06      7770      7930
## S_7771 CYM50-210735-0728 Product      2021-08-03       M50   肠菌胶囊 CYM50-210735    2021.07.28     2021-08-06      7771      7931
## S_7772 CYM50-210735-0729 Product      2021-08-03       M50   肠菌胶囊 CYM50-210735    2021.07.29     2021-08-06      7772      7932
## S_7773 CYM50-210735-0730 Product      2021-08-03       M50   肠菌胶囊 CYM50-210735    2021.07.30     2021-08-06      7773      7933
## S_7222         Community      QC            &lt;NA&gt;       Ref         QC         &lt;NA&gt;          &lt;NA&gt;           &lt;NA&gt;      7222      7327
##                                                    Pipeline_MGS                                                         Pipeline_16s
## S_7769 /share/work/HPC/work_tmp/PipelineJob_180_20210923/output /share/projects/Engineering/pipeline_output/PipelineJob_304_20211203
## S_7770 /share/work/HPC/work_tmp/PipelineJob_180_20210923/output /share/projects/Engineering/pipeline_output/PipelineJob_304_20211203
## S_7771 /share/work/HPC/work_tmp/PipelineJob_180_20210923/output /share/projects/Engineering/pipeline_output/PipelineJob_304_20211203
## S_7772 /share/work/HPC/work_tmp/PipelineJob_180_20210923/output /share/projects/Engineering/pipeline_output/PipelineJob_304_20211203
## S_7773 /share/work/HPC/work_tmp/PipelineJob_180_20210923/output /share/projects/Engineering/pipeline_output/PipelineJob_304_20211203
## S_7222 /share/work/HPC/work_tmp/PipelineJob_180_20210923/output /share/projects/Engineering/pipeline_output/PipelineJob_304_20211203</code></pre>
<div class="sourceCode" id="cb117"><pre class="sourceCode markdown"><code class="sourceCode markdown"><span id="cb117-1"><a href="test-example.html#cb117-1" tabindex="-1"></a>run_RefCheck2(</span>
<span id="cb117-2"><a href="test-example.html#cb117-2" tabindex="-1"></a>    ps = metaphlan2_ps_LOD_species,</span>
<span id="cb117-3"><a href="test-example.html#cb117-3" tabindex="-1"></a>    BRS_ID = &quot;S_7222&quot;,</span>
<span id="cb117-4"><a href="test-example.html#cb117-4" tabindex="-1"></a>    Ref_type = &quot;MGS&quot;L)</span></code></pre></div>
<pre><code>## Noting: the Reference Matrix is for MGS
## 
## ############Matched baterica of the BRS sample#############
## The number of BRS&#39; bacteria matched the Reference Matrix is [2]
## s__Enterococcus_faecalis
## s__Escherichia_coli
## The number of bacteria unmatched the Reference Matrix is [24]
## s__Propionibacterium_acnes
## s__Bifidobacterium_adolescentis
## s__Bifidobacterium_bifidum
## s__Bifidobacterium_longum
## s__Bifidobacterium_pseudocatenulatum
## s__Collinsella_aerofaciens
## s__Eggerthella_unclassified
## s__Bacteroides_fragilis
## s__Bacteroides_intestinalis
## s__Bacteroides_ovatus
## s__Bacteroides_thetaiotaomicron
## s__Bacteroides_uniformis
## s__Bacteroides_vulgatus
## s__Bacteroides_xylanisolvens
## s__Parabacteroides_goldsteinii
## s__Prevotella_copri
## s__Enterococcus_faecium
## s__Lactobacillus_pentosus
## s__Lactobacillus_salivarius
## s__Streptococcus_salivarius
## s__Coprococcus_comes
## s__Dorea_formicigenerans
## s__Roseburia_hominis
## s__Faecalibacterium_prausnitzii
## The number of the additional bacteria compared to the Reference Matrix is [2]
## ###########################################################
## 
## ##################Status of the BRS sample##################
## Whether the BRS has the all bateria of Reference Matrix: FALSE
## Correlation Coefficient of the BRS is: 2.22e-16
## Bray Curtis of the BRS is: 0.1017
## Impurity of the BRS is: 72.95
## ###########################################################
## #####Final Evaluation Results of the BRS #######
## The BRS of sequencing dataset didn&#39;t pass the cutoff of the Reference Matrix
## ###########################################################
## $Matrix
##                                       7682     7683     7684     7685     7842     7843     7844     7845   S_7222        mean
## Propionibacterium_acnes            0.28649  0.22355  0.32199  0.28276  0.37568  0.44201  0.41135  0.39489       NA  0.30430222
## Bifidobacterium_adolescentis       7.05611  6.28460  6.57297  6.25448  4.69357  4.80628  4.94943  4.84278       NA  5.05113556
## Bifidobacterium_bifidum            1.32803  1.19346  1.26883  1.05154  1.34640  1.40471  1.43721  1.48097       NA  1.16790556
## Bifidobacterium_longum            10.31832  9.25812  9.69184  7.76031 11.03311 11.61484 12.29030 11.69019       NA  9.29522556
## Bifidobacterium_pseudocatenulatum  6.89760  6.36177  7.39605  5.76804  6.19464  6.47094  7.41615  6.27837       NA  5.86484000
## Collinsella_aerofaciens            0.56189  0.53249  0.60476  0.47934  0.65513  0.68833  0.76063  0.66896       NA  0.55017000
## Eggerthella_unclassified           1.21416  1.06143  1.19122  0.85743  1.36030  1.48457  1.61204  1.50700       NA  1.14312778
## Bacteroides_fragilis               6.56752  7.19828  7.27658  7.31998  7.10174  6.70626  6.60219  6.11088       NA  6.09815889
## Bacteroides_intestinalis           0.09448  0.09550  0.10216  0.09326  0.08448  0.09556  0.08852  0.10469       NA  0.08429444
## Bacteroides_ovatus                 3.08552  3.27226  3.10904  3.24565  3.51376  3.50063  3.37872  3.42300       NA  2.94762000
## Bacteroides_thetaiotaomicron       3.24207  3.31897  3.22418  3.43809  3.35611  3.38323  3.29098  3.30355       NA  2.95079778
## Bacteroides_uniformis              2.24271  2.24035  1.92015  2.18435  2.43230  2.38180  2.13830  2.41437       NA  1.99492556
## Bacteroides_vulgatus               3.06672  3.20369  3.14979  3.15352  3.24822  3.09280  3.13038  3.06113       NA  2.78958333
## Bacteroides_xylanisolvens          1.55687  1.84824  1.91166  1.85273  1.75220  1.74002  1.69811  1.67676       NA  1.55962111
## Parabacteroides_goldsteinii        5.92564  5.95499  6.04638  5.85282  6.85811  6.50800  6.35758  6.96646       NA  5.60777556
## Prevotella_copri                   2.03757  1.99619  1.92504  2.15422  1.57638  1.60584  1.57913  1.59224       NA  1.60740111
## Enterococcus_faecalis             11.46695 12.46200 11.92117 11.95884 13.71474 13.67805 13.21782 13.27297 17.19286 13.20948889
## Enterococcus_faecium               4.57147  4.64943  4.52463  4.76407  5.45590  5.17641  5.15276  5.24896       NA  4.39373667
## Lactobacillus_pentosus             0.75844  0.75524  0.72958  0.80796  0.91548  0.89940  0.81308  0.94831       NA  0.73638778
## Lactobacillus_salivarius           4.89005  5.18462  5.12745  7.79194  2.72469  2.25673  2.10356  2.02202       NA  3.56678444
## Streptococcus_salivarius           3.54220  3.74119  3.62036  4.01546  2.90216  2.70193  2.61760  2.57348       NA  2.85715333
## Coprococcus_comes                  2.28009  2.42978  2.25227  2.82904  1.21527  1.16485  1.07160  1.07834       NA  1.59124889
## Dorea_formicigenerans              4.83413  5.02149  5.18268  5.56891  3.34720  3.09877  3.08677  2.70732       NA  3.64969667
## Roseburia_hominis                  0.04384  0.04183  0.04107  0.02304  0.03853  0.03464  0.03532  0.03597       NA  0.03269333
## Faecalibacterium_prausnitzii       0.65572  0.60153  0.60112  0.62079  0.54147  0.55383  0.58806  0.54655       NA  0.52323000
## Escherichia_coli                   8.76957  8.33337  7.70416  8.11247 10.44118 10.33823  9.96261 11.06988  9.85431  9.39842000
## Impurity_level                     2.70584  2.73563  2.58287  1.75896  3.12125  4.17134  4.20980  4.97996 72.95000 11.02396111
##                                                                                Evaluation
## Propionibacterium_acnes           S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Bifidobacterium_adolescentis      S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Bifidobacterium_bifidum           S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Bifidobacterium_longum            S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Bifidobacterium_pseudocatenulatum S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Collinsella_aerofaciens           S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Eggerthella_unclassified          S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Bacteroides_fragilis              S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Bacteroides_intestinalis          S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Bacteroides_ovatus                S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Bacteroides_thetaiotaomicron      S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Bacteroides_uniformis             S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Bacteroides_vulgatus              S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Bacteroides_xylanisolvens         S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Parabacteroides_goldsteinii       S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Prevotella_copri                  S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Enterococcus_faecalis             S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Enterococcus_faecium              S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Lactobacillus_pentosus            S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Lactobacillus_salivarius          S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Streptococcus_salivarius          S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Coprococcus_comes                 S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Dorea_formicigenerans             S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Roseburia_hominis                 S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Faecalibacterium_prausnitzii      S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Escherichia_coli                  S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## Impurity_level                    S_7222 didn&#39;t pass the threshold (2023-10-27 16:59:47).
## 
## $Assess
##          Gold_Cutoff       BRS
## Coef          0.8726 2.220e-16
## Bray          0.2064 1.017e-01
## Impurity      6.4400 7.295e+01</code></pre>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="test-example.html#cb119-1" tabindex="-1"></a>metaphlan2_ps_LOD_species_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb119-2"><a href="test-example.html#cb119-2" tabindex="-1"></a>                                           <span class="at">ps =</span> metaphlan2_ps_LOD_species,</span>
<span id="cb119-3"><a href="test-example.html#cb119-3" tabindex="-1"></a>                                           <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb119-4"><a href="test-example.html#cb119-4" tabindex="-1"></a>                                           <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>,</span>
<span id="cb119-5"><a href="test-example.html#cb119-5" tabindex="-1"></a>                                           <span class="at">discard =</span> <span class="cn">TRUE</span>)</span>
<span id="cb119-6"><a href="test-example.html#cb119-6" tabindex="-1"></a>metaphlan2_ps_LOD_species_remove_BRS</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 180 taxa and 144 samples ]
## sample_data() Sample Data:       [ 144 samples by 12 sample variables ]
## tax_table()   Taxonomy Table:    [ 180 taxa by 7 taxonomic ranks ]</code></pre>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="test-example.html#cb121-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="st">&quot;DataSet/Step3/&quot;</span>)) {</span>
<span id="cb121-2"><a href="test-example.html#cb121-2" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="st">&quot;DataSet/Step3/&quot;</span>)</span>
<span id="cb121-3"><a href="test-example.html#cb121-3" tabindex="-1"></a>}</span>
<span id="cb121-4"><a href="test-example.html#cb121-4" tabindex="-1"></a><span class="fu">saveRDS</span>(metaphlan2_ps_LOD_species_remove_BRS, <span class="st">&quot;DataSet/Step3/Donor_MGS_phyloseq_LOD_species_remove_BRS.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="step4-extracting-specific-taxonomic-level" class="section level2 hasAnchor" number="10.6">
<h2><span class="header-section-number">10.6</span> Step4: Extracting specific taxonomic level<a href="test-example.html#step4-extracting-specific-taxonomic-level" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Removing spike-in sample (BRS)</li>
</ul>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="test-example.html#cb122-1" tabindex="-1"></a>metaphlan2_ps <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;DataSet/Step1/Donor_MGS_phyloseq.RDS&quot;</span>)</span>
<span id="cb122-2"><a href="test-example.html#cb122-2" tabindex="-1"></a>metaphlan2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb122-3"><a href="test-example.html#cb122-3" tabindex="-1"></a>                               <span class="at">ps =</span> metaphlan2_ps,</span>
<span id="cb122-4"><a href="test-example.html#cb122-4" tabindex="-1"></a>                               <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb122-5"><a href="test-example.html#cb122-5" tabindex="-1"></a>                               <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>,</span>
<span id="cb122-6"><a href="test-example.html#cb122-6" tabindex="-1"></a>                               <span class="at">discard =</span> <span class="cn">TRUE</span>)</span>
<span id="cb122-7"><a href="test-example.html#cb122-7" tabindex="-1"></a>metaphlan2_ps_LOD_species_remove_BRS</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 180 taxa and 144 samples ]
## sample_data() Sample Data:       [ 144 samples by 12 sample variables ]
## tax_table()   Taxonomy Table:    [ 180 taxa by 7 taxonomic ranks ]</code></pre>
<ul>
<li>Species</li>
</ul>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="test-example.html#cb124-1" tabindex="-1"></a>metaphlan2_ps_remove_BRS_LOD_species <span class="ot">&lt;-</span> <span class="fu">aggregate_LOD_taxa</span>(</span>
<span id="cb124-2"><a href="test-example.html#cb124-2" tabindex="-1"></a>                                                <span class="at">ps =</span> metaphlan2_ps_remove_BRS, </span>
<span id="cb124-3"><a href="test-example.html#cb124-3" tabindex="-1"></a>                                                <span class="at">taxa_level =</span> <span class="st">&quot;Species&quot;</span>, </span>
<span id="cb124-4"><a href="test-example.html#cb124-4" tabindex="-1"></a>                                                <span class="at">cutoff =</span> <span class="fl">1e-04</span>)</span>
<span id="cb124-5"><a href="test-example.html#cb124-5" tabindex="-1"></a>metaphlan2_ps_remove_BRS_LOD_species</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 180 taxa and 144 samples ]
## sample_data() Sample Data:       [ 144 samples by 12 sample variables ]
## tax_table()   Taxonomy Table:    [ 180 taxa by 7 taxonomic ranks ]</code></pre>
<ul>
<li>Genus</li>
</ul>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="test-example.html#cb126-1" tabindex="-1"></a>metaphlan2_ps_remove_BRS_LOD_genus <span class="ot">&lt;-</span> <span class="fu">aggregate_LOD_taxa</span>(</span>
<span id="cb126-2"><a href="test-example.html#cb126-2" tabindex="-1"></a>                                              <span class="at">ps =</span> metaphlan2_ps_remove_BRS, </span>
<span id="cb126-3"><a href="test-example.html#cb126-3" tabindex="-1"></a>                                              <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>, </span>
<span id="cb126-4"><a href="test-example.html#cb126-4" tabindex="-1"></a>                                              <span class="at">cutoff =</span> <span class="fl">1e-04</span>)</span>
<span id="cb126-5"><a href="test-example.html#cb126-5" tabindex="-1"></a>metaphlan2_ps_remove_BRS_LOD_genus</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 67 taxa and 144 samples ]
## sample_data() Sample Data:       [ 144 samples by 12 sample variables ]
## tax_table()   Taxonomy Table:    [ 67 taxa by 6 taxonomic ranks ]</code></pre>
<ul>
<li>Phylum</li>
</ul>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="test-example.html#cb128-1" tabindex="-1"></a>metaphlan2_ps_remove_BRS_LOD_phylum <span class="ot">&lt;-</span> <span class="fu">aggregate_LOD_taxa</span>(</span>
<span id="cb128-2"><a href="test-example.html#cb128-2" tabindex="-1"></a>                                               <span class="at">ps =</span> metaphlan2_ps_remove_BRS, </span>
<span id="cb128-3"><a href="test-example.html#cb128-3" tabindex="-1"></a>                                               <span class="at">taxa_level =</span> <span class="st">&quot;Phylum&quot;</span>, </span>
<span id="cb128-4"><a href="test-example.html#cb128-4" tabindex="-1"></a>                                               <span class="at">cutoff =</span> <span class="fl">1e-04</span>)</span>
<span id="cb128-5"><a href="test-example.html#cb128-5" tabindex="-1"></a>metaphlan2_ps_remove_BRS_LOD_phylum</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 7 taxa and 144 samples ]
## sample_data() Sample Data:       [ 144 samples by 12 sample variables ]
## tax_table()   Taxonomy Table:    [ 7 taxa by 2 taxonomic ranks ]</code></pre>
<ul>
<li>output</li>
</ul>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="test-example.html#cb130-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="st">&quot;DataSet/Step4/&quot;</span>)) {</span>
<span id="cb130-2"><a href="test-example.html#cb130-2" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="st">&quot;DataSet/Step4/&quot;</span>)</span>
<span id="cb130-3"><a href="test-example.html#cb130-3" tabindex="-1"></a>}</span>
<span id="cb130-4"><a href="test-example.html#cb130-4" tabindex="-1"></a><span class="fu">saveRDS</span>(metaphlan2_ps_remove_BRS_LOD_species, <span class="st">&quot;DataSet/Step4/Donor_MGS_phyloseq_remove_BRS_LOD_species.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb130-5"><a href="test-example.html#cb130-5" tabindex="-1"></a><span class="fu">saveRDS</span>(metaphlan2_ps_remove_BRS_LOD_genus, <span class="st">&quot;DataSet/Step4/Donor_MGS_phyloseq_remove_BRS_LOD_genus.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb130-6"><a href="test-example.html#cb130-6" tabindex="-1"></a><span class="fu">saveRDS</span>(metaphlan2_ps_remove_BRS_LOD_phylum, <span class="st">&quot;DataSet/Step4/Donor_MGS_phyloseq_remove_BRS_LOD_phylum.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="step5-globalview" class="section level2 hasAnchor" number="10.7">
<h2><span class="header-section-number">10.7</span> Step5: GlobalView<a href="test-example.html#step5-globalview" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="test-example.html#cb131-1" tabindex="-1"></a>metaphlan2_ps_remove_BRS_LOD_species <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;DataSet/Step4/Donor_MGS_phyloseq_remove_BRS_LOD_species.RDS&quot;</span>)</span>
<span id="cb131-2"><a href="test-example.html#cb131-2" tabindex="-1"></a></span>
<span id="cb131-3"><a href="test-example.html#cb131-3" tabindex="-1"></a><span class="co"># alpha</span></span>
<span id="cb131-4"><a href="test-example.html#cb131-4" tabindex="-1"></a>metaphlan2_ps_remove_BRS_species_alpha <span class="ot">&lt;-</span> <span class="fu">run_alpha_diversity</span>(</span>
<span id="cb131-5"><a href="test-example.html#cb131-5" tabindex="-1"></a>                                            <span class="at">ps =</span> metaphlan2_ps_remove_BRS_LOD_species, </span>
<span id="cb131-6"><a href="test-example.html#cb131-6" tabindex="-1"></a>                                            <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Simpson&quot;</span>, <span class="st">&quot;InvSimpson&quot;</span>))</span>
<span id="cb131-7"><a href="test-example.html#cb131-7" tabindex="-1"></a><span class="fu">plot_boxplot</span>(<span class="at">data =</span> metaphlan2_ps_remove_BRS_species_alpha,</span>
<span id="cb131-8"><a href="test-example.html#cb131-8" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Simpson&quot;</span>, <span class="st">&quot;InvSimpson&quot;</span>),</span>
<span id="cb131-9"><a href="test-example.html#cb131-9" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb131-10"><a href="test-example.html#cb131-10" tabindex="-1"></a>             <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Stool&quot;</span>, <span class="st">&quot;Product&quot;</span>),</span>
<span id="cb131-11"><a href="test-example.html#cb131-11" tabindex="-1"></a>             <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-75-1"></span>
<img src="MGS-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-75-1.png" alt="diversity and ordination and composition(Example)" width="768" />
<p class="caption">
Figure 10.1: diversity and ordination and composition(Example)
</p>
</div>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="test-example.html#cb132-1" tabindex="-1"></a><span class="co"># beta</span></span>
<span id="cb132-2"><a href="test-example.html#cb132-2" tabindex="-1"></a>metaphlan2_ps_remove_BRS_species_beta <span class="ot">&lt;-</span> <span class="fu">run_beta_diversity</span>(</span>
<span id="cb132-3"><a href="test-example.html#cb132-3" tabindex="-1"></a>                                            <span class="at">ps =</span> metaphlan2_ps_remove_BRS_LOD_species, </span>
<span id="cb132-4"><a href="test-example.html#cb132-4" tabindex="-1"></a>                                            <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>)</span>
<span id="cb132-5"><a href="test-example.html#cb132-5" tabindex="-1"></a><span class="fu">plot_distance_corrplot</span>(<span class="at">datMatrix =</span> metaphlan2_ps_remove_BRS_species_beta<span class="sc">$</span>BetaDistance)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-75-2"></span>
<img src="MGS-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-75-2.png" alt="diversity and ordination and composition(Example)" width="768" />
<p class="caption">
Figure 10.2: diversity and ordination and composition(Example)
</p>
</div>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="test-example.html#cb133-1" tabindex="-1"></a><span class="co"># permanova</span></span>
<span id="cb133-2"><a href="test-example.html#cb133-2" tabindex="-1"></a><span class="fu">run_permanova</span>(<span class="at">ps =</span> metaphlan2_ps_remove_BRS_LOD_species, </span>
<span id="cb133-3"><a href="test-example.html#cb133-3" tabindex="-1"></a>              <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>, </span>
<span id="cb133-4"><a href="test-example.html#cb133-4" tabindex="-1"></a>              <span class="at">columns =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<pre><code>##       SumsOfSample Df SumsOfSqs  MeanSqs  F.Model         R2 Pr(&gt;F) AdjustedPvalue
## Group          144  1  1.328635 1.328635 6.307557 0.04253025  0.001          0.001</code></pre>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="test-example.html#cb135-1" tabindex="-1"></a><span class="co"># beta dispersion</span></span>
<span id="cb135-2"><a href="test-example.html#cb135-2" tabindex="-1"></a>beta_df <span class="ot">&lt;-</span> <span class="fu">run_beta_diversity</span>(<span class="at">ps =</span> metaphlan2_ps_remove_BRS_LOD_species, </span>
<span id="cb135-3"><a href="test-example.html#cb135-3" tabindex="-1"></a>                              <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>, </span>
<span id="cb135-4"><a href="test-example.html#cb135-4" tabindex="-1"></a>                              <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<pre><code>## 
## Permutation test for homogeneity of multivariate dispersions
## Permutation: free
## Number of permutations: 999
## 
## Response: Distances
##            Df  Sum Sq  Mean Sq      F N.Perm Pr(&gt;F)
## Groups      1 0.01652 0.016518 1.1697    999  0.282
## Residuals 142 2.00532 0.014122                     
## 
## Pairwise comparisons:
## (Observed p-value below diagonal, permuted p-value above diagonal)
##         Product Stool
## Product         0.281
## Stool   0.28129</code></pre>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="test-example.html#cb137-1" tabindex="-1"></a><span class="co"># ordination</span></span>
<span id="cb137-2"><a href="test-example.html#cb137-2" tabindex="-1"></a>metaphlan2_ps_ordination <span class="ot">&lt;-</span> <span class="fu">run_ordination</span>(</span>
<span id="cb137-3"><a href="test-example.html#cb137-3" tabindex="-1"></a>                       <span class="at">ps =</span> metaphlan2_ps_remove_BRS_LOD_species,</span>
<span id="cb137-4"><a href="test-example.html#cb137-4" tabindex="-1"></a>                       <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb137-5"><a href="test-example.html#cb137-5" tabindex="-1"></a>                       <span class="at">method =</span> <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb137-6"><a href="test-example.html#cb137-6" tabindex="-1"></a></span>
<span id="cb137-7"><a href="test-example.html#cb137-7" tabindex="-1"></a><span class="fu">plot_Ordination</span>(<span class="at">ResultList =</span> metaphlan2_ps_ordination, </span>
<span id="cb137-8"><a href="test-example.html#cb137-8" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>, </span>
<span id="cb137-9"><a href="test-example.html#cb137-9" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Stool&quot;</span>, <span class="st">&quot;Product&quot;</span>),</span>
<span id="cb137-10"><a href="test-example.html#cb137-10" tabindex="-1"></a>                <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-75-3"></span>
<img src="MGS-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-75-3.png" alt="diversity and ordination and composition(Example)" width="768" />
<p class="caption">
Figure 10.3: diversity and ordination and composition(Example)
</p>
</div>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="test-example.html#cb138-1" tabindex="-1"></a><span class="co"># Microbial composition</span></span>
<span id="cb138-2"><a href="test-example.html#cb138-2" tabindex="-1"></a><span class="fu">plot_stacked_bar_XIVZ</span>(</span>
<span id="cb138-3"><a href="test-example.html#cb138-3" tabindex="-1"></a>        <span class="at">phyloseq =</span> metaphlan2_ps_remove_BRS_LOD_species,</span>
<span id="cb138-4"><a href="test-example.html#cb138-4" tabindex="-1"></a>        <span class="at">level =</span> <span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb138-5"><a href="test-example.html#cb138-5" tabindex="-1"></a>        <span class="at">feature =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-75-4"></span>
<img src="MGS-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-75-4.png" alt="diversity and ordination and composition(Example)" width="768" />
<p class="caption">
Figure 10.4: diversity and ordination and composition(Example)
</p>
</div>
</div>
<div id="step6-differential-analysis" class="section level2 hasAnchor" number="10.8">
<h2><span class="header-section-number">10.8</span> Step6: Differential Analysis<a href="test-example.html#step6-differential-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="test-example.html#cb139-1" tabindex="-1"></a>metaphlan2_ps_remove_BRS_LOD_species <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;DataSet/Step4/Donor_MGS_phyloseq_remove_BRS_LOD_species.RDS&quot;</span>)</span>
<span id="cb139-2"><a href="test-example.html#cb139-2" tabindex="-1"></a></span>
<span id="cb139-3"><a href="test-example.html#cb139-3" tabindex="-1"></a><span class="co"># filter &amp; trim</span></span>
<span id="cb139-4"><a href="test-example.html#cb139-4" tabindex="-1"></a>metaphlan2_ps_remove_BRS_species_filter <span class="ot">&lt;-</span> <span class="fu">run_filter</span>(<span class="at">ps =</span> metaphlan2_ps_remove_BRS_LOD_species, </span>
<span id="cb139-5"><a href="test-example.html#cb139-5" tabindex="-1"></a>                                                      <span class="at">cutoff =</span> <span class="fl">1e-4</span>, </span>
<span id="cb139-6"><a href="test-example.html#cb139-6" tabindex="-1"></a>                                                      <span class="at">unclass =</span> <span class="cn">TRUE</span>)</span>
<span id="cb139-7"><a href="test-example.html#cb139-7" tabindex="-1"></a>metaphlan2_ps_remove_BRS_species_filter_trim <span class="ot">&lt;-</span> <span class="fu">run_trim</span>(<span class="at">object =</span> metaphlan2_ps_remove_BRS_species_filter, </span>
<span id="cb139-8"><a href="test-example.html#cb139-8" tabindex="-1"></a>                                                         <span class="at">cutoff =</span> <span class="fl">0.1</span>, <span class="at">trim =</span> <span class="st">&quot;feature&quot;</span>)</span>
<span id="cb139-9"><a href="test-example.html#cb139-9" tabindex="-1"></a>metaphlan2_ps_remove_BRS_species_filter_trim</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 110 taxa and 144 samples ]
## sample_data() Sample Data:       [ 144 samples by 12 sample variables ]
## tax_table()   Taxonomy Table:    [ 110 taxa by 7 taxonomic ranks ]</code></pre>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="test-example.html#cb141-1" tabindex="-1"></a><span class="co"># lefse</span></span>
<span id="cb141-2"><a href="test-example.html#cb141-2" tabindex="-1"></a><span class="co"># metaphlan2_ps_lefse &lt;- run_lefse(</span></span>
<span id="cb141-3"><a href="test-example.html#cb141-3" tabindex="-1"></a><span class="co">#                           ps = metaphlan2_ps_remove_BRS_species_filter_trim,</span></span>
<span id="cb141-4"><a href="test-example.html#cb141-4" tabindex="-1"></a><span class="co">#                           group = &quot;Group&quot;,</span></span>
<span id="cb141-5"><a href="test-example.html#cb141-5" tabindex="-1"></a><span class="co">#                           group_names = c(&quot;Stool&quot;, &quot;Product&quot;),</span></span>
<span id="cb141-6"><a href="test-example.html#cb141-6" tabindex="-1"></a><span class="co">#                           norm = &quot;CPM&quot;,</span></span>
<span id="cb141-7"><a href="test-example.html#cb141-7" tabindex="-1"></a><span class="co">#                           Lda = 2)</span></span>
<span id="cb141-8"><a href="test-example.html#cb141-8" tabindex="-1"></a>metaphlan2_ps_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse2</span>(</span>
<span id="cb141-9"><a href="test-example.html#cb141-9" tabindex="-1"></a>                          <span class="at">ps =</span> metaphlan2_ps_remove_BRS_species_filter_trim,</span>
<span id="cb141-10"><a href="test-example.html#cb141-10" tabindex="-1"></a>                          <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb141-11"><a href="test-example.html#cb141-11" tabindex="-1"></a>                          <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Stool&quot;</span>, <span class="st">&quot;Product&quot;</span>),</span>
<span id="cb141-12"><a href="test-example.html#cb141-12" tabindex="-1"></a>                          <span class="at">norm =</span> <span class="st">&quot;CPM&quot;</span>,</span>
<span id="cb141-13"><a href="test-example.html#cb141-13" tabindex="-1"></a>                          <span class="at">lda_cutoff =</span> <span class="dv">2</span>)</span>
<span id="cb141-14"><a href="test-example.html#cb141-14" tabindex="-1"></a></span>
<span id="cb141-15"><a href="test-example.html#cb141-15" tabindex="-1"></a><span class="co"># # don&#39;t run this code when you do lefse in reality</span></span>
<span id="cb141-16"><a href="test-example.html#cb141-16" tabindex="-1"></a><span class="co"># metaphlan2_ps_lefse$LDA_Score &lt;- metaphlan2_ps_lefse$LDA_Score * 1000</span></span>
<span id="cb141-17"><a href="test-example.html#cb141-17" tabindex="-1"></a><span class="fu">plot_lefse</span>(</span>
<span id="cb141-18"><a href="test-example.html#cb141-18" tabindex="-1"></a>    <span class="at">da_res =</span> metaphlan2_ps_lefse,</span>
<span id="cb141-19"><a href="test-example.html#cb141-19" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;LDA_Score&quot;</span>,</span>
<span id="cb141-20"><a href="test-example.html#cb141-20" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="dv">2</span>,</span>
<span id="cb141-21"><a href="test-example.html#cb141-21" tabindex="-1"></a>    <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-76-1"></span>
<img src="MGS-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-76-1.png" alt="Differential Analysis (Example)" width="768" />
<p class="caption">
Figure 10.5: Differential Analysis (Example)
</p>
</div>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="test-example.html#cb142-1" tabindex="-1"></a>metaphlan2_ps_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb142-2"><a href="test-example.html#cb142-2" tabindex="-1"></a>                          <span class="at">ps =</span> metaphlan2_ps_remove_BRS_species_filter_trim,</span>
<span id="cb142-3"><a href="test-example.html#cb142-3" tabindex="-1"></a>                          <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb142-4"><a href="test-example.html#cb142-4" tabindex="-1"></a>                          <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Stool&quot;</span>, <span class="st">&quot;Product&quot;</span>))</span>
<span id="cb142-5"><a href="test-example.html#cb142-5" tabindex="-1"></a><span class="fu">plot_volcano</span>(</span>
<span id="cb142-6"><a href="test-example.html#cb142-6" tabindex="-1"></a>    <span class="at">da_res =</span> metaphlan2_ps_wilcox,</span>
<span id="cb142-7"><a href="test-example.html#cb142-7" tabindex="-1"></a>    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Stool&quot;</span>, <span class="st">&quot;Product&quot;</span>),</span>
<span id="cb142-8"><a href="test-example.html#cb142-8" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;Log2FoldChange (Rank)</span><span class="sc">\n</span><span class="st">Stool_vs_Product&quot;</span>,</span>
<span id="cb142-9"><a href="test-example.html#cb142-9" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb142-10"><a href="test-example.html#cb142-10" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb142-11"><a href="test-example.html#cb142-11" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb142-12"><a href="test-example.html#cb142-12" tabindex="-1"></a>    <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb142-13"><a href="test-example.html#cb142-13" tabindex="-1"></a>    <span class="at">topN =</span> <span class="dv">5</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-76-2"></span>
<img src="MGS-microbiota-data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-76-2.png" alt="Differential Analysis (Example)" width="768" />
<p class="caption">
Figure 10.6: Differential Analysis (Example)
</p>
</div>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="test-example.html#cb143-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="st">&quot;DataSet/Step8/&quot;</span>)) {</span>
<span id="cb143-2"><a href="test-example.html#cb143-2" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="st">&quot;DataSet/Step8/&quot;</span>)</span>
<span id="cb143-3"><a href="test-example.html#cb143-3" tabindex="-1"></a>}</span>
<span id="cb143-4"><a href="test-example.html#cb143-4" tabindex="-1"></a><span class="fu">saveRDS</span>(metaphlan2_ps_remove_BRS_species_filter_trim, <span class="st">&quot;DataSet/Step6/Donor_MGS_phyloseq_remove_BRS_species_filter_trim.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="systematic-information-8" class="section level2 hasAnchor" number="10.9">
<h2><span class="header-section-number">10.9</span> Systematic Information<a href="test-example.html#systematic-information-8" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="test-example.html#cb144-1" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.1.3 (2022-03-10)
##  os       macOS Monterey 12.2.1
##  system   x86_64, darwin17.0
##  ui       RStudio
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       Asia/Shanghai
##  date     2023-10-27
##  rstudio  2023.09.0+463 Desert Sunflower (desktop)
##  pandoc   3.1.1 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/ (via rmarkdown)
## 
## ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version   date (UTC) lib source
##  abind                  1.4-5     2016-07-21 [2] CRAN (R 4.1.0)
##  ade4                   1.7-22    2023-02-06 [2] CRAN (R 4.1.2)
##  ALDEx2                 1.30.0    2022-11-01 [2] Bioconductor
##  annotate               1.72.0    2021-10-26 [2] Bioconductor
##  AnnotationDbi          1.60.2    2023-03-10 [2] Bioconductor
##  ape                  * 5.7-1     2023-03-13 [2] CRAN (R 4.1.2)
##  askpass                1.1       2019-01-13 [2] CRAN (R 4.1.0)
##  backports              1.4.1     2021-12-13 [2] CRAN (R 4.1.0)
##  base64enc              0.1-3     2015-07-28 [2] CRAN (R 4.1.0)
##  bayesm                 3.1-5     2022-12-02 [2] CRAN (R 4.1.2)
##  Biobase              * 2.54.0    2021-10-26 [2] Bioconductor
##  BiocGenerics         * 0.40.0    2021-10-26 [2] Bioconductor
##  BiocParallel           1.28.3    2021-12-09 [2] Bioconductor
##  biomformat             1.22.0    2021-10-26 [2] Bioconductor
##  Biostrings             2.62.0    2021-10-26 [2] Bioconductor
##  bit                    4.0.5     2022-11-15 [2] CRAN (R 4.1.2)
##  bit64                  4.0.5     2020-08-30 [2] CRAN (R 4.1.0)
##  bitops                 1.0-7     2021-04-24 [2] CRAN (R 4.1.0)
##  blob                   1.2.4     2023-03-17 [2] CRAN (R 4.1.2)
##  bookdown               0.34      2023-05-09 [2] CRAN (R 4.1.2)
##  broom                  1.0.5     2023-06-09 [2] CRAN (R 4.1.3)
##  bslib                  0.5.0     2023-06-09 [2] CRAN (R 4.1.3)
##  cachem                 1.0.8     2023-05-01 [2] CRAN (R 4.1.2)
##  callr                  3.7.3     2022-11-02 [2] CRAN (R 4.1.2)
##  car                    3.1-2     2023-03-30 [2] CRAN (R 4.1.2)
##  carData                3.0-5     2022-01-06 [2] CRAN (R 4.1.2)
##  caTools                1.18.2    2021-03-28 [2] CRAN (R 4.1.0)
##  cccd                   1.6       2022-04-08 [2] CRAN (R 4.1.2)
##  cellranger             1.1.0     2016-07-27 [2] CRAN (R 4.1.0)
##  checkmate              2.2.0     2023-04-27 [2] CRAN (R 4.1.2)
##  class                  7.3-22    2023-05-03 [2] CRAN (R 4.1.2)
##  classInt               0.4-9     2023-02-28 [2] CRAN (R 4.1.2)
##  cli                    3.6.1     2023-03-23 [2] CRAN (R 4.1.2)
##  cluster                2.1.4     2022-08-22 [2] CRAN (R 4.1.2)
##  coda                   0.19-4    2020-09-30 [2] CRAN (R 4.1.0)
##  codetools              0.2-19    2023-02-01 [2] CRAN (R 4.1.2)
##  coin                   1.4-2     2021-10-08 [2] CRAN (R 4.1.0)
##  colorspace             2.1-0     2023-01-23 [2] CRAN (R 4.1.2)
##  compositions           2.0-6     2023-04-13 [2] CRAN (R 4.1.2)
##  conflicted           * 1.2.0     2023-02-01 [2] CRAN (R 4.1.2)
##  corpcor                1.6.10    2021-09-16 [2] CRAN (R 4.1.0)
##  corrplot               0.92      2021-11-18 [2] CRAN (R 4.1.0)
##  cowplot                1.1.1     2020-12-30 [2] CRAN (R 4.1.0)
##  crayon                 1.5.2     2022-09-29 [2] CRAN (R 4.1.2)
##  crosstalk              1.2.0     2021-11-04 [2] CRAN (R 4.1.0)
##  data.table             1.14.8    2023-02-17 [2] CRAN (R 4.1.2)
##  DBI                    1.1.3     2022-06-18 [2] CRAN (R 4.1.2)
##  DelayedArray           0.20.0    2021-10-26 [2] Bioconductor
##  deldir                 1.0-9     2023-05-17 [2] CRAN (R 4.1.3)
##  DEoptimR               1.0-14    2023-06-09 [2] CRAN (R 4.1.3)
##  DESeq2                 1.34.0    2021-10-26 [2] Bioconductor
##  devtools             * 2.4.5     2022-10-11 [2] CRAN (R 4.1.2)
##  digest                 0.6.33    2023-07-07 [1] CRAN (R 4.1.3)
##  doParallel             1.0.17    2022-02-07 [2] CRAN (R 4.1.2)
##  doSNOW                 1.0.20    2022-02-04 [2] CRAN (R 4.1.2)
##  dplyr                * 1.1.2     2023-04-20 [2] CRAN (R 4.1.2)
##  DT                     0.28      2023-05-18 [2] CRAN (R 4.1.3)
##  dynamicTreeCut         1.63-1    2016-03-11 [2] CRAN (R 4.1.0)
##  e1071                  1.7-13    2023-02-01 [2] CRAN (R 4.1.2)
##  edgeR                  3.36.0    2021-10-26 [2] Bioconductor
##  ellipsis               0.3.2     2021-04-29 [2] CRAN (R 4.1.0)
##  emmeans                1.8.7     2023-06-23 [1] CRAN (R 4.1.3)
##  estimability           1.4.1     2022-08-05 [2] CRAN (R 4.1.2)
##  evaluate               0.21      2023-05-05 [2] CRAN (R 4.1.2)
##  FactoMineR             2.8       2023-03-27 [2] CRAN (R 4.1.2)
##  fansi                  1.0.4     2023-01-22 [2] CRAN (R 4.1.2)
##  farver                 2.1.1     2022-07-06 [2] CRAN (R 4.1.2)
##  fastcluster            1.2.3     2021-05-24 [2] CRAN (R 4.1.0)
##  fastmap                1.1.1     2023-02-24 [2] CRAN (R 4.1.2)
##  fdrtool                1.2.17    2021-11-13 [2] CRAN (R 4.1.0)
##  filematrix             1.3       2018-02-27 [2] CRAN (R 4.1.0)
##  flashClust             1.01-2    2012-08-21 [2] CRAN (R 4.1.0)
##  FNN                    1.1.3.2   2023-03-20 [2] CRAN (R 4.1.2)
##  foreach                1.5.2     2022-02-02 [2] CRAN (R 4.1.2)
##  foreign                0.8-84    2022-12-06 [2] CRAN (R 4.1.2)
##  forestplot             3.1.1     2022-12-06 [2] CRAN (R 4.1.2)
##  formatR                1.14      2023-01-17 [2] CRAN (R 4.1.2)
##  Formula                1.2-5     2023-02-24 [2] CRAN (R 4.1.2)
##  fs                     1.6.2     2023-04-25 [2] CRAN (R 4.1.2)
##  futile.logger        * 1.4.3     2016-07-10 [2] CRAN (R 4.1.0)
##  futile.options         1.0.1     2018-04-20 [2] CRAN (R 4.1.0)
##  genefilter             1.76.0    2021-10-26 [2] Bioconductor
##  geneplotter            1.72.0    2021-10-26 [2] Bioconductor
##  generics               0.1.3     2022-07-05 [2] CRAN (R 4.1.2)
##  GenomeInfoDb         * 1.30.1    2022-01-30 [2] Bioconductor
##  GenomeInfoDbData       1.2.7     2022-03-09 [2] Bioconductor
##  GenomicRanges        * 1.46.1    2021-11-18 [2] Bioconductor
##  ggiraph                0.8.7     2023-03-17 [2] CRAN (R 4.1.2)
##  ggiraphExtra           0.3.0     2020-10-06 [2] CRAN (R 4.1.2)
##  ggplot2              * 3.4.2     2023-04-03 [2] CRAN (R 4.1.2)
##  ggpubr               * 0.6.0     2023-02-10 [2] CRAN (R 4.1.2)
##  ggrepel                0.9.3     2023-02-03 [2] CRAN (R 4.1.2)
##  ggsci                  3.0.0     2023-03-08 [2] CRAN (R 4.1.2)
##  ggsignif               0.6.4     2022-10-13 [2] CRAN (R 4.1.2)
##  ggVennDiagram          1.2.2     2022-09-08 [2] CRAN (R 4.1.2)
##  glasso                 1.11      2019-10-01 [2] CRAN (R 4.1.0)
##  glmnet                 4.1-7     2023-03-23 [2] CRAN (R 4.1.2)
##  glue                 * 1.6.2     2022-02-24 [2] CRAN (R 4.1.2)
##  Gmisc                * 3.0.2     2023-03-13 [2] CRAN (R 4.1.2)
##  GO.db                  3.14.0    2022-04-11 [2] Bioconductor
##  gplots                 3.1.3     2022-04-25 [2] CRAN (R 4.1.2)
##  gridExtra              2.3       2017-09-09 [2] CRAN (R 4.1.0)
##  gtable                 0.3.3     2023-03-21 [2] CRAN (R 4.1.2)
##  gtools                 3.9.4     2022-11-27 [2] CRAN (R 4.1.2)
##  highr                  0.10      2022-12-22 [2] CRAN (R 4.1.2)
##  Hmisc                  5.1-0     2023-05-08 [2] CRAN (R 4.1.2)
##  hms                    1.1.3     2023-03-21 [2] CRAN (R 4.1.2)
##  htmlTable            * 2.4.1     2022-07-07 [2] CRAN (R 4.1.2)
##  htmltools              0.5.5     2023-03-23 [2] CRAN (R 4.1.2)
##  htmlwidgets            1.6.2     2023-03-17 [2] CRAN (R 4.1.2)
##  httpuv                 1.6.11    2023-05-11 [2] CRAN (R 4.1.3)
##  httr                   1.4.6     2023-05-08 [2] CRAN (R 4.1.2)
##  huge                   1.3.5     2021-06-30 [2] CRAN (R 4.1.0)
##  igraph                 1.5.0     2023-06-16 [1] CRAN (R 4.1.3)
##  impute                 1.68.0    2021-10-26 [2] Bioconductor
##  insight                0.19.3    2023-06-29 [2] CRAN (R 4.1.3)
##  IRanges              * 2.28.0    2021-10-26 [2] Bioconductor
##  irlba                  2.3.5.1   2022-10-03 [2] CRAN (R 4.1.2)
##  iterators              1.0.14    2022-02-05 [2] CRAN (R 4.1.2)
##  jpeg                   0.1-10    2022-11-29 [2] CRAN (R 4.1.2)
##  jquerylib              0.1.4     2021-04-26 [2] CRAN (R 4.1.0)
##  jsonlite               1.8.7     2023-06-29 [2] CRAN (R 4.1.3)
##  kableExtra             1.3.4     2021-02-20 [2] CRAN (R 4.1.2)
##  KEGGREST               1.34.0    2021-10-26 [2] Bioconductor
##  KernSmooth             2.23-22   2023-07-10 [2] CRAN (R 4.1.3)
##  knitr                  1.43      2023-05-25 [2] CRAN (R 4.1.3)
##  labeling               0.4.2     2020-10-20 [2] CRAN (R 4.1.0)
##  lambda.r               1.2.4     2019-09-18 [2] CRAN (R 4.1.0)
##  later                  1.3.1     2023-05-02 [2] CRAN (R 4.1.2)
##  lattice              * 0.21-8    2023-04-05 [2] CRAN (R 4.1.2)
##  lavaan                 0.6-15    2023-03-14 [2] CRAN (R 4.1.2)
##  leaps                  3.1       2020-01-16 [2] CRAN (R 4.1.0)
##  libcoin                1.0-9     2021-09-27 [2] CRAN (R 4.1.0)
##  lifecycle              1.0.3     2022-10-07 [2] CRAN (R 4.1.2)
##  limma                  3.50.3    2022-04-07 [2] Bioconductor
##  locfit                 1.5-9.8   2023-06-11 [2] CRAN (R 4.1.3)
##  LOCOM                  1.1       2022-08-05 [2] Github (yijuanhu/LOCOM@c181e0f)
##  lubridate              1.9.2     2023-02-10 [2] CRAN (R 4.1.2)
##  magrittr             * 2.0.3     2022-03-30 [2] CRAN (R 4.1.2)
##  MASS                   7.3-60    2023-05-04 [2] CRAN (R 4.1.2)
##  Matrix                 1.6-0     2023-07-08 [2] CRAN (R 4.1.3)
##  MatrixGenerics       * 1.6.0     2021-10-26 [2] Bioconductor
##  matrixStats          * 1.0.0     2023-06-02 [2] CRAN (R 4.1.3)
##  mbzinb                 0.2       2022-03-16 [2] local
##  memoise                2.0.1     2021-11-26 [2] CRAN (R 4.1.0)
##  metagenomeSeq          1.36.0    2021-10-26 [2] Bioconductor
##  mgcv                   1.8-42    2023-03-02 [2] CRAN (R 4.1.2)
##  microbiome             1.16.0    2021-10-26 [2] Bioconductor
##  mime                   0.12      2021-09-28 [2] CRAN (R 4.1.0)
##  miniUI                 0.1.1.1   2018-05-18 [2] CRAN (R 4.1.0)
##  mixedCCA               1.6.2     2022-09-09 [2] CRAN (R 4.1.2)
##  mnormt                 2.1.1     2022-09-26 [2] CRAN (R 4.1.2)
##  modeltools             0.2-23    2020-03-05 [2] CRAN (R 4.1.0)
##  multcomp               1.4-25    2023-06-20 [2] CRAN (R 4.1.3)
##  multcompView           0.1-9     2023-04-09 [2] CRAN (R 4.1.2)
##  multtest               2.50.0    2021-10-26 [2] Bioconductor
##  munsell                0.5.0     2018-06-12 [2] CRAN (R 4.1.0)
##  mvtnorm                1.2-2     2023-06-08 [2] CRAN (R 4.1.3)
##  mycor                  0.1.1     2018-04-10 [2] CRAN (R 4.1.0)
##  NADA                   1.6-1.1   2020-03-22 [2] CRAN (R 4.1.0)
##  NetCoMi              * 1.0.3     2022-07-14 [2] Github (stefpeschel/NetCoMi@d4d80d3)
##  nlme                 * 3.1-162   2023-01-31 [2] CRAN (R 4.1.2)
##  nnet                   7.3-19    2023-05-03 [2] CRAN (R 4.1.2)
##  openssl                2.0.6     2023-03-09 [2] CRAN (R 4.1.2)
##  pbapply                1.7-2     2023-06-27 [2] CRAN (R 4.1.3)
##  pbivnorm               0.6.0     2015-01-23 [2] CRAN (R 4.1.0)
##  pcaPP                  2.0-3     2022-10-24 [2] CRAN (R 4.1.2)
##  permute              * 0.9-7     2022-01-27 [2] CRAN (R 4.1.2)
##  pheatmap             * 1.0.12    2019-01-04 [2] CRAN (R 4.1.0)
##  phyloseq             * 1.38.0    2021-10-26 [2] Bioconductor
##  picante              * 1.8.2     2020-06-10 [2] CRAN (R 4.1.0)
##  pillar                 1.9.0     2023-03-22 [2] CRAN (R 4.1.2)
##  pkgbuild               1.4.2     2023-06-26 [2] CRAN (R 4.1.3)
##  pkgconfig              2.0.3     2019-09-22 [2] CRAN (R 4.1.0)
##  pkgload                1.3.2.1   2023-07-08 [2] CRAN (R 4.1.3)
##  plyr                   1.8.8     2022-11-11 [2] CRAN (R 4.1.2)
##  png                    0.1-8     2022-11-29 [2] CRAN (R 4.1.2)
##  ppcor                  1.1       2015-12-03 [2] CRAN (R 4.1.0)
##  preprocessCore         1.56.0    2021-10-26 [2] Bioconductor
##  prettyunits            1.1.1     2020-01-24 [2] CRAN (R 4.1.0)
##  processx               3.8.2     2023-06-30 [2] CRAN (R 4.1.3)
##  profvis                0.3.8     2023-05-02 [2] CRAN (R 4.1.2)
##  promises               1.2.0.1   2021-02-11 [2] CRAN (R 4.1.0)
##  protoclust             1.6.4     2022-04-01 [2] CRAN (R 4.1.2)
##  proxy                  0.4-27    2022-06-09 [2] CRAN (R 4.1.2)
##  ps                     1.7.5     2023-04-18 [2] CRAN (R 4.1.2)
##  pscl                   1.5.5.1   2023-05-10 [2] CRAN (R 4.1.2)
##  psych                  2.3.6     2023-06-21 [2] CRAN (R 4.1.3)
##  pulsar                 0.3.10    2023-01-26 [2] CRAN (R 4.1.2)
##  purrr                * 1.0.1     2023-01-10 [2] CRAN (R 4.1.2)
##  qgraph                 1.9.5     2023-05-16 [2] CRAN (R 4.1.3)
##  quadprog               1.5-8     2019-11-20 [2] CRAN (R 4.1.0)
##  qvalue                 2.26.0    2021-10-26 [2] Bioconductor
##  R6                     2.5.1     2021-08-19 [2] CRAN (R 4.1.0)
##  RAIDA                  1.0       2022-03-14 [2] local
##  rbibutils              2.2.13    2023-01-13 [2] CRAN (R 4.1.2)
##  RColorBrewer         * 1.1-3     2022-04-03 [2] CRAN (R 4.1.2)
##  Rcpp                 * 1.0.11    2023-07-06 [1] CRAN (R 4.1.3)
##  RcppZiggurat           0.1.6     2020-10-20 [2] CRAN (R 4.1.0)
##  RCurl                  1.98-1.12 2023-03-27 [2] CRAN (R 4.1.2)
##  Rdpack                 2.4       2022-07-20 [2] CRAN (R 4.1.2)
##  readr                * 2.1.4     2023-02-10 [2] CRAN (R 4.1.2)
##  readxl               * 1.4.3     2023-07-06 [2] CRAN (R 4.1.3)
##  remotes                2.4.2     2021-11-30 [2] CRAN (R 4.1.0)
##  reshape2               1.4.4     2020-04-09 [2] CRAN (R 4.1.0)
##  reticulate             1.30      2023-06-09 [2] CRAN (R 4.1.3)
##  Rfast                  2.0.8     2023-07-03 [2] CRAN (R 4.1.3)
##  rhdf5                  2.38.1    2022-03-10 [2] Bioconductor
##  rhdf5filters           1.6.0     2021-10-26 [2] Bioconductor
##  Rhdf5lib               1.16.0    2021-10-26 [2] Bioconductor
##  rlang                  1.1.1     2023-04-28 [1] CRAN (R 4.1.2)
##  rmarkdown              2.23      2023-07-01 [2] CRAN (R 4.1.3)
##  robustbase             0.99-0    2023-06-16 [2] CRAN (R 4.1.3)
##  rootSolve              1.8.2.3   2021-09-29 [2] CRAN (R 4.1.0)
##  rpart                  4.1.19    2022-10-21 [2] CRAN (R 4.1.2)
##  RSpectra               0.16-1    2022-04-24 [2] CRAN (R 4.1.2)
##  RSQLite                2.3.1     2023-04-03 [2] CRAN (R 4.1.2)
##  rstatix                0.7.2     2023-02-01 [2] CRAN (R 4.1.2)
##  rstudioapi             0.15.0    2023-07-07 [2] CRAN (R 4.1.3)
##  Rtsne                  0.16      2022-04-17 [2] CRAN (R 4.1.2)
##  RVenn                  1.1.0     2019-07-18 [2] CRAN (R 4.1.0)
##  rvest                  1.0.3     2022-08-19 [2] CRAN (R 4.1.2)
##  S4Vectors            * 0.32.4    2022-03-29 [2] Bioconductor
##  sandwich               3.0-2     2022-06-15 [2] CRAN (R 4.1.2)
##  sass                   0.4.6     2023-05-03 [2] CRAN (R 4.1.2)
##  scales                 1.2.1     2022-08-20 [2] CRAN (R 4.1.2)
##  scatterplot3d          0.3-44    2023-05-05 [2] CRAN (R 4.1.2)
##  sessioninfo            1.2.2     2021-12-06 [2] CRAN (R 4.1.0)
##  sf                     1.0-7     2022-03-07 [2] CRAN (R 4.1.2)
##  shape                  1.4.6     2021-05-19 [2] CRAN (R 4.1.0)
##  shiny                  1.7.4.1   2023-07-06 [2] CRAN (R 4.1.3)
##  sjlabelled             1.2.0     2022-04-10 [2] CRAN (R 4.1.2)
##  sjmisc                 2.8.9     2021-12-03 [2] CRAN (R 4.1.0)
##  snow                   0.4-4     2021-10-27 [2] CRAN (R 4.1.0)
##  SpiecEasi            * 1.1.2     2022-07-14 [2] Github (zdk123/SpiecEasi@c463727)
##  SPRING                 1.0.4     2022-08-03 [2] Github (GraceYoon/SPRING@3d641a4)
##  stringi                1.7.12    2023-01-11 [2] CRAN (R 4.1.2)
##  stringr                1.5.0     2022-12-02 [2] CRAN (R 4.1.2)
##  SummarizedExperiment * 1.24.0    2021-10-26 [2] Bioconductor
##  survival               3.5-5     2023-03-12 [2] CRAN (R 4.1.2)
##  svglite                2.1.1     2023-01-10 [2] CRAN (R 4.1.2)
##  systemfonts            1.0.4     2022-02-11 [2] CRAN (R 4.1.2)
##  tensorA                0.36.2    2020-11-19 [2] CRAN (R 4.1.0)
##  TH.data                1.1-2     2023-04-17 [2] CRAN (R 4.1.2)
##  tibble               * 3.2.1     2023-03-20 [2] CRAN (R 4.1.2)
##  tidyr                * 1.3.0     2023-01-24 [2] CRAN (R 4.1.2)
##  tidyselect             1.2.0     2022-10-10 [2] CRAN (R 4.1.2)
##  timechange             0.2.0     2023-01-11 [2] CRAN (R 4.1.2)
##  truncnorm              1.0-9     2023-03-20 [2] CRAN (R 4.1.2)
##  tzdb                   0.4.0     2023-05-12 [2] CRAN (R 4.1.3)
##  umap                   0.2.10.0  2023-02-01 [2] CRAN (R 4.1.2)
##  units                  0.8-2     2023-04-27 [2] CRAN (R 4.1.2)
##  urlchecker             1.0.1     2021-11-30 [2] CRAN (R 4.1.0)
##  usethis              * 2.2.2     2023-07-06 [2] CRAN (R 4.1.3)
##  utf8                   1.2.3     2023-01-31 [2] CRAN (R 4.1.2)
##  uuid                   1.1-0     2022-04-19 [2] CRAN (R 4.1.2)
##  vctrs                  0.6.3     2023-06-14 [1] CRAN (R 4.1.3)
##  vegan                * 2.6-4     2022-10-11 [2] CRAN (R 4.1.2)
##  VennDiagram          * 1.7.3     2022-04-12 [2] CRAN (R 4.1.2)
##  VGAM                   1.1-8     2023-03-09 [2] CRAN (R 4.1.2)
##  viridis              * 0.6.3     2023-05-03 [2] CRAN (R 4.1.2)
##  viridisLite          * 0.4.2     2023-05-02 [2] CRAN (R 4.1.2)
##  vroom                  1.6.3     2023-04-28 [2] CRAN (R 4.1.2)
##  webshot                0.5.5     2023-06-26 [2] CRAN (R 4.1.3)
##  WGCNA                  1.72-1    2023-01-18 [2] CRAN (R 4.1.2)
##  withr                  2.5.0     2022-03-03 [2] CRAN (R 4.1.2)
##  Wrench                 1.12.0    2021-10-26 [2] Bioconductor
##  xfun                   0.40      2023-08-09 [1] CRAN (R 4.1.3)
##  XMAS2                * 2.2.0     2023-10-27 [1] local
##  XML                    3.99-0.14 2023-03-19 [2] CRAN (R 4.1.2)
##  xml2                   1.3.5     2023-07-06 [2] CRAN (R 4.1.3)
##  xtable                 1.8-4     2019-04-21 [2] CRAN (R 4.1.0)
##  XVector                0.34.0    2021-10-26 [2] Bioconductor
##  yaml                   2.3.7     2023-01-23 [2] CRAN (R 4.1.2)
##  zCompositions          1.4.0-1   2022-03-26 [2] CRAN (R 4.1.2)
##  zlibbioc               1.40.0    2021-10-26 [2] Bioconductor
##  zoo                    1.8-12    2023-04-13 [2] CRAN (R 4.1.2)
## 
##  [1] /Users/zouhua/Library/R/x86_64/4.1/library
##  [2] /Library/Frameworks/R.framework/Versions/4.1/Resources/library
## 
## ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="network-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/MGS_DataAnalysis_template/blob/main/19-Example.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["MGS microbiota data analysis based on XMAS package.pdf", "MGS microbiota data analysis based on XMAS package.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true,
"toc_depth": 3
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
